\chapter*{List of Abbreviations}

\begin{longtable}[l]{@{}p{2.5cm}p{12cm}@{}}
\textbf{017} & Darunavir \\
\textbf{3D} & Three-dimensional \\
\textbf{8PR} & Paroxetine \\
\textbf{AA} & Amino acid \\
\textbf{ADMET} & Absorption, distribution, metabolism, excretion, toxicity \\
\textbf{ADP} & Adenosine-5'-diphosphate \\
\textbf{AMI} & Adjusted mutual information \\
\textbf{ANN} & Artificial neural network \\
\textbf{AP} & Average precision \\
\textbf{ARI} & Adjusted Rand index \\
\textbf{ASA} & Accessible surface area \\
\textbf{ATP} & Adenosine-5’-triphosphate \\
\textbf{AUC} & Area under the curve \\
\textbf{BGC} & Glucose \\
\textbf{BLOSUM} & Block substitution matrix \\
\textbf{CC9} & Curcumin \\
\textbf{CCD} & Chemical component dictionary \\
\textbf{CHI} & Calinski-Harabasz index \\
\textbf{CI} & Confidence interval \\
\textbf{CLR} & Cholesterol \\
\textbf{CM} & Centre of mass \\
\textbf{CMD} & Conserved and missense depleted \\
\textbf{CME} & Conserved and missense enriched \\
\textbf{CNN} & Convolutional neural network \\
\textbf{Cryo-EM} & Cryogenic electron microscopy \\
\textbf{DBI} & Davies-Bouldin index \\
\textbf{DNA} & Deoxyribonucleic acid \\
\textbf{DBSCAN} & Density-based spatial clustering of applications with noise \\
\textbf{DCA} & Distance to closest ligand atom \\
\textbf{DCC} & Distance centroid to centroid \\
\textbf{DNN} & Deep neural network \\
\textbf{DRN} & Deep residual network \\
\textbf{DWF} & Debye-Waller factor \\
\textbf{EDO} & Etylene glycol \\
\textbf{EGNN} & Equivariant graph neural network \\
\textbf{EMA} & European Medicines Agency \\
\textbf{ENA} & European Nucleotide Archive \\
\textbf{FAD} & Flavin-adenine dinucleotide \\
\textbf{FBDD} & Fragment-based drug discovery \\
\textbf{FDA} & Food and Drug Administration \\
\textbf{FDR} & False discovery rate \\
\textbf{FMN} & Flavin mononucleotide \\
\textbf{FN} & False negative \\
\textbf{FP} & False positive \\
\textbf{FPR} & False positive rate \\
\textbf{FS} & Fragment screening \\
\textbf{FUC} & Fucose \\
\textbf{GAL} & Galactose \\
\textbf{GAT} & Graph attention network \\
\textbf{GNN} & Graph neural network \\
\textbf{gnomAD} & Genome aggregation database \\
\textbf{GOL} & Glycerol \\
\textbf{GPCR} & G-protein coupled receptor \\
\textbf{GSH} & Glutathione \\
\textbf{GTP} & Guanosine-5′-triphosphate \\
\textbf{HAP} & HOLO4K-AlphaFold2 paired \\
\textbf{HEM} & Haem \\
\textbf{HOH} & Water \\
\textbf{HPC} & High performance computing \\
\textbf{HTS} & High-throughput screening \\
\textbf{IDR} & Intrinsically disordered region \\
\textbf{IOU} & Intersection over union \\
\textbf{IQR} & Interquartile range \\
\textbf{\textit{I\textsubscript{rel}}} & Relative intersection \\
\textbf{JFP} & N-(4-methyl-1,3-thiazol-2-yl)propanamide \\
\textbf{JI} & Jaccard index \\
\textbf{LBVS} & Ligand-based virtual screening \\
\textbf{LGBM} & Light gradient boosting machine \\
\textbf{LJ} & Lennard-Jones \\
\textbf{MAN} & Mannose \\
\textbf{MCC} & Matthews correlation coefficient \\
\textbf{MCD} & Minimum centroid distance \\
\textbf{MDS} & Multi dimensional scaling \\
\textbf{MES} & Missense enrichment score \\
\textbf{MHRA} & Medicines and Healthcare products Regulatory Agency \\
\textbf{ML} & Machine learning \\
\textbf{MLP} & Multi layer perceptron \\
\textbf{MOAD} & Mother Of All Databases \\
\textbf{mRNA} & Messenger RNA \\
\textbf{MRO} & Maximum residue overlap \\
\textbf{MTD} & Maximum tolerated dose \\
\textbf{NAD} & Nicotinamide-adenine-dinucleotide \\
\textbf{NAP} & Nicotinamide-adenine-dinucleotide phosphate \\
\textbf{NME} & New molecular entity \\
\textbf{\textit{N}-mer} & Protein peptide of \textit{N} amino acids \\
\textbf{NMR} & Nuclear magnetic resonance \\
\textbf{NOAEL} & No observed adverse effect level \\
\textbf{NR} & Non-redundant \\
\textbf{OR} & Odds ratio \\
\textbf{PAM} & Point accepted mutation \\
\textbf{PanDDA} & Pan-dataset density analysis \\
\textbf{PCA} & Principal component analysis \\
\textbf{PD} & Pharmacodynamics \\ 
\textbf{PDB} & Protein Data Bank \\
\textbf{PDBe} & Protein Data Bank Europe \\
\textbf{PDBe-KB} & Protein Data Bank Europe Knowledge Base \\
\textbf{PK} & Pharmacokinetics \\
\textbf{PLP} & Vitamin B6 phosphate \\
\textbf{PR} & Precision-recall curve \\
\textbf{QSAR} & Quantitative structure-activity relationship \\
\textbf{\textit{R\textsubscript{g}}} & Radius of gyration \\
\textbf{ROC} & Receiver operating characteristic \\
\textbf{RMSD} & Root mean square deviation \\
\textbf{RNA} & Ribonucleic acid \\
\textbf{RRO} & Relative residue overlap \\
\textbf{RVO} & Relative volume overlap \\
\textbf{RSA} & Relative solvent accessibility \\
\textbf{SAH} & S-adenosyl-L-homocysteine \\
\textbf{SAS} & Solvent accessible surface \\
\textbf{SASA} & Solvent accessible surface area \\
\textbf{SBVS} & Structure-based virtual screening \\
\textbf{SD} & Standard deviation \\
\textbf{SDP} & Specificity-determining position \\
\textbf{SES} & Solvent-excluded surface \\
\textbf{SS} & Sum of squares \\
\textbf{TN} & True negative \\
\textbf{TP} & True positive \\
\textbf{TPR} & True positive rate \\
\textbf{TRS} & Tris buffer \\
\textbf{\textit{U\textsubscript{D}}} & Distance U \\
\textbf{\textit{U\textsubscript{max}}} & Maximum U statistic \\
\textbf{UMD} & Unconserved and missense depleted \\
\textbf{UME} & Unconserved and missense enriched \\
\textbf{UPGMA} & Unweighted pair-group method with arithmetic mean \\
\textbf{\textit{U\textsubscript{rel}}} & Relative U statistic \\
\textbf{VDW} & Van der Waals \\
\textbf{VN} & Virtual node \\
\textbf{\textit{V\textsubscript{R}}} & Volume ratio \\
\textbf{VS} & Virtual screening \\
\textbf{VUS} & Variant of uncertain significance \\
\textbf{XRC} & X-ray crystallography \\
\textbf{XYP} & Xylose \\

\end{longtable}
