\chapter*{List of Abbreviations}

\begin{longtable}[l]{@{}p{2.5cm}p{12cm}@{}}
\textbf{\large \#} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/017}{017}} & Darunavir \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/0XR}{0XR}} & Ethyl caffeate \\
\textmd{\href{https://en.wikipedia.org/wiki/Three-dimensional_space}{3D}} & Three-dimensional \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/8PR}{8PR}} & Paroxetine \\[0.3175cm]
\textbf{\large A} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Adenine}{A}} & Adenine \\
\textmd{\href{https://en.wikipedia.org/wiki/Amino_acid}{AA}} & Amino acid \\
\textmd{\href{https://en.wikipedia.org/wiki/ADME}{ADMET}} & Absorption, distribution, metabolism, excretion, toxicity \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/ADP}{ADP}} & Adenosine-5'-diphosphate \\
\textmd{\href{https://alphafold.ebi.ac.uk/}{AFDB}} & AlphaFold Protein Structure Database \\
\textmd{\href{https://en.wikipedia.org/wiki/Adjusted_mutual_information}{AMI}} & Adjusted mutual information \\
\textmd{\href{https://en.wikipedia.org/wiki/Neural_network_(machine_learning)}{ANN}} & Artificial neural network \\
\textmd{\href{https://en.wikipedia.org/wiki/Evaluation_measures_(information_retrieval)}{AP}} & Average precision \\
\textmd{\href{https://en.wikipedia.org/wiki/Rand_index}{ARI}} & Adjusted Rand index \\
\textmd{\href{https://en.wikipedia.org/wiki/Accessible_surface_area}{ASA}} & Accessible surface area \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/ATP}{ATP}} & Adenosine-5’-triphosphate \\
\textmd{\href{https://en.wikipedia.org/wiki/Receiver_operating_characteristic}{AUC}} & Area under the curve \\[0.3175cm]
\textbf{\large B} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/BGC}{BGC}} & Glucose \\
\textmd{\href{https://en.wikipedia.org/wiki/BLOSUM}{BLOSUM}} & Block substitution matrix \\[0.3175cm]
\textbf{\large C} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Cytosine}{C}} & Cytosine \\
\textmd{\href{https://en.wikipedia.org/wiki/Amino_acid\#Chirality}{CA}} & α-carbon \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/CC9}{CC9}} & Curcumin \\
\textmd{\href{https://www.wwpdb.org/data/ccd}{CCD}} & Chemical component dictionary \\
\textmd{\href{https://en.wikipedia.org/wiki/Calinski–Harabasz_index}{CHI}} & Calinski-Harabasz index \\
\textmd{\href{https://en.wikipedia.org/wiki/Confidence_interval}{CI}} & Confidence interval \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/CLA}{CLA}} & Chlorophyll A \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/CLR}{CLR}} & Cholesterol \\
\textmd{\href{https://en.wikipedia.org/wiki/Center_of_mass}{CM}} & Centre of mass \\
\textmd{\href{https://www.nature.com/articles/s42003-024-06117-5}{CMD}} & Conserved and missense-depleted \\
\textmd{\href{https://www.nature.com/articles/s42003-024-06117-5}{CME}} & Conserved and missense-enriched \\
\textmd{\href{https://en.wikipedia.org/wiki/Convolutional_neural_network}{CNN}} & Convolutional neural network \\
\textmd{\href{https://en.wikipedia.org/wiki/Carboxylic_acid}{COOH}} & Carboxyl group \\
\textmd{\href{https://en.wikipedia.org/wiki/Cryogenic_electron_microscopy}{Cryo-EM}} & Cryogenic electron microscopy \\[0.3175cm]
\textbf{\large D} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Davies–Bouldin_index}{DBI}} & Davies-Bouldin index \\
\textmd{\href{https://en.wikipedia.org/wiki/DNA}{DNA}} & Deoxyribonucleic acid \\
\textmd{\href{https://en.wikipedia.org/wiki/DBSCAN}{DBSCAN}} & Density-based spatial clustering of applications with noise \\
\textmd{\href{https://academic.oup.com/bioinformatics/article/37/12/1681/6104838}{DCA}} & Distance to closest ligand atom \\
\textmd{\href{https://academic.oup.com/bioinformatics/article/37/12/1681/6104838}{DCC}} & Distance centroid to centroid \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/DGD}{DGD}} & Digalactosyl diacyl glycerol \\
\textmd{\href{https://en.wikipedia.org/wiki/Deep_learning\#Deep_neural_networks}{DNN}} & Deep neural network \\
\textmd{\href{https://en.wikipedia.org/wiki/Residual_neural_network}{DRN}} & Deep residual network \\
\textmd{\href{https://onlinelibrary.wiley.com/doi/10.1002/prot.340110408}{DS}} & Divergence score \\
\textmd{\href{https://en.wikipedia.org/wiki/DSSP_(algorithm)}{DSSP}} & Define secondary structure of proteins \\
\textmd{\href{https://en.wikipedia.org/wiki/Debye–Waller_factor}{DWF}} & Debye-Waller factor \\[0.3175cm]
\textbf{\large E} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/EDO}{EDO}} & Etylene glycol \\
\textmd{\href{https://arxiv.org/abs/2404.07194}{EGNN}} & Equivariant graph neural network \\
\textmd{\href{https://en.wikipedia.org/wiki/European_Medicines_Agency}{EMA}} & European Medicines Agency \\
\textmd{\href{https://en.wikipedia.org/wiki/European_Nucleotide_Archive}{ENA}} & European Nucleotide Archive \\
\textmd{\href{https://en.wikipedia.org/wiki/Electric_potential}{ESP}} & Electrostatic potential \\[0.3175cm]
\textbf{\large F} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/FAD}{FAD}} & Flavin-adenine dinucleotide \\
\textmd{\href{https://en.wikipedia.org/wiki/Fragment-based_lead_discovery}{FBDD}} & Fragment-based drug discovery \\
\textmd{\href{https://en.wikipedia.org/wiki/Food_and_Drug_Administration}{FDA}} & Food and Drug Administration \\
\textmd{\href{https://en.wikipedia.org/wiki/False_discovery_rate}{FDR}} & False discovery rate \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/FMN}{FMN}} & Flavin mononucleotide \\
\textmd{\href{https://en.wikipedia.org/wiki/False_positives_and_false_negatives}{FN}} & False negative \\
\textmd{\href{https://en.wikipedia.org/wiki/False_positives_and_false_negatives}{FP}} & False positive \\
\textmd{\href{https://en.wikipedia.org/wiki/False_positive_rate}{FPR}} & False positive rate \\
\textmd{\href{https://en.wikipedia.org/wiki/Fragment-based_lead_discovery}{FS}} & Fragment screening \\
\textmd{\href{https://www.compbio.dundee.ac.uk/ligysis/help}{FS}} & Functional score \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/FUC}{FUC}} & Fucose \\[0.3175cm]
\textbf{\large G} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Guanine}{G}} & Guanine \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/GAL}{GAL}} & Galactose \\
\textmd{\href{https://en.wikipedia.org/wiki/Graph_neural_network\#Graph_attention_network}{GAT}} & Graph attention network \\
\textmd{\href{https://en.wikipedia.org/wiki/Graph_neural_network}{GNN}} & Graph neural network \\
\textmd{\href{https://gnomad.broadinstitute.org/}{gnomAD}} & Genome aggregation database \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/GOL}{GOL}} & Glycerol \\
\textmd{\href{https://en.wikipedia.org/wiki/G_protein-coupled_receptor}{GPCR}} & G-protein coupled receptor \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/GSH}{GSH}} & Glutathione \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/GTP}{GTP}} & Guanosine-5′-triphosphate \\[0.3175cm]
\textbf{\large H} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Hydrogen}{H}} & Hydrogen \\
\textmd{\href{https://jcheminf.biomedcentral.com/articles/10.1186/s13321-024-00821-4}{HAP}} & HOLO4K-AlphaFold2 paired \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/HEM}{HEM}} & Haem \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/HOH}{HOH}} & Water \\
\textmd{\href{https://en.wikipedia.org/wiki/High-performance_computing}{HPC}} & High-performance computing \\
\textmd{\href{https://en.wikipedia.org/wiki/High-throughput_screening}{HTS}} & High-throughput screening \\[0.3175cm]
\textbf{\large I} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Intrinsically_disordered_proteins}{IDR}} & Intrinsically disordered region \\
\textmd{\href{https://en.wikipedia.org/wiki/Jaccard_index}{IOU}} & Intersection over union \\
\textmd{\href{https://en.wikipedia.org/wiki/Interquartile_range}{IQR}} & Interquartile range \\
\textmd{\href{https://www.nature.com/articles/s42003-024-05970-8}{\textit{I\textsubscript{rel}}}} & Relative intersection \\[0.3175cm]
\textbf{\large J} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/JFP}{JFP}} & N-(4-methyl-1,3-thiazol-2-yl)propanamide \\
\textmd{\href{https://en.wikipedia.org/wiki/Jaccard_index}{JI}} & Jaccard index \\[0.3175cm]
\textbf{\large L} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Virtual_screening\#Ligand-based_methods_2}{LBVS}} & Ligand-based virtual screening \\
\textmd{\href{https://en.wikipedia.org/wiki/LightGBM}{LGBM}} & Light gradient boosting machine \\
\textmd{\href{https://en.wikipedia.org/wiki/Lennard-Jones_potential}{LJ}} & Lennard-Jones \\[0.3175cm]
\textbf{\large M} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/MAFFT}{MAFFT}} & Multiple alignment using fast Fourier transform \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/MAN}{MAN}} & Mannose \\
\textmd{\href{https://en.wikipedia.org/wiki/Relative_accessible_surface_area}{MaxASA}} & Maximum accessible surface area \\
\textmd{\href{https://en.wikipedia.org/wiki/Phi_coefficient}{MCC}} & Matthews correlation coefficient \\
\textmd{\href{https://jcheminf.biomedcentral.com/articles/10.1186/s13321-024-00923-z}{MCD}} & Minimum centroid distance \\
\textmd{\href{https://en.wikipedia.org/wiki/Multidimensional_scaling}{MDS}} & Multi-dimensional scaling \\
\textmd{\href{https://www.nature.com/articles/s42003-024-06117-5}{MES}} & Missense enrichment score \\
\textmd{\href{https://en.wikipedia.org/wiki/Medicines_and_Healthcare_products_Regulatory_Agency}{MHRA}} & Medicines and Healthcare products Regulatory Agency \\
\textmd{\href{https://en.wikipedia.org/wiki/Machine_learning}{ML}} & Machine learning \\
\textmd{\href{https://www.cgl.ucsf.edu/chimera/data/mlp-sep2016/index.html}{MLP}} & Molecular lipophilicity potential\\
\textmd{\href{https://en.wikipedia.org/wiki/Multilayer_perceptron}{MLP}} & Multi-layer perceptron \\
\textmd{\href{https://www.nature.com/articles/s41598-023-29996-w}{MOAD}} & Mother Of All Databases \\
\textmd{\href{https://en.wikipedia.org/wiki/Messenger_RNA}{mRNA}} & Messenger RNA \\
\textmd{\href{https://jcheminf.biomedcentral.com/articles/10.1186/s13321-024-00923-z}{MRO}} & Maximum residue overlap \\
\textmd{\href{https://en.wikipedia.org/wiki/Multiple_sequence_alignment}{MSA}} & Multiple sequence alignment \\
\textmd{\href{https://en.wikipedia.org/wiki/Therapeutic_index\#Maximum_tolerated_dose}{MTD}} & Maximum tolerated dose \\
\textmd{\href{https://en.wikipedia.org/wiki/MUSCLE_(alignment_software)}{MUSCLE}} & Multiple sequence comparison by log-expectation  \\[0.3175cm]
\textbf{\large N} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Nitrogen}{N}} & Nitrogen \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/NAD}{NAD}} & Nicotinamide-adenine-dinucleotide \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/NAP}{NAP}} & Nicotinamide-adenine-dinucleotide phosphate \\
\textmd{\href{https://en.wikipedia.org/wiki/Amine}{NH2}} & Amino group \\
\textmd{\href{https://jcheminf.biomedcentral.com/articles/10.1186/s13321-024-00923-z}{NI}} & No ions \\
\textmd{\href{https://en.wikipedia.org/wiki/New_chemical_entity}{NME}} & New molecular entity \\
\textmd{\href{https://en.wikipedia.org/wiki/Peptide}{\textit{N}-mer}} & Protein peptide of \textit{N} amino acids \\
\textmd{\href{https://en.wikipedia.org/wiki/Nuclear_magnetic_resonance_spectroscopy}{NMR}} & Nuclear magnetic resonance \\
\textmd{\href{https://en.wikipedia.org/wiki/No-observed-adverse-effect_level}{NOAEL}} & No observed adverse effect level \\
\textmd{\href{https://jcheminf.biomedcentral.com/articles/10.1186/s13321-024-00923-z}{NR}} & Non-redundant \\[0.3175cm]
\textbf{\large O} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Oxygen}{O}} & Oxygen \\
\textmd{\href{https://en.wikipedia.org/wiki/Odds_ratio}{OR}} & Odds ratio \\[0.3175cm]
\textbf{\large P} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Point_accepted_mutation}{PAM}} & Point accepted mutation \\
\textmd{\href{https://www.nature.com/articles/ncomms15123}{PanDDA}} & Pan-dataset density analysis \\
\textmd{\href{https://en.wikipedia.org/wiki/Principal_component_analysis}{PCA}} & Principal component analysis \\
\textmd{\href{https://en.wikipedia.org/wiki/Pharmacodynamics}{PD}} & Pharmacodynamics \\ 
\textmd{\href{https://en.wikipedia.org/wiki/Protein_Data_Bank}{PDB}} & Protein Data Bank \\
\textmd{\href{https://www.ebi.ac.uk/pdbe/}{PDBe}} & Protein Data Bank Europe \\
\textmd{\href{https://www.ebi.ac.uk/pdbe/pdbe-kb/}{PDBe-KB}} & Protein Data Bank Europe Knowledge Base \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/PID}{PID}} & Peridinin \\
\textmd{\href{https://en.wikipedia.org/wiki/Pharmacokinetics}{PK}} & Pharmacokinetics \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/PLP}{PLP}} & Vitamin B6 phosphate \\
\textmd{\href{https://en.wikipedia.org/wiki/Precision_and_recall}{PR}} & Precision-recall curve \\
\textmd{\href{https://en.wikipedia.org/wiki/Post-translational_modification}{PTM}} & Post-translational modification \\[0.3175cm]
\textbf{\large Q} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Quantitative_structure–activity_relationship}{QSAR}} & Quantitative structure-activity relationship \\[0.3175cm]
\textbf{\large R} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Amino_acid\#Side_chains}{R}} & Side chain \\
\textmd{\href{https://en.wikipedia.org/wiki/Radius_of_gyration}{\textit{R\textsubscript{g}}}} & Radius of gyration \\
\textmd{\href{https://en.wikipedia.org/wiki/Receiver_operating_characteristic}{ROC}} & Receiver operating characteristic \\
\textmd{\href{https://en.wikipedia.org/wiki/Root_mean_square_deviation}{RMSD}} & Root mean square deviation \\
\textmd{\href{https://en.wikipedia.org/wiki/RNA}{RNA}} & Ribonucleic acid \\
\textmd{\href{https://jcheminf.biomedcentral.com/articles/10.1186/s13321-024-00923-z}{RRO}} & Relative residue overlap \\
\textmd{\href{https://jcheminf.biomedcentral.com/articles/10.1186/s13321-024-00923-z}{RVO}} & Relative volume overlap \\
\textmd{\href{https://en.wikipedia.org/wiki/Relative_accessible_surface_area}{RSA}} & Relative solvent accessibility \\[0.3175cm]
\textbf{\large S} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/SAH}{SAH}} & S-adenosyl-L-homocysteine \\
\textmd{\href{https://en.wikipedia.org/wiki/Accessible_surface_area}{SAS}} & Solvent accessible surface \\
\textmd{\href{https://en.wikipedia.org/wiki/Accessible_surface_area}{SASA}} & Solvent accessible surface area \\
\textmd{\href{https://en.wikipedia.org/wiki/Virtual_screening\#Structure-based_methods_known_protein_ligand_docking}{SBVS}} & Structure-based virtual screening \\
\textmd{\href{https://en.wikipedia.org/wiki/Standard_deviation}{SD}} & Standard deviation \\
\textmd{\href{https://www.nature.com/articles/s42003-024-06117-5}{SDP}} & Specificity-determining position \\
\textmd{\href{https://en.wikipedia.org/wiki/Accessible_surface_area\#Relation_to_solvent-excluded_surface}{SES}} & Solvent excluded surface \\
\textmd{\href{https://en.wikipedia.org/wiki/Single-nucleotide_polymorphism}{SNP}} & Single nucleotide polymorphism \\
\textmd{\href{https://en.wikipedia.org/wiki/Sum_of_squares}{SS}} & Sum of squares \\[0.3175cm]
\textbf{\large T} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Thymine}{T}} & Thymine \\
\textmd{\href{https://en.wikipedia.org/wiki/False_positives_and_false_negatives}{TN}} & True negative \\
\textmd{\href{https://en.wikipedia.org/wiki/False_positives_and_false_negatives}{TP}} & True positive \\
\textmd{\href{https://en.wikipedia.org/wiki/TPR}{TPR}} & True positive rate \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/TRS}{TRS}} & Tris buffer \\[0.3175cm]
\textbf{\large U} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Uracil}{U}} & Uracil \\
\textmd{\href{https://www.nature.com/articles/s42003-024-05970-8}{\textit{U\textsubscript{D}}}} & Distance U \\
\textmd{\href{https://www.nature.com/articles/s42003-024-05970-8}{\textit{U\textsubscript{max}}}} & Maximum U statistic \\
\textmd{\href{https://www.nature.com/articles/s42003-024-06117-5}{UMD}} & Unconserved and missense depleted \\
\textmd{\href{https://www.nature.com/articles/s42003-024-06117-5}{UME}} & Unconserved and missense enriched \\
\textmd{\href{https://en.wikipedia.org/wiki/UPGMA}{UPGMA}} & Unweighted pair-group method with arithmetic mean \\
\textmd{\href{https://www.uniprot.org/help/uniprotkb}{UPKB}} & UniProt Knowledgebase \\
\textmd{\href{https://www.nature.com/articles/s42003-024-05970-8}{\textit{U\textsubscript{rel}}}} & Relative U statistic \\[0.3175cm]
\textbf{\large V} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Van_der_Waals_force}{VDW}} & Van der Waals \\
\textmd{\href{https://arxiv.org/abs/2404.07194}{VN}} & Virtual node \\
\textmd{\href{https://jcheminf.biomedcentral.com/articles/10.1186/s13321-024-00923-z}{\textit{V\textsubscript{R}}}} & Volume ratio \\
\textmd{\href{https://en.wikipedia.org/wiki/Virtual_screening}{VS}} & Virtual screening \\
\textmd{\href{https://en.wikipedia.org/wiki/Variant_of_uncertain_significance}{VUS}} & Variant of uncertain significance \\[0.3175cm]
\textbf{\large X} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/X-ray_crystallography}{XRC}} & X-ray crystallography \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/XYP}{XYP}} & Xylose \\[0.3175cm]

\end{longtable}
