\chapter*{List of Abbreviations}

\begin{longtable}[l]{@{}p{2.5cm}p{12cm}@{}}
\textbf{\large \#} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/017}{017}} & Darunavir \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/0XR}{0XR}} & Ethyl caffeate \\
\textmd{\href{https://en.wikipedia.org/wiki/Three-dimensional_space}{3D}} & Three-dimensional \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/8PR}{8PR}} & Paroxetine \\[0.3175cm]
\textbf{\large A} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Amino_acid}{AA}} & Amino acid \\
\textmd{\href{https://en.wikipedia.org/wiki/ADME}{ADMET}} & Absorption, distribution, metabolism, excretion, toxicity \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/ADP}{ADP}} & Adenosine-5'-diphosphate \\
\textmd{\href{https://en.wikipedia.org/wiki/Adjusted_mutual_information}{AMI}} & Adjusted mutual information \\
\textmd{\href{https://en.wikipedia.org/wiki/Neural_network_(machine_learning)}{ANN}} & Artificial neural network \\
\textmd{\href{https://en.wikipedia.org/wiki/Evaluation_measures_(information_retrieval)}{AP}} & Average precision \\
\textmd{\href{https://en.wikipedia.org/wiki/Rand_index}{ARI}} & Adjusted Rand index \\
\textmd{\href{https://en.wikipedia.org/wiki/Accessible_surface_area}{ASA}} & Accessible surface area \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/ATP}{ATP}} & Adenosine-5’-triphosphate \\
\textmd{\href{https://en.wikipedia.org/wiki/Receiver_operating_characteristic}{AUC}} & Area under the curve \\[0.3175cm]
\textbf{\large B} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/BGC}{BGC}} & Glucose \\
\textmd{\href{https://en.wikipedia.org/wiki/BLOSUM}{BLOSUM}} & Block substitution matrix \\[0.3175cm]
\textbf{\large C} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/CC9}{CC9}} & Curcumin \\
\textmd{\href{https://www.wwpdb.org/data/ccd}{CCD}} & Chemical component dictionary \\
\textmd{\href{https://en.wikipedia.org/wiki/Calinski–Harabasz_index}{CHI}} & Calinski-Harabasz index \\
\textmd{\href{https://en.wikipedia.org/wiki/Confidence_interval}{CI}} & Confidence interval \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/CLA}{CLA}} & Chlorophyll A \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/CLR}{CLR}} & Cholesterol \\
\textmd{\href{https://en.wikipedia.org/wiki/Center_of_mass}{CM}} & Centre of mass \\
\textmd{CMD} & Conserved and missense-depleted \\
\textmd{CME} & Conserved and missense-enriched \\
\textmd{\href{https://en.wikipedia.org/wiki/Convolutional_neural_network}{CNN}} & Convolutional neural network \\
\textmd{\href{https://en.wikipedia.org/wiki/Cryogenic_electron_microscopy}{Cryo-EM}} & Cryogenic electron microscopy \\[0.3175cm]
\textbf{\large D} & \\[0.25cm]
\textmd{\href{https://en.wikipedia.org/wiki/Davies–Bouldin_index}{DBI}} & Davies-Bouldin index \\
\textmd{DNA} & Deoxyribonucleic acid \\
\textmd{DBSCAN} & Density-based spatial clustering of applications with noise \\
\textmd{DCA} & Distance to closest ligand atom \\
\textmd{DCC} & Distance centroid to centroid \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/DGD}{DGD}} & Digalactosyl diacyl glycerol \\
\textmd{DNN} & Deep neural network \\
\textmd{DRN} & Deep residual network \\
\textmd{DWF} & Debye-Waller factor \\[0.3175cm]
\textbf{\large E} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/EDO}{EDO}} & Etylene glycol \\
\textmd{EGNN} & Equivariant graph neural network \\
\textmd{EMA} & European Medicines Agency \\
\textmd{ENA} & European Nucleotide Archive \\[0.3175cm]
\textbf{\large F} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/FAD}{FAD}} & Flavin-adenine dinucleotide \\
\textmd{FBDD} & Fragment-based drug discovery \\
\textmd{FDA} & Food and Drug Administration \\
\textmd{FDR} & False discovery rate \\
\textmd{FMN} & Flavin mononucleotide \\
\textmd{FN} & False negative \\
\textmd{FP} & False positive \\
\textmd{FPR} & False positive rate \\
\textmd{FS} & Fragment screening \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/FUC}{FUC}} & Fucose \\[0.3175cm]
\textbf{\large G} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/GAL}{GAL}} & Galactose \\
\textmd{GAT} & Graph attention network \\
\textmd{GNN} & Graph neural network \\
\textmd{gnomAD} & Genome aggregation database \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/GOL}{GOL}} & Glycerol \\
\textmd{GPCR} & G-protein coupled receptor \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/GSH}{GSH}} & Glutathione \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/GTP}{GTP}} & Guanosine-5′-triphosphate \\[0.3175cm]
\textbf{\large H} & \\[0.25cm]
\textmd{HAP} & HOLO4K-AlphaFold2 paired \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/HEM}{HEM}} & Haem \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/HOH}{HOH}} & Water \\
\textmd{HPC} & High-performance computing \\
\textmd{HTS} & High-throughput screening \\[0.3175cm]
\textbf{\large I} & \\[0.25cm]
\textmd{IDR} & Intrinsically disordered region \\
\textmd{IOU} & Intersection over union \\
\textmd{IQR} & Interquartile range \\
\textmd{\textit{I\textsubscript{rel}}} & Relative intersection \\[0.3175cm]
\textbf{\large J} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/JFP}{JFP}} & N-(4-methyl-1,3-thiazol-2-yl)propanamide \\
\textmd{JI} & Jaccard index \\[0.3175cm]
\textbf{\large L} & \\[0.25cm]
\textmd{LBVS} & Ligand-based virtual screening \\
\textmd{LGBM} & Light gradient boosting machine \\
\textmd{LJ} & Lennard-Jones \\[0.3175cm]
\textbf{\large M} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/MAN}{MAN}} & Mannose \\
\textmd{MCC} & Matthews correlation coefficient \\
\textmd{MCD} & Minimum centroid distance \\
\textmd{MDS} & Multi-dimensional scaling \\
\textmd{MES} & Missense enrichment score \\
\textmd{MHRA} & Medicines and Healthcare products Regulatory Agency \\
\textmd{ML} & Machine learning \\
\textmd{MLP} & Multi-layer perceptron \\
\textmd{MOAD} & Mother Of All Databases \\
\textmd{mRNA} & Messenger RNA \\
\textmd{MRO} & Maximum residue overlap \\
\textmd{MTD} & Maximum tolerated dose \\[0.3175cm]
\textbf{\large N} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/NAD}{NAD}} & Nicotinamide-adenine-dinucleotide \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/NAP}{NAP}} & Nicotinamide-adenine-dinucleotide phosphate \\
\textmd{NI} & No ions \\
\textmd{NME} & New molecular entity \\
\textmd{\textit{N}-mer} & Protein peptide of \textit{N} amino acids \\
\textmd{NMR} & Nuclear magnetic resonance \\
\textmd{NOAEL} & No observed adverse effect level \\
\textmd{NR} & Non-redundant \\[0.3175cm]
\textbf{\large O} & \\[0.25cm]
\textmd{OR} & Odds ratio \\[0.3175cm]
\textbf{\large P} & \\[0.25cm]
\textmd{PAM} & Point accepted mutation \\
\textmd{PanDDA} & Pan-dataset density analysis \\
\textmd{PCA} & Principal component analysis \\
\textmd{PD} & Pharmacodynamics \\ 
\textmd{PDB} & Protein Data Bank \\
\textmd{PDBe} & Protein Data Bank Europe \\
\textmd{PDBe-KB} & Protein Data Bank Europe Knowledge Base \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/PID}{PID}} & Peridinin \\
\textmd{PK} & Pharmacokinetics \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/PLP}{PLP}} & Vitamin B6 phosphate \\
\textmd{PR} & Precision-recall curve \\[0.3175cm]
\textbf{\large Q} & \\[0.25cm]
\textmd{QSAR} & Quantitative structure-activity relationship \\[0.3175cm]
\textbf{\large R} & \\[0.25cm]
\textmd{\textit{R\textsubscript{g}}} & Radius of gyration \\
\textmd{ROC} & Receiver operating characteristic \\
\textmd{RMSD} & Root mean square deviation \\
\textmd{RNA} & Ribonucleic acid \\
\textmd{RRO} & Relative residue overlap \\
\textmd{RVO} & Relative volume overlap \\
\textmd{RSA} & Relative solvent accessibility \\[0.3175cm]
\textbf{\large S} & \\[0.25cm]
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/SAH}{SAH}} & S-adenosyl-L-homocysteine \\
\textmd{SAS} & Solvent accessible surface \\
\textmd{SASA} & Solvent accessible surface area \\
\textmd{SBVS} & Structure-based virtual screening \\
\textmd{SD} & Standard deviation \\
\textmd{SDP} & Specificity-determining position \\
\textmd{SES} & Solvent excluded surface \\
\textmd{SS} & Sum of squares \\[0.3175cm]
\textbf{\large T} & \\[0.25cm]
\textmd{TN} & True negative \\
\textmd{TP} & True positive \\
\textmd{TPR} & True positive rate \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/TRS}{TRS}} & Tris buffer \\[0.3175cm]
\textbf{\large U} & \\[0.25cm]
\textmd{\textit{U\textsubscript{D}}} & Distance U \\
\textmd{\textit{U\textsubscript{max}}} & Maximum U statistic \\
\textmd{UMD} & Unconserved and missense depleted \\
\textmd{UME} & Unconserved and missense enriched \\
\textmd{UPGMA} & Unweighted pair-group method with arithmetic mean \\
\textmd{\textit{U\textsubscript{rel}}} & Relative U statistic \\[0.3175cm]
\textbf{\large V} & \\[0.25cm]
\textmd{VDW} & Van der Waals \\
\textmd{VN} & Virtual node \\
\textmd{\textit{V\textsubscript{R}}} & Volume ratio \\
\textmd{VS} & Virtual screening \\
\textmd{VUS} & Variant of uncertain significance \\[0.3175cm]
\textbf{\large X} & \\[0.25cm]
\textmd{XRC} & X-ray crystallography \\
\textmd{\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/XYP}{XYP}} & Xylose \\[0.3175cm]

\end{longtable}
