\chapter{Classification of likely functional class for ligand binding sites identified from fragment screening}

\section*{Preface}

On this chapter, a series of methods to group small molecule ligands by protein interactions and cluster ligand sites by relative solvent accessibility profile are introduced. 293 unique ligand binding sites are defined from 37 fragment screening experiments and grouped into four clusters which are differentially enriched in known functional sites. A multi-layer perceptron is developed to predict cluster labels with an accuracy of 96\% so allowing functional classification of sites for proteins not in this set. 

\section*{Publications}

Utg√©s, J.S., MacGowan, S.A., Ives, C.M., Barton, G.J. Classification of likely functional class for ligand binding sites identified from fragment screening. \textit{Commun Biol} \textbf{7}, 320 (2024). \url{https://doi.org/10.1038/s42003-024-05970-8}.

\section*{Author contributions}

J.S.U., G.J.B., S.A.M. and C.M.I. conceived, designed, and developed the research. J.S.U. and C.M.I. analysed the data. J.S.U., C.M.I. and S.A.M. developed the software. J.S.U. and G.J.B. wrote, reviewed and edited the manuscript. G.J.B. secured funding and supervised.

\section{Introduction}

Fragment-based drug discovery, or fragment screening, is a widely used technique to identify lead compounds against a specific protein target \cite{MURRAY_2009_FBDD}. Fragment screening typically uses X-ray crystallography to provide detailed information on the binding mode of small molecule fragments that bind to a target protein. Fragments can then be linked or grown to form more potent leads \cite{CONGREVE_2003_RO3, REES_2004_FBLD, SCHIEBEL_2016_FRAGMENTS}. A typical fragment screening experiment will generate a collection of three-dimensional structures with fragments bound to different regions of the protein. While many fragments group around well understood catalytic or binding sites, and so provide a scaffold for drug discovery, fragments are also observed bound to regions of the protein where the functional significance is unclear. Such sites may be functionally irrelevant or could identify previously unknown allosteric or other functionally important sites worthy of experimental investigation. 

On this chapter, a strategy is presented to identify which fragment binding sites are most likely to be of functional importance and so prioritise sites for further investigation. The first step is to identify binding sites from the fragment data. Ligand binding sites are not being predicted, as is done by P2Rank \cite{KRIVAK_2018_P2RANK}, fpocket \cite{GUILLOUX_2009_FPOCKET}, or molecular dynamics-based methods such as MixMD \cite{LEXA_2011_FLEXIBILITY, GHANAKOTA_2018_MIXMD}, MDmix \cite{ALVAREZ_2014_MIXMD}, or SILCS \cite{FALLER_2015_SILCS}. Instead, from a set of experimentally determined three-dimensional structures of protein-ligand complexes, ligands are defined as binding to the same site based on their interactions with the protein.

In most previous studies the focus has been on clustering ligands by root mean square deviation (RMSD) \cite{SHIN_2005_PDBLIGAND} or Euclidean distances \cite{KOZAKOV_2005_CLUSTERING} after ligand superposition. Ligand site prediction methods, such as 3DLigandSite \cite{WASS_2010_3DLIGANDSITE, MCGREIG_2022_3DLIGANDSITE} also define sites based on ligand structure superposition and RMSD. Here, an algorithm is described that defines ligand binding sites from analysis of ligand interaction residues on the protein. The method allows the extent of a fragment binding site to be described without the need for superposition. Unsupervised methods are then applied to group the defined sites into four robust clusters based on their relative solvent accessibility profiles. This analysis suggests which sites in a set of 39 fragment screening experiments are most likely to be of functional significance through further stratification by evolutionary conservation and human population missense depletion \cite{MACGOWAN_2017_VARIANTS, MACGOWAN_2024_VARIANTS}. A machine learning method is subsequently developed to classify a set of interacting residues from an experimentally determined structure or a predicted ligand binding site into one of the four defined classes.

The work described in this chapter is likely to be of interest to groups focusing on fragment screening studies, but wider applications to ligand site classification from experimentally determined or predicted structures are also discussed. 

\section{Methods}
\label{sec:ch_FRAGSYS_methods}

\subsection{Structure dataset}

The Pan-Dataset Density Analysis (PanDDA) algorithm characterises a set of related crystallographic data sets of the same crystal form and identifies binding events by isomorphous difference maps \cite{PEARCE_2017_PANDDA}. Initially, 3021 three-dimensional structures determined by X-ray crystallography were selected by querying the PDBe \cite{wwPDB_2019_PDB} for entries containing the string ``PanDDA'' in their title. 1542 of the structures included bound ligands for 39 different proteins. Four proteins which were in multi-protein complexes including additional ligands were excluded to leave single-protein-ligand complexes from 35 different proteins and a total of 1450 structures. The structures presented resolutions from 0.9-3.3 \AA{}, with a mean resolution of $\approx$1.5 \AA{}. Preferred biological assemblies, as defined by PISA \cite{KRISSINEL_2007_PISA}, were downloaded from the PDBe via ProIntVar \cite{MACGOWAN_2020_DRSASP}. 

\subsection{Binding site definition}

Ligand binding site definition or prediction approaches are usually based on the spatial superposition of ligand molecules and their clustering on metrics such as Euclidean distance or RMSD \cite{SHIN_2005_PDBLIGAND, KOZAKOV_2005_CLUSTERING, WASS_2010_3DLIGANDSITE, MCGREIG_2022_3DLIGANDSITE}. These methods rely on structural superposition ans can be computationally expensive when dealing with large numbers of structures. Here, binding sites are defined from protein-ligand interactions without the need for superposition (\autoref{fig:fragsys_bs_def}).

Only non-ion ligands of interest were used for the binding site definition. These do not include water molecules (\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/HOH}{HOH}), nor other by-products of the experimental conditions, such as ethylene glycol (\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/EDO}{EDO}), glycerol (\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/GOL}{GOL}) or tris (\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/TRS}{TRS}). Ligand contacts were determined with Arpeggio \cite{JUBB_2017_ARPEGGIO}. For a given ligand, a binding fingerprint is defined as the UniProt sequence numbers of the residues the ligand interacts with. For a pair of ligands $L_{A}$ and $L_{B}$, with interaction fingerprints $A$ and $B$, their relative intersection, $I_{rel}$, is defined (\autoref{eq:relative_intersection}) by dividing the intersection of sets $A$ and $B$ by the maximum possible intersection between the two sets, given by the minimum fingerprint length (\autoref{eq:max_intersection}).

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/FIG8_bs_def.png}
    \caption[Ligand binding site definition algorithm]{\textbf{Ligand binding site definition algorithm.} The method defines ligand binding sites from a set of three-dimensional structures portraying the complex of a protein of interest bound to ligands. \textbf{(A)} Protein-ligand complex exemplified by Tyrosine-protein phosphatase non-receptor type 1, PTP-1B, (\href{https://www.uniprot.org/uniprotkb/P18031/entry}{P18031}) bound to N-(4-methyl-1,3-thiazol-2-yl)propanamide (\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/JFP}{JFP}). PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qdj}{5QDJ} \cite{KEEDY_2018_PTP1B}; \textbf{(B)} Ligand binding fingerprint, comprised by the UniProt sequence numbers of ligand-interacting residues; \textbf{(C)} Formula of the employed similarity metric: relative intersection, $I_{rel}$; \textbf{(D)} Hierarchical clustering tree resulting from the similarity matrix, cut at a threshold to determine distinct clusters of ligands. This is a subtree of the full tree, showing 7/18 binding sites defined on PTP-1B. This is represented by a dashed line pointing downwards on the tree; \textbf{(E)} Superoposition of ligands binding to the protein, coloured by ligand cluster. Only ligands found in groups (sites) 1-7 are coloured by their membership. The rest are coloured in grey. Cartoon PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qdj}{5QDJ}.}
    \label{fig:fragsys_bs_def}
\end{figure}

\begin{equation}
    I_{rel} = \frac{A \cap B}{A \cap B_{max}}
\label{eq:relative_intersection} % Label for the equation
\end{equation}
\myequations{Relative Intersection}

\vspace{-35pt} % Adjust this value as needed

\begin{equation}
    A \cap B_{max} = \text{min}(\text{len}(A), \text{len}(B))
\label{eq:max_intersection}
\end{equation}
\myequations{Maximum Intersection}
\vspace{-21pt} % Adjust this value as needed

$I_{rel}$ is thus a similarity metric, ranging 0-1, that can be used to perform hierarchical clustering on the ligands. Single-linkage hierarchical clustering was performed with the OC software \cite{BARTON_1993_OC}. After exploring various $I_{rel}$ threshold values for cutting the tree, $I_{rel}$ = 0.66 was selected as the optimal threshdold for this dataset. As $I_{rel}$ is a similarity metric, this threshold indicates that a ligand shares at least two-thirds of its binding residues with at least one other ligand within the same cluster. A total of 293 ligand binding sites across 37 protein domains were defined this way. For each protein, all structures were multiply aligned by STAMP \cite{RUSSELL_1992_STAMP}. Ligand binding sites were visualised in UCSF Chimera \cite{PETTERSEN_2004_CHIMERA}.

\subsection{Multiple sequence alignments}

Two of the 35 proteins included fragment screening experiments targeting multiple domains, or protein products, resulting in 39 protein-fragments sets. A representative sequence was selected for each of the 39 sets of structures. These sequences were used to search SwissProt \cite{BOUTET_2016_UNIPROT} for homologues and generate multiple sequence alignments. The search was performed with jackHMMER \cite{EDDY_1995_HMMER} using default parameters and 5 iterations. Evolutionary divergence within the alignments was quantified with the Shenkin divergence score, $V_{Shenkin}$, \cite{SHENKIN_1991_SCORE} and the normalised $N_{Shenkin}$, as defined by Utg√©s \textit{et al.} \cite{UTGES_2021_ANKS}.

\subsection{Human variants and enrichment}

VarAlign \cite{MACGOWAN_2020_DRSASP} was used to retrieve genetic variants from gnomAD v2.1 \cite{KARCZEWSKI_2020_GNOMAD} found in the human sequences within the multiple sequence alignment generated for each target protein. gnomAD contains exomes and genomes from 141,456 unrelated individuals with no known phenotypic conditions and is therefore a reasonable representation of the general \textit{healthy} population. Variants found in the human sequences within the alignments were mapped to individual alignment columns and missense enrichment scores (MES) were calculated. MES represents the enrichment in missense variants of an alignment column relative to the average of the other columns in the alignment \cite{MACGOWAN_2017_VARIANTS, MACGOWAN_2024_VARIANTS}. 95\% confidence intervals (CI) and $p$-values were used to assess the significance of these ratios \cite{SZUMILAS_2010_ODDSRATIOS}. MES was also calculated for the defined ligand binding sites. The MES of a binding site represents the enrichment in missense variants of a binding site relative to the rest of protein residues. Alignment columns as well as binding sites were classified as enriched (MES $>$ 0), depleted (MES $<$ 0) or neutral (MES = 0). Enrichment was not calculated for two of the 39 proteins since no human homologues were identified.

\subsection{Binding site clustering}

Secondary structures were defined with DSSP \cite{KABSCH_1983_DSSP} via ProIntVar, and relative solvent accessibility (RSA) was calculated with the method of  Tien \textit{et al.} \cite{TIEN_2013_RSA}. The defined binding sites were grouped according to the pattern of RSA as follows and summarised in \autoref{fig:fragsys_bs_clust}.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/FIG9_bs_clust.png}
    \caption[Binding site clustering algorithm]{\textbf{Binding site clustering algorithm.} The method here clusters ligand binding sites defined across different proteins based on their solvent accessibility profiles. \textbf{(A)} Defined ligand binding site for PTP-1B on PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qdu}{5QDU} \cite{KEEDY_2018_PTP1B}; \textbf{(B)} Relative solvent accessibility profile of a binding site, represented by the RSA of the site residues; \textbf{(C)} Formula of the employed distance metric: distance $U$, $U_{D}$; \textbf{(D)} Multidimensional scaling (MDS) representation of binding sites coloured according to the four clusters determined by the $K$-means algorithm. Dashed lines represent the cluster limits.}
    \label{fig:fragsys_bs_clust}
\end{figure}

\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.7\textwidth]{figures/ch_FRAGSYS/PNG/SUPP_FIG1_ward_clust.png}
    \caption[Binding sites Ward clustering]{\textbf{Binding sites Ward clustering.} Cluster map of the $U$ distance, $U_{D}$, matrix of the 293 defined binding sites clustered by the Ward hierarchical clustering method implemented in SciPy. The tree is cut at $D_{Ward}$ = 1.8, giving four clear clusters. These clusters are labelled so they correspond to the ones obtained with $K$-means. Clusters in the heat map are represented by dark squares around the diagonal. $U_{D}$ is a distance; therefore, clusters include sites that are similar to each other, and present lower distances (dark colour).}
    \label{fig:fragsys_bs_clust_ward}
\end{figure}

Given two binding sites, $A$ and $B$, with RSA profiles $r_{A}$ and $r_{B}$ and sizes $n_{A}$ and $n_{B}$ respectively, in amino acid residues, $U_{A}$ and $U_{B}$ can be calculated (\autoref{eq:mann_whitney_U}). The Mann-Whitney $U$ statistic \cite{MANN_WHITNEY_1947_TEST}, as implemented in SciPy \cite{VIRTANEN_2020_SCIPY}, was chosen as it has a maximum theoretical value ($U_{max}$) (\autoref{eq:maximum_U}). A relative $U$ value, $U_{rel}$, ranging 0-1 is obtained by dividing the $U$ value by $U_{max}$. The more similar $r_{A}$ and $r_{B}$ are, the bigger $U$ and $U_{rel}$ are. Thus, $U_{rel}$ is a similarity score. Subtracting $U_{rel}$ from 1 gives the $U$ distance, $U_{D}$, (\autoref{eq:relative_U}). $U_{D}$ is indicative of how different $r_{A}$ and $r_{B}$ are and can be used to cluster binding sites by their RSA profiles.

\begin{equation}
U_A = R_A - \frac{n_A(n_A + 1)}{2}, \quad U_B = R_B - \frac{n_B(n_B + 1)}{2} \quad
\label{eq:mann_whitney_U}
\end{equation}
\myequations{Mann-Whitney's U}

\vspace{-31pt} % Adjust this value as needed

\begin{equation}
U_A + U_B = n_A n_B, \quad U = \min(U_A, U_B) \rightarrow U_{max} = \frac{n_A n_B}{2} \quad
\label{eq:maximum_U} 
\end{equation}
\myequations{Maximum U}

\vspace{-31pt} % Adjust this value as needed

\begin{equation}
U_{rel} = \frac{U}{U_{max}} \rightarrow U_D = 1 - U_{rel} \quad 
\label{eq:relative_U}
\end{equation}
\myequations{Relative U}

\vspace{-12pt} % Adjust this value as needed

After calculating pairwise distances between the RSA profiles of the defined binding sites, $K$-means clustering \cite{LLOYD_1982_KMEANS} was performed. Several clustering algorithms were tried to realise this task, including some hierarchical, or connectivity-based, such as single and complete-linkage \cite{SORENSEN_1948_CL}, unweighted average linkage clustering (UPGMA) \cite{SOKAL_1958_UPGMA}, or Ward linkage \cite{WARD_1963_CLUSTERING}, as well as centroid-based, such as $K$-means. Overall, the clusters obtained by the different methods were similar. Ward linkage and $K$-means resulted in the most similar clusters, displaying an average similarity between clusters of 85\% (\autoref{fig:fragsys_bs_clust_ward}).

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/SUPP_FIG2_KNN_robustness.png}
    \caption[$K$-means clustering robustness]{\textbf{K-means clustering robustness.} Cluster analysis to assess the quality of the $K$-means clustering. For each $K \in$ [2, 14], clustering is bootstrapped 1000 times with different initial random states. Error bars indicate 1 SD. \textbf{(A)} Calinski-Harabasz Index (CHI); \textbf{(B)} Davies-Bouldin Index (DBI); \textbf{(C)} Inertia; \textbf{(D)} Silhouette. All methods agree the optimal clustering of this dataset lies in $K \in$ [4, 6].}
    \label{fig:kmeans_robustness}
\end{figure}

Finally, multidimensional scaling (MDS) \cite{MEAD_1992_MDS} with $N$ = 2 dimensions was performed to visualise the clusters. $K$-means was selected as it presented better contained clusters, i.e., less overlapping between members of different clusters. The silhouette \cite{ROUSSEEUW_1987_SILHOUETTES}, and elbow \cite{THORNDIKE_1953_ELBOW} methods, as well as Calinski-Harabasz index (CHI) \cite{CALINSKI_1974_CHI} and Davies-Bouldin index (DBI) \cite{DAVIES_1979_DBI} were used for finding optimal $K$ (\autoref{fig:kmeans_robustness}), in conjunction with the MDS, trees resulting from hierarchical clustering algorithms, and the visual representation of the RSA profiles, to decide on a final number of $K$ = 4 clusters: C1, C2, C3, and C4. Clustering was repeated 1000 times with different random states and 289/293 (98.6\%) sites were always present in the same cluster, thus suggesting the clusters are robust.

\subsection{Binding site cluster prediction}

Two different predictive models were developed with the aim of classifying binding sites into the defined RSA-based clusters obtained with $K$-means, as described above. The first  uses the $K$-nearest neighbour (KNN) algorithm as implemented in Scikit-learn \cite{PEDREGOSA_2011_SKLEARN}, with $K$ = 3. The input for this KNN model is the rows of the $U_{D}$ matrix, containing the distances between pairs of binding site RSA profiles.

The second model is a multi-layer perceptron (MLP) \cite{CYBENKO_1989_MLP}, a type of artificial neural network (ANN) constructed with Keras \cite{CHOLLET_2015_KERAS} with a single hidden fully connected layer between the input layer of 11 neurons, and the output layer of 4 neurons, one for each cluster label. RSA profiles present different lengths depending on the size (number of amino acids) of the binding site. As this input is not suitable for the neural network, binding sites were encoded as an 11-element vector. The first element of the vector encodes the size of the binding site relative to the maximum site size of 40 residues within this dataset. The other 10 elements represent the proportion of residues forming the binding site with an RSA \% within a 10-unit interval: [0, 10), [10, 20), ‚Ä¶, and [90, 100].

\subsubsection{MLP ablation}

A thorough hyperparameter optimisation was carried out by examining the effect that a series of hyperparameter changes have on the prediction accuracy relative to our current ML setup, labelled as \textit{current}. Sixty-four single-hyperparameter changes were performed, one at a time. For each variation, 100 models were trained with different seeds and the average validation accuracies compared to our current MLP. Sixty-four pairwise $t$-tests were conducted to compare the accuracy means, and Benjamini-Hochberg correction \cite{BENJAMINI_1995_FDR} applied. FDR and increment in accuracy, $\Delta_{acc}$ (\autoref{eq:delta_accuracy}) are used to describe the results, where $acc_{current}$ is the average validation accuracy of our \textit{current} ML setup across the 100 seeds, and $acc_{variant}$ is the average accuracy across 100 seeds of each one of the 64 variant models. $\Delta_{acc} <$ 0 represents a decrease in performance respect our current ML architecture, whereas $\Delta_{acc} >$ 0 corresponds to a higher accuracy. The results of this ablation study are described below and graphically represented in \autoref{fig:mlp_ablation} and \autoref{tab:MLP_ablation}.

\begin{equation}
\Delta_{acc} = acc_{variant} - acc_{current}
\label{eq:delta_accuracy}
\end{equation}
\myequations{Increment in MLP accuracy}
\vspace{-37pt} % Adjust this value as needed

\paragraph{Number of layers}

Removing the single hidden layer resulted in a significant decrease in accuracy, $\Delta_{acc}$ = \textminus11\% (FDR $<$ 0.05). The addition of more layers did not improve accuracy: 2-layer  $\Delta_{acc}$ = \textminus1\% (FDR $<$ 0.05), 10-layer $\Delta_{acc}$ = \textminus8.9\% (FDR $<$ 0.05), or was not statistically different from our current setup baseline: 5-layer $\Delta_{acc}$ = \textminus0.15\% (FDR = 0.42).

\paragraph{Neurons per layer}

The addition of neurons N\textsubscript{neurons} = [11, 20, 25, 50, 100] in the single layer did not improve the current accuracy (FDR $>$ 0.05). The removal of neurons did not have an effect of performance for N\textsubscript{neurons} = [4, 5, 6, 7, 8, 9] (FDR $>$ 0.05), or a significant negative effect for 1 neuron, $\Delta_{acc}$ = \textminus15\% (FDR $<$ 0.05), 2 neurons $\Delta_{acc}$ = \textminus4\% (FDR $<$ 0.05), and 3 neurons, $\Delta_{acc}$ = \textminus1\% (FDR $<$ 0.05). This result suggests that 5 neurons on a single hidden layer might be enough to achieve a comparable accuracy to our current model.

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/SUPP_FIG3_MLP_ablation.png}
    \caption[MLP ablation study]{\textbf{MLP ablation study.} Ablation study performed on the MLP. Sixty-four single hyperparameter changes are conducted one at a time to explore the hyperparameter space and the effect they have on the prediction accuracy relative to our current machine learning setup, labelled as \textit{current}. Box and whiskers represent the distribution of validation accuracy across 100 random seeds. The box contains the central 50\% of the data, i.e., Q1 ‚Äì median (Q2) ‚Äì Q3  also known as interquartile range (IQR). Whiskers extend to 1.5 $\times$ IQR, and beyond them are the outliers. Dashed lines mark the separation between different hyperparameters: number of layers, neurons, activation, loss functions, weight initialisers, optimisers, learning, dropout rates, and regularisation techniques.}
    \label{fig:mlp_ablation}
\end{figure}
%\vspace{-17pt} % Adjust this value as needed

\paragraph{Activation function}

The usage of different activation functions either negatively affected the accuracy of the MLP ($\Delta_{acc} <$ 0) or had no effect (FDR $>$ 0.05).

\paragraph{Loss function}

Different loss functions resulted in terrible loss of accuracy $\Delta_{acc} \approx$ \textminus50\% (FDR $<$ 0.05). This is expected as they are not appropriate for a multi-label classifier, unlike sparse categorical cross entropy.

\paragraph{Weight initialiser}

Most weight initialisers were tested and either negatively affected the accuracy of the MLP ($\Delta_{acc} <$ 0) or had no effect (FDR $>$ 0.05). However, \texttt{RandomNormal}, \texttt{RandomUniform}, and \texttt{TruncatedNormal} did improve the accuracy but by less than 1\%, $\Delta_{acc} <$ +1\% (FDR $<$ 0.05).

\paragraph{Optimiser}

Regarding optimisers, they either severely negatively affected accuracy $\Delta_{acc} \approx$ \textminus30\% (FDR $<$ 0.05), had no significant effect (FDR $>$ 0.05), or very slightly improved accuracy, such as \texttt{RMSProp} $\Delta_{acc} <$ +1\% (FDR $<$ 0.05).

\paragraph{Learning rate}

Extreme learning rates of 0.001 (too small), and 1.0 (too big) negatively affected prediction $\Delta_{acc} <$ \textminus5\% (FDR $<$ 0.05). Intermediate rates had either no significant effect (FDR $>$ 0.05) or a small effect $\|\Delta_{acc}\| <$ 1\%.

\paragraph{Dropout rate}

Regarding dropout rates, a rate = 75\%, negatively affected prediction  $\Delta_{acc} <$ \textminus2\%, (FDR $<$ 0.05). Lower dropout rates: 0.1, 0.25, and 0.33 did improve the accuracy, but the effect size is very small, $\Delta_{acc} <$ +1\% (FDR $<$ 0.05). This result agrees with the effect of the removal of neurons per layer and shows that fewer neurons on a single hidden layer might be enough to achieve a comparable accuracy to our current model, as dropping them out has no effect.

\paragraph{Regularisation}

Overall, implementing kernel, bias, or activity regularisation techniques did not improve prediction accuracy, but worsened it $\Delta_{acc} \ni$ [\textminus2.56, \textminus0.46] (FDR $<$ 0.05).

\vspace{+19pt} % Adjust this value as needed

\begin{longtable}{|c|c|c|c|}
\hline
\textbf{Model}              & \textbf{Accuracy} (\%) & \textbf{$\Delta_{acc}$} (\%) & \textbf{FDR} \\ \hline
\endfirsthead
%
\multicolumn{4}{c}%
{{\bfseries Table \thetable} (continued)} \\
\hline
\textbf{Model}              & \textbf{Accuracy} (\%) & \textbf{$\Delta_{acc}$} (\%) & \textbf{FDR} \\ \hline
\endhead{}
%
\textit{current}            & 93.9              & -                       & -            \\ \hline
\texttt{N\textsubscript{LAYERS} = 0}                   & 82.9              &  \textminus11.0                     & 0            \\ \hline
\texttt{N\textsubscript{LAYERS} = 2}                    & 92.9              &  \textminus1.0                      & 0            \\ \hline
\texttt{N\textsubscript{LAYERS} = 5}                    & 93.8              &  \textminus0.1                   & 0.42         \\ \hline
\texttt{N\textsubscript{LAYERS} = 10}                   & 85.0              &  \textminus8.9                   & 0            \\ \hline
\texttt{N\textsubscript{NEURONS} = 1}                  & 79.0              &  \textminus14.9                  & 0            \\ \hline
\texttt{N\textsubscript{NEURONS} = 2}                  & 89.8               &  \textminus4.2                   & 0            \\ \hline
\texttt{N\textsubscript{NEURONS} = 3}                  & 92.9              &  \textminus1.1                   & 0            \\ \hline
\texttt{N\textsubscript{NEURONS} = 4}                  & 93.7              &  \textminus0.3                   & 0.24         \\ \hline
\texttt{N\textsubscript{NEURONS} = 5}                  & 93.8              &  \textminus0.1                   & 0.54         \\ \hline
\texttt{N\textsubscript{NEURONS} = 6}                  & 93.7              &  \textminus0.2                   & 0.26         \\ \hline
\texttt{N\textsubscript{NEURONS} = 7}                  & 93.6              &  \textminus0.3                   & 0.08         \\ \hline
\texttt{N\textsubscript{NEURONS} = 8}                  & 93.6              &  \textminus0.3                   & 0.08         \\ \hline
\texttt{N\textsubscript{NEURONS} = 9}                  & 93.8              &  \textminus0.1                   & 0.49         \\ \hline
\texttt{N\textsubscript{NEURONS} = 11}                 & 94.0              & +0.1                    & 0.65         \\ \hline
\texttt{N\textsubscript{NEURONS} = 20}                 & 94.0              & +0.1                    & 0.68         \\ \hline
\texttt{N\textsubscript{NEURONS} = 25}                 & 94.0              &  \textminus0.0                   & 0.83         \\ \hline
\texttt{N\textsubscript{NEURONS} = 50}                 & 93.9              &  \textminus0.0                   & 0.91         \\ \hline
\texttt{N\textsubscript{NEURONS} = 100}                & 93.8              &  \textminus0.1                   & 0.49         \\ \hline
\texttt{sigmoid}                     & 92.4              &  \textminus1.5                   & 0            \\ \hline
\texttt{elu}                         & 94.1              & +0.2                    & 0.38         \\ \hline
\texttt{selu}                        & 93.7              &  \textminus0.2                   & 0.15         \\ \hline
\texttt{exponential}                         & 92.8              &  \textminus1.1                   & 0            \\ \hline
\texttt{tanh}                        & 93.8              &  \textminus0.1                   & 0.41         \\ \hline
\texttt{softplus}                    & 93.3              &  \textminus0.6                   & 0            \\ \hline
\texttt{softmax}                     & 90.0               &  \textminus3.9                   & 0            \\ \hline
\texttt{softsign}                    & 93.5              &  \textminus0.4                   & 0.02         \\ \hline
\texttt{MSE}             & 44.1              &  \textminus49.8                  & 0            \\ \hline
\texttt{Poisson}         & 43.6              &  \textminus50.3                  & 0            \\ \hline
\texttt{KLDivergence}             & 43.9              &  \textminus50.0                     & 0            \\ \hline
\texttt{HeNormal}       & 93.7              &  \textminus0.2                    & 0.37         \\ \hline
\texttt{HeUniform}        & 93.7              &  \textminus0.2                   & 0.21         \\ \hline
\texttt{RandomNormal}     & 94.4              & +0.5                    & 0.02         \\ \hline
\texttt{RandomUniform}      & 94.4              & +0.5                    & 0.01         \\ \hline
\texttt{TruncatedNormal}    & 94.5              & +0.6                    & 0            \\ \hline
\texttt{Ones}           & 90.0               &  \textminus3.9                   & 0            \\ \hline
\texttt{Zeros}          & 42.6              &  \textminus51.3                  & 0            \\ \hline
\texttt{GlorotNormal} & 93.9              &  \textminus0.0                   & 0.79         \\ \hline
\texttt{SGD}             & 46.7              &  \textminus47.2                  & 0            \\ \hline
\texttt{RMSProp}          & 94.5              & +0.6                    & 0            \\ \hline
\texttt{Adadelta}         & 29.4              &  \textminus64.5                   & 0            \\ \hline
\texttt{Adagrad}         & 55.1              &  \textminus38.8                  & 0            \\ \hline
\texttt{Adamax}           & 94.1              & +0.2                    & 0.6          \\ \hline
\texttt{Nadam}            & 94.1              & +0.2                    & 0.48         \\ \hline
\texttt{Ftrl}             & 34.7              &  \textminus59.2                  & 0            \\ \hline
\texttt{Learning rate = 0.001}       & 89.0              &  \textminus4.9                   & 0            \\ \hline
\texttt{Learning rate = 0.005}       & 93.5              &  \textminus0.4                   & 0.04         \\ \hline
\texttt{Learning rate = 0.05}        & 93.6              &  \textminus0.3                   & 0.06         \\ \hline
\texttt{Learning rate = 0.1}         & 94.3              & +0.4                    & 0.05         \\ \hline
\texttt{Learning rate = 0.25}        & 93.1              & \textminus0.8                   & 0.27         \\ \hline
\texttt{Learning rate = 0.5}         & 88.2              &  \textminus5.7                    & 0            \\ \hline
\texttt{Learning rate = 1.0}         & 69.6               &  \textminus24.3                  & 0            \\ \hline
\texttt{Dropout rate = 0.1}                & 94.5              & +0.6                    & 0            \\ \hline
\texttt{Dropout rate = 0.25}               & 94.6              & +0.7                    & 0            \\ \hline
\texttt{Dropout rate = 0.33}               & 94.7              & +0.8                     & 0            \\ \hline
\texttt{Dropout rate = 0.5}                & 94.1              & +0.2                    & 0.54         \\ \hline
\texttt{Dropout rate = 0.75}               & 91.9              &  \textminus2.0                   & 0            \\ \hline
\texttt{L1}                 & 92.5              &  \textminus1.4                   & 0            \\ \hline
\texttt{L2}                 & 93.5              &  \textminus0.4                   & 0.04         \\ \hline
\texttt{L1L2}               & 92.0              &  \textminus1.9                   & 0            \\ \hline
\texttt{Orthogonal rows}        & 93.5              &  \textminus0.4                   & 0.02         \\ \hline
\texttt{Orthogonal columns}        & 93.9              &  \textminus0.0                   & 0.62         \\ \hline
\texttt{\textit{all} L1}            & 91.8              &  \textminus2.1                   & 0            \\ \hline
\texttt{\textit{all} L2}            & 92.2              &  \textminus1.7                    & 0            \\ \hline
\texttt{\textit{all} L1L2}          & 91.4              &  \textminus2.5                   & 0            \\ \hline
\end{longtable}

% Prevent LaTeX from incrementing the table counter
\addtocounter{table}{-1}
\addcontentsline{lot}{table}{\protect\numberline{2.1}MLP ablation study}


% Separate caption for the table
\begin{center}
\captionsetup{type=table} % Specify the type of the caption
\caption[MLP ablation study]{\textbf{MLP ablation study.} Sixty-four single hyperparameter changes were conducted one at a time to explore the hyperparameter space and the effect they have on the prediction accuracy relative to our current ML setup, labelled as \textit{current}. Accuracy (\%) represents the validation accuracy average across 100 random seeds. $\Delta_{acc}$ (\%) represents the difference in accuracy between the variant MLP model and our \textit{current} setup. Negative values result from a decrease in performance, whereas positive ones mean an improvement in classification accuracy. FDR was employed to assess the significance of these differences.}
\label{tab:MLP_ablation}
\end{center}

\vspace{-23pt} % Adjust this value as needed
\vspace{-13pt} % Adjust this value as needed

\subsubsection{Performance evaluation}

The complete dataset ($N$ = 293) was split into a blind test set (1/11 = 27), and a training set (10/11 = 266). Ten repeats of a stratified 10-fold cross-validation were performed to assess the robustness of the ANN and compare it with the KNN model, as well as a baseline of the same models trained on randomly shuffled data and completely random label assignment ($p$ = 0.25). The reliability of the ANN predictions was assessed by means of a confidence score calculated as in \cite{CUFF_2000_PROFILES}, which represents how certain the MLP is of each individual prediction (\autoref{eq:confidence_score}). The score is based on the difference between the top first and second probabilities assigned by the network to each of the classes, $p_{1}$, and $p_{2}$, respectively. For example, if the output of the network were $P$ = [0.95, 0.02, 0.03, 0.0]. The probabilities would be sorted, so $p_{1}$ = 0.95, $p_{2}$ = 0.03, and a confidence score of 9 would be obtained.

\vspace{-12pt} % Adjust this value as needed

\begin{equation}
\text{confidence score} = \left\lfloor 10 \times (p_1 - p_2) \right\rfloor
\label{eq:confidence_score}
\end{equation}
\myequations{Confidence Score}

\vspace{-20pt} % Adjust this value as needed

The KNN is based on distances to all training data and so, as expected, consistently gives higher classification accuracy than the ANN model where sites are represented by their binned RSA profile, and are thus completely unaware of other sites, and their distances to them (\autorefpanel{fig:MLP_CV_blind_test}{A}). Both methods are significantly better than random. The average cross-validation accuracy across all repeats is of 98\%, 90\%, 33\%, 31\%, and 24\% for KNN, ANN models, their randomly trained versions, and completely random label assignment, respectively. The baseline accuracy of the randomly trained models is higher than 25\% since the dataset is unbalanced, with classes, C1 and C2 overrepresented.

\autorefpanel{fig:MLP_CV_blind_test}{B} shows the confidence of the ANN predictions across the 10 repeats of the 10-fold cross-validation. The overall accuracy is 90\%. Those predictions presenting a confidence score greater or equal to 5 present an accuracy of 97\% and cover 75\% of all predictions. Finally, \autorefpanel{fig:MLP_CV_blind_test}{C} shows the same two-dimensional representation of the K-means clusters found on \autorefpanel{fig:fragsys_bs_clust}{D} and demonstrates that those binding sites with lower prediction confidence are mostly located at the borders between clusters. Sites that switch cluster labels depending on the seed are also located in these regions.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/FIG10_ml_results.png}
    \caption[MLP cross-validation and blind test results]{\textbf{MLP cross-validation and blind test results.} \textbf{(A)} Average accuracy of the 10-repeat 10-fold ($N$ = 100) cross-validation of the KNN, and ANN predictive models compared to a baseline of the same models trained on randomly shuffled data, as well as complete random prediction ($p$ = 0.25). Boxes represent the central 50\% of the data. Whiskers extend to 1.5 $\times$ IQR, and beyond them are the outliers; \textbf{(B)} Cross-validation accuracy and proportion of binding sites against cumulative confidence score from the trained ANN. Sites with a confidence score greater or equal to 5 present an average accuracy of 97\%, and represent 75\% of all predictions. Predictions are for the 2660 cross-validation data points, 10 different repeats of 10 distinct splits of 26-27 binding sites each. Accuracy error bars indicate 95\% CI of the proportion \cite{WILSON_197_PROP_CI}; \textbf{(C)} MDS representation of the 293 binding sites. Training data are coloured according to the average confidence of their prediction in the cross-validation. Test data are coloured according to whether they were correctly predicted or not. Dashed lines indicate the limits of $K$-means clusters.}
    \label{fig:MLP_CV_blind_test}
\end{figure}

Once the model hyperparameters were optimised, 50 models were trained on 10/11 of the data ($N$ = 266) for the 10 different seeds used to initialise the models. From a final pool of 500 models, the one presenting the highest validation accuracy and lowest validation loss was chosen, with a validation accuracy of 96\%. This model, as well as KNN were used to predict on the blind test set. There was no significant difference in performance of the ANN and KNN models. Accuracies were 96\% (26/27), 95\% CI = [82, 99], and 100\% (27/27), 95\% CI = [88, 100], for ANN and KNN, respectively. The adjusted Rand index (ARI) \cite{RAND_1971_ARI, HUBERT_1985_ARI}, as well as adjusted mutual information (AMI) \cite{VINH_2009_AMI, VINH_2010_AMI} were calculated. ARI\textsubscript{ANN} = 0.93, 95\% CI = [0.81, 1.0] \cite{STEINLEY_2016_ARI}, AMI\textsubscript{ANN} = 0.93, 95\% CI = [0.82, 1.0]. ARI\textsubscript{KNN} = 1.0, AMI\textsubscript{KNN} = 1.0. 95\% CI of AMI was calculated by bootstrap resampling ($N$ = 10,000). The three metrics all agreed on the high performance of the MLP. \autorefpanel{fig:MLP_CV_blind_test}{C} illustrates how the binding site, which label was wrongly predicted by the ANN model is located on the limits between adjacent clusters C3 and C4. This result agrees with the $K$-means clustering reliability, and confidence score analysis of the cross-validation, where the same inter-cluster regions are highlighted due to their lower clustering reliability, and low confidence prediction. This suggests that the core of the clusters is stable, and that the ANN confidence score may be used to identify binding sites that are at the borders of clusters.

\subsection{Site function classification}

Ligand binding sites were divided into two groups - \textit{known function} and \textit{unknown function} - by searching UniProt \cite{UNIPROT_2019_UNIPROT} for feature annotations indicative of function, e.g., metal, substrate binding, active site, etc via the UniProt proteins API \cite{NIGHTINGALE_2017_API}. Seventeen out of the 35 proteins presented at least one UniProt annotated residue in one binding site. Manual curation using protein homology within the proteins in the data set added 9 more functionally annotated proteins. This gave a total of 44 sites from 26 proteins classified as of known function. All other sites were classified as \textit{unknown function}.

\subsection{Statistics and reproducibility}

All data analyses were carried out primarily with  the following Python libraries: NumPy \cite{HARRIS_2020_NUMPY}, Pandas \cite{MCKINNEY_2010_PANDAS,PANDAS_2022_PANDAS} and SciPy. Keras and Scikit-learn were used for machine learning, with Matplotlib \cite{HUNTER_2007_MATPLOTLIB}, and Seaborn \cite{WASKOM_2021_SEABORN} for plotting. All statistical tests performed are two-tailed, and significance level $\alpha$ = 0.05. Sample sizes and measures of significance are reported and described in the text, figures and legends.

\subsection{Data and code availability}

The software developed to carry out this analysis as well as the main summary result tables are available in the following repository: \url{https://github.com/bartongroup/FRAGSYS} (DOI: 10.5281/zenodo.10606595) \cite{UTGES_2024_FRAGSYS_ZENODO}.

\section{Results}

\subsection{Defined binding sites}

The focus here is on human proteins to allow the additional information from human population variation data to be explored. For this reason, two of the 39 protein domains, products of the SARS-CoV-2 Replicase polyprotein 1ab (\href{https://www.uniprot.org/uniprotkb/P0DTD1/entry}{P0DTD1}), were removed since they did not include any human homologues. The remaining 37 protein domains accounted for 1309 three-dimensional structures that included interactions with 1601 ligands of interest, of which 998 were unique. 293 ligand binding sites were defined across these domains, formed by 2664 unique ligand binding residues. The total number of binding sites per domain ranges from 1 to 24, with 33/37 domains presenting more than one defined binding site. The median number of sites per domain is seven.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/FIG1_bs_examples.png}
    \caption[Ligand clusters defined by the binding site definition algorithm]{\textbf{Ligand clusters defined by the binding site definition algorithm.} A single protein cartoon representation, coloured in white, is shown for simplicity for each example. Ligands are coloured according to the site they bind to. Identifiers are from UniProt. \textbf{(A)} There were 110 structures depicting human tyrosine-protein phosphatase non-receptor type 1 (\href{https://www.uniprot.org/uniprotkb/P18031/entry}{P18031}), binding 143 ligand molecules, 104 of which were unique. 18 binding sites were defined. Cartoon PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qdu}{5QDU} \cite{KEEDY_2018_PTP1B}; \textbf{(B)} The 68 ligands, 30 unique, found across 50 structures of the chestnut blight fungus endothiapepsin (\href{https://www.uniprot.org/uniprotkb/P11838/entry}{P11838}), were classified in 12 distinct binding sites. Cartoon PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5r1y}{5R1Y} \cite{WOLLENHAUPT_2020_F2X}; \textbf{(C)} For mouse mitogen-activated protein kinase 14 (\href{https://www.uniprot.org/uniprotkb/P47811/entry}{P47811}), 52 structures portrayed the interaction with 53 ligand molecules, 50 unique, which clustered in 10 ligand binding sites. Cartoon PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/1lew}{1LEW} \cite{CHANG_2002_MapKinase}.}
    \label{fig:bs_def_examples}
\end{figure}

\autoref{fig:bs_def_examples} illustrates three examples of the 37 domains for which ligand binding sites were defined by the algorithm presented on this chapter. The grouping of the ligands into the defined sites reflects the similarity between the interaction fingerprints of the different ligands with the target protein.

\autoref{fig:bss_features} shows the 293 defined binding sites are diverse in size (number of amino acids), solvent accessibility, evolutionary divergence, and missense depletion. Binding site size ranges from 2-40 residues with a median of 9, while median site RSA ranges from 4-80\%, with a median of 30\%. For evolutionary divergence, the average site $N_{Shenkin}$ spread from 0-80, peaking at 40. Lastly, MES spans \textminus0.75 to 1.0, peaking at neutrality (MES $\approx$ 0).

\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.65\textwidth]{figures/ch_FRAGSYS/PNG/FIG2_bs_feats_2x2.png}
    \caption[Variation in binding site features]{\textbf{Variation in binding site features.} Distribution of size \textbf{(A)}, median RSA \textbf{(B)}, average $N_{Shenkin}$ \textbf{(C)} and MES \textbf{(D)} across the 293 binding sites defined from our data set. Black dashed lines indicate the median of each distribution.}
    \label{fig:bss_features}
\end{figure}

Despite the diversity among sites, some general trends can be observed. \autorefpanel{fig:bss_feats_corr}A shows that larger binding sites tend to be less accessible to solvent ($r$ = \textminus0.4, p $\approx$ 0). \autorefpanel{fig:bss_feats_corr}{B} illustrates that larger sites are less divergent across homologues ($r$ = \textminus0.21, $p$ = $10^{-4}$) while \autorefpanel{fig:bss_feats_corr}{C} presents how larger sites show lower enrichment in neutral missense variants within the human population, i.e., are on average more depleted in missense variants than sites of a smaller size ($r$ = \textminus0.15, $p$ = 0.008). Correlations between MES and $N_{Shenkin}$, and RSA and $N_{Shenkin}$ were not significant, i.e., 95\% CI $r \ni$ 0.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/FIG3_bs_feats_corr.png}
    \caption[Relation between different binding site properties]{\textbf{Relation between different binding site properties.} A regression line is fitted to all data points previous to binning ($N$ = 293 binding sites). Pearson‚Äôs correlation coefficient $r$ \cite{RODGERS_1988_CORRELATION} and associated $p$-value and 95\% CI of $r$ \cite{BOWLEY_1928_R_CI} are also shown. Data points are grouped into bins according to different binding site size intervals, represented by box and swarm plots. \textbf{(A)} Median site RSA (\%) \textit{vs} binding site size, in amino acids; \textbf{(B)} Average $N_{Shenkin}$ \textit{vs} binding site size; \textbf{(C)} Average site MES \textit{vs} site size. Boxes represent the IQR, and whiskers extend to 1.5 $\times$ IQR.}
    \label{fig:bss_feats_corr}
\end{figure}

\vspace{-17pt} % Adjust this value as needed

\subsection{RSA-based binding site clustering}

\autoref{fig:bss_clust_profiles} depicts the four clusters defined by our method and the RSA profiles of the sites within them while \autoref{fig:bss_clusts_examples} illustrates six binding sites from each cluster to highlight the range of binding site size. Cluster 1 includes 46 sites, 127 sites are found on C2, 91 in C3 and 29 in C4. The proportion of residues with an RSA \textless~25\% in \autoref{fig:bss_clust_profiles} follows a different profile in each cluster, which is confirmed in \autorefpanel{fig:bs_clusts_feats}{A}. C1 is the most buried with a proportion of residues with RSA \textless~25\% of 0.68, ($p_{buried}$ = 0.68), followed by C2 with $p_{buried}$ = 0.47, then C3, $p_{buried}$ = 0.30, and lastly C4 ($p_{buried}$ = 0.10). \autorefpanel{fig:bs_clusts_feats}{B} displays the difference in binding site size between the clusters. There is variation within clusters in site size, but certain patterns are still apparent. C1 includes the largest sites, with an average size of $\bar{s}$ = 15 residues, followed by C2 with $\bar{s}$ = 11, then C3 with $\bar{s}$ = 8, and finally C4 with $\bar{s}$ = 5. \autorefpanel{fig:bs_clusts_feats}{C} shows the two-dimensional MDS representation of the binding sites, also shown in \autorefpanel{fig:fragsys_bs_clust}{D}, and \autorefpanel{fig:MLP_CV_blind_test}{C} in the Methods section of this chapter. While there is some overlap between neighbouring clusters, C1 and C4 are significantly different. Sites near the cluster borders are those that switch groups depending on the random initialisation of the clustering. To summarise, C1 includes on average the largest, most buried sites, whereas C4 includes the smallest and most accessible. C2 and C3 are not as different as C1 and C4, but still differ in size and burial proportion with C2 including larger and overall, less accessible sites than C3.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/FIG4_bs_clust_results_SPLIT_1.png}
    \captionof{figure}[Profiles of RSA-based binding site clusters]{\textbf{Profiles of RSA-based binding site clusters.} RSA profiles of the 293 binding sites that were grouped in four, C1-C4, clusters by $K$-means based on the difference between their RSA profiles ($U_{D}$). N\textsubscript{C1} = 46; N\textsubscript{C2} = 127; N\textsubscript{C3} = 91; N\textsubscript{C4} = 29. Each binding site is represented by a vector, plotted as a bar here. The elements of the vector represent the residues that form the binding site and are sorted according to their RSA, so buried residues are at the beginning of the vector (bottom), and more accessible residues towards the end (top). Each element of the vector, or section of the bar, is coloured according to RSA, using the matplotlib  \textit{cividis} colour palette. Within each cluster, binding sites are sorted based on the number of amino acids. Over each cluster, a line is drawn at RSA = 25\%.}
\label{fig:bss_clust_profiles}
\end{figure}

These results support $U_{D}$ as a metric that effectively quantifies the difference in solvent exposure profile across binding sites resulting in four clusters which differ not only in RSA, but also binding site size. This effect might be explained by the negative correlation between solvent accessibility and binding site size shown in \autoref{fig:bss_feats_corr}.

\autorefpanel{fig:bs_clusts_feats}{D} shows the RSA distribution across the four clusters. C1 presents a strikingly different distribution to the rest of clusters, peaking at RSA $\approx$ 5\%, indicating a high density of buried residues. C2 still presents an excess of buried residues relative to  C3-4, though not as high as C1. C4 presents the most different distribution to C1, peaking at RSA $\approx$ 50-70\%. This definition agrees with \autoref{fig:bss_clust_profiles} and \autorefpanel{fig:bs_clusts_feats}{A}. 

\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/FIG4_bs_clust_results_SPLIT_2.png}
    \captionof{figure}[Examples of RSA-based binding site clusters]{\textbf{Examples of RSA-based binding site clusters.} Six examples of binding sites are shown in structure for each cluster. Examples were selected to represent the range of binding site sizes within each cluster. Binding site residues are coloured according to their RSA, using the \textit{cividis} colour scheme. The rest of the protein is coloured in white. Ligands binding to the site in question are coloured in red.  IDs are UniProt accession codes and PDB IDs are provided for protein representative structures. C1 -- \href{https://www.uniprot.org/uniprotkb/Q9UKK9/entry}{Q9UKK9} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qjl}{5QJL} \cite{PDB_5QJL}, \href{https://www.uniprot.org/uniprotkb/Q9UGL1/entry}{Q9UGL1} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5fz0}{5FZ0} \cite{PDB_5FZ0}, \href{https://www.uniprot.org/uniprotkb/Q9UJM8/entry}{Q9UJM8} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qib}{5QIB} \cite{PDB_5QIB}, \href{https://www.uniprot.org/uniprotkb/Q8WS26/entry}{Q8WS26} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qpm}{5QPM} \cite{PDB_5QPM}, \href{https://www.uniprot.org/uniprotkb/Q7LBC6/entry}{Q7LBC6} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5ran}{5RAN} \cite{PDB_5RAN}, \href{https://www.uniprot.org/uniprotkb/Q5T0W9/entry}{Q5T0W9} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qhn}{5QHN} \cite{PDB_5QHN}; C2 -- \href{https://www.uniprot.org/uniprotkb/P11838/entry}{P11838} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5r1y}{5R1Y} \cite{WOLLENHAUPT_2020_F2X}, \href{https://www.uniprot.org/uniprotkb/P0DTD1/entry}{P0DTD1} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5s4b}{5S4B} \cite{SCHULLER_2021_NSP3}, \href{https://www.uniprot.org/uniprotkb/Q8IU60/entry}{Q8IU60} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qp9}{5QP9} \cite{PDB_5QP9}, \href{https://www.uniprot.org/uniprotkb/P18031/entry}{P18031} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qdu}{5QDU} \cite{KEEDY_2018_PTP1B}, \href{https://www.uniprot.org/uniprotkb/P0C024/entry}{P0C024} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qgi}{5QGI} \cite{PDB_5QGI}, \href{https://www.uniprot.org/uniprotkb/Q9UKK9/entry}{Q9UKK9} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qjl}{5QJL}; C3 -- \href{https://www.uniprot.org/uniprotkb/P47811/entry}{P47811} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5ra5}{5RA5} \cite{NICHOLS_2020_MINING}, \href{https://www.uniprot.org/uniprotkb/O15178/entry}{O15178} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qsa}{5QSA} \cite{PDB_5QSA}, \href{https://www.uniprot.org/uniprotkb/Q6B0I6/entry}{Q6B0I6} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5phl}{5PHL} \cite{PEARCE_2017_PANDDA}, \href{https://www.uniprot.org/uniprotkb/P47811/entry}{P47811} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5ra5}{5RA5}, \href{https://www.uniprot.org/uniprotkb/Q92835/entry}{Q92835} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5ry0}{5RY0} \cite{BRADSHAW_2024_SHIP1}, \href{https://www.uniprot.org/uniprotkb/P18031/entry}{P18031} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qdu}{5QDU}; C4 -- \href{https://www.uniprot.org/uniprotkb/Q9BRQ3/entry}{Q9BRQ3} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5ra5}{5RA5} \cite{PDB_5R55}, \href{https://www.uniprot.org/uniprotkb/Q8WWQ0/entry}{Q8WWQ0} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5rju}{5RJU} \cite{PDB_5RJU}, \href{https://www.uniprot.org/uniprotkb/Q32ZE1/entry}{Q32ZE1} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5rhi}{5RHI} \cite{PDB_5RHI}, \href{https://www.uniprot.org/uniprotkb/O95696/entry}{O95696} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5pny}{5PNY} \cite{PEARCE_2017_PANDDA}, \href{https://www.uniprot.org/uniprotkb/Q7LBC6/entry}{Q7LBC6} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5ran}{5RAN}, \href{https://www.uniprot.org/uniprotkb/Q6PJP8/entry}{Q6PJP8} - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5q1z}{5Q1Z} \cite{PDB_5Q1Z}.}
\label{fig:bss_clusts_examples}
\end{figure}

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/FIG5_bs_clust_feats_FULL.png}
    \caption[Binding site cluster features]{\textbf{Binding site cluster features.} \textbf{(A)} Box plot of the proportion of residues with RSA \textless~25\% per binding site across the four clusters defined by $K$-means clustering;\textbf{(B)} Box plot of the binding site size, in amino acids, across clusters. Pairwise Mann-Whitney-Wilcoxon tests were performed to assess the differences between the clusters. Boxes represent the IQR, and whiskers extend to 1.5 $\times$ IQR. $p$-value annotation legend - \textit{ns}: $p >$ 0.05, *: 0.01 $< p \leq$ 0.05, **: $10^{-2} < p \leq 10^{-3}$, ***: $10^{-4} < p \leq 10^{-3}$, ****: $p \leq 10^{-4}$; \textbf{(C)} MDS representation of the 293 binding sites on 2 dimensions. Data points represent binding sites and are coloured based on the cluster they group in; \textbf{(D)} Histogram of RSA (\%) of the residues found within the ligand binding sites in each cluster; \textbf{(E)} Histogram of $N_{Shenkin}$ within cluster residues; \textbf{(F)} MES histogram plots for the 4 clusters defined.}
    \label{fig:bs_clusts_feats}
\end{figure}

To further characterise the defined clusters, the distributions of the normalised Shenkin divergence score ($N_{Shenkin}$) and Missense Enrichment Score (MES) of the residues found in the clusters were analysed. Regarding evolutionary divergence (\autorefpanel{fig:bs_clusts_feats}{E}), C1 also presents a different distribution to the rest of the clusters, with a peak at $N_{Shenkin} \approx$ 5, i.e., most of the residues conforming the sites within this cluster are highly conserved. The other clusters present flatter distributions with increasing proportion of divergent residues ($N_{Shenkin} >$ 25) $p_{C2}$ = 0.55, $p_{C3}$ = 0.67, and $p_{C4}$ = 0.69. $N_{Shenkin}$ is a divergence score ranging from 0-100, therefore residues with $N_{Shenkin} \leq$ 25, $p_{C1}$ = 0.58, $p_{C2}$ = 0.45, $p_{C3}$ = 0.33, and $p_{C4}$ = 0.31, represent stronger residue conservation, or lower divergence, than $N_{Shenkin} >$ 25. This agrees with the pattern observed on the RSA distributions (\autorefpanel{fig:bs_clusts_feats}{D}), as buried residues tend to be evolutionarily conserved \cite{CHOTHIA_1986_CONSERVATION, RUSSELL_1994_UNCONSERVATION}. In terms of missense depletion (\autorefpanel{fig:bs_clusts_feats}{F}), the distribution of C1 is slightly shifted to the left, towards more negative values, i.e., more missense depleted residues, with $\overline{\text{MES}}_{C1}$ = \textminus 0.17. The distributions of C2-4 are not statistically different, but present increasing average missense enrichment scores: $\overline{\text{MES}}_{C2}$ = \textminus 0.07, $\overline{\text{MES}}_{C3}$ = \textminus 0.02, and $\overline{\text{MES}}_{C1}$ = +0.06. Once again, this pattern agrees with the ones observed with site size, solvent accessibility, and evolutionary divergence. Sites that are more buried tend to be bigger in size, more conserved across homologues, as well as depleted in missense variation in human.

%\vspace{-8pt} % Adjust this value as needed

\subsection{Clusters predict differential functional enrichment}

A key goal of the work presented in this chapter is to identify which sites from a fragment screening experiment are most likely to be functional and so worth investigating further. \autoref{fig:bs_clusts_enrichment} shows the relative enrichment in functional sites across the four defined clusters. C1 is the most enriched in functional sites, with 17/46 sites being classed as of \textit{known function}, (OR = 4.46, $p \approx$ 0). C2 was next with 21/127 (OR = 1.15, $p$ = 0.75). C3 with 6/91 is depleted relative to the other clusters, (OR = 0.33, $p$ = 0.01), and finally C4 with 0/29, (OR = 0.16, $p$ = 0.04). RSA-based defined clusters are differentially enriched in functional sites. Based on their enrichment, a binding site found in C1 is $\approx$4, $\approx$14, and $\approx$28-fold more likely to be functional than a site in C2, C3, and C4, respectively.

Functional definitions in UniProt tend to lag behind the literature. A literature search found support for 12 sites in C1 that are without UniProt annotations with two examples discussed below. We found no literature support for the remaining seventeen sites in C1 suggesting they may be novel, functionally important sites.  \autoref{tab:novel_c1_sites} shows the full list of C1 sites that are predicted to be functionally important with 2/17 examples discussed below.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.50\textwidth]{figures/ch_FRAGSYS/PNG/FIG6_bs_clust_func_enrichment.png}
    \caption[Binding site cluster enrichment in known functional sites]{\textbf{Binding site cluster enrichment in known functional sites.} This enrichment score is an odds ratio (OR). Error bars indicate 95\% CI of the OR. Y-axis is in $\log_{10}$ scale. A pseudo-count of 1 was added to each cell of the contingency table, to be able to calculate the score.}
    \label{fig:bs_clusts_enrichment}
\end{figure}

\subsection{Example C1 site functional predictions supported by literature but not annotated in UniProt}

\subsubsection{Zika virus NS3}

The Zika virus (ZIKV) genome polyprotein (\href{https://www.uniprot.org/uniprotkb/Q32ZE1/entry}{Q32ZE1}) is 3419 amino acids long and codes for three structural proteins: capsid (C), envelope (E), and membrane (M) as well as seven non-structural proteins: NS1, NS2A, NS2B, NS3, NS4A, NS4B, and NS5. NS3 is a critical serine proteinase for viral polyprotein processing and genomic regulation. It includes a protease domain at the N-terminus, and a helicase domain on the C-terminus. The helicase is responsible for RNA unwinding during replication, and thus makes an interesting drug target against ZIKV \cite{LUO_2015_FLAVIVIRUS}.

There are 10 sites in NS3 identified from 17 structures with 17 unique ligands and all are functionally unannotated in UniProt. The analysis here shows Binding Site 7 (BS7) to lie in Cluster 1 and so is likely to be functional.

The site is located between domains I-III, involving residues from $\eta$2, $\alpha$3 on domain I, and $\alpha$10, $\alpha$11 on domain III as defined in \cite{TIAN_2016_ZIKV} (\autoref{fig:c1_example_1}). Mottin \textit{et al.} \cite{MOTTIN_2017_ZIKA_HELICASE} predicted four RNA binding sites on NS3. One of them, the RNA exit crevice is located between domains I‚ÄìIII, and involves $\alpha$3, $\alpha$10 residues. Raubenolt \textit{et al.} \cite{RAUBENOLT_2021_ZIKA_ALLOSTERIC} probed four different allosteric sites on this protein. One of them, D3, was manually curated, included $\alpha$11, $\alpha$12, and overlapped with BS7. Later, Durgam and Guruprasad \cite{DURGAM_2022_ZIKA_ATP} stated that four of the ten residues forming this site: Ala264, Thr265, Lys537 and Asp540 bind to RNA when in complex with NS3. These results strongly suggest that this region plays an important role in RNA binding to NS3 and so is a good candidate site to target to modulate function. Moreover, the site is on average missense-depleted: MES =  \textminus0.28. A264 ($N_{Shenkin}$ = 18, MES = \textminus0.79), T267 ($N_{Shenkin}$ = 53, MES = \textminus0.55), and S293 ($N_{Shenkin}$ = 72, MES =  \textminus0.48) are the three key positions out of the 10 forming this binding site, as they are all constrained within the human orthologs of this protein. A264 is conserved across homologues, whereas T267 and S293 are divergent yet missense depleted, so could be important for binding specificity.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/FIG7_c1_example_1.png}
    \caption[Binding Site 7 of Zika virus NS3]{\textbf{Binding Site 7 of Zika virus NS3.} Non-structural protein NS3 of Zika virus (\href{https://www.uniprot.org/uniprotkb/Q32ZE1/entry}{Q32ZE1}) binding to N-(2-methoxy-5-methylphenyl)glycinamide (\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/NY7}{NY7}) in BS7 (PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5RHG}{5RHG}) \cite{PDB_5RHG}. Domains I, II, and III are coloured in pink, blue, and green respectively. Binding Site 7, which is in Cluster 1, is highlighted. The other 9 binding sites, which fall in C2 (3), C3 (3) and C4 (3), are hidden. Ligand binding residues in red, and NY7 in yellow. Protein-ligand interactions are represented by black lines.}
    \label{fig:c1_example_1}
\end{figure}

\vspace{-20pt} % Adjust this value as needed

\subsubsection{SARS-CoV-2 NSP13}

The Severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) replicase polyprotein 1ab (\href{https://www.uniprot.org/uniprotkb/P0DTD1/entry}{P0DTD1}) is 7096 amino acids long and codes for 16 non-structural proteins \cite{NAQVI_2020_SARSCOV2}. NSP13 is a helicase that unwinds dsRNA in the 5‚Äô-3‚Äô direction to provide a single-stranded template for viral RNA amplification \cite{YUE_2022_SARSCOV2}. NSP13 also has NTPase activity, which provides the energy for the RNA unwinding \cite{SHU_2020_SARSCOV2}. NSP13 plays a fundamental role in the replication and transcription of the SARS-CoV-2 genome and is thought to be a good drug target against SARS-CoV-2 virus infection \cite{ZENG_2021_SARSCOV2}. NSP13 has five domains. Two ``RecA like'' subdomains 1A and 2A, in charge of nucleotide binding and hydrolysis, as well as three other domains: an N-terminal zinc-binding domain, the helical ``stalk'' domain, and a beta-barrel 1B domain \cite{ROMEO_2022_SARSCOV2}. It is the most conserved protein across coronaviruses, with sequence identity $>$ 99\% \cite{RICCI_2022_SARSCOV2}.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/FIG7_c1_example_2.png}
    \caption[Binding Site 6+16 of SARS-CoV-2 NSP13]{\textbf{Binding Site 6+16 of SARS-CoV-2 NSP13.} Non-structural protein NSP13 of SARS-CoV-2 (\href{https://www.uniprot.org/uniprotkb/P0DTD1/entry}{P0DTD1}) binding to 3 ligands in BS6+16 (Cartoon PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5RMH}{5RMH}) \cite{NEWMAN_2021_SARSCOV2}. 1A, 1B, 2A, stalk and zinc domains are coloured in yellow, pink, green, brown, and grey respectively. Ligand binding residues in red, and ligands in yellow. Interactions are not shown here for simplicity.}
    \label{fig:c1_example_2}
\end{figure}

Twenty-four sites are defined on the surface of NSP13. Two binding sites were identified as C1: BS6, and BS16 (\autoref{fig:c1_example_2}). Visual inspection shows the two sites to be adjacent with a total of 16 residues. Three fragments bind to the site, which is located in the nucleotide and RNA binding interface of NSP13 between the 1B and 2A domains. This is the region where the 5‚Äô end of the RNA binds \cite{YAN_2020_SARSCOV2}. This pocket is determined to be highly druggable, and drugs binding to it might be effective against other coronaviruses, due to the pocket‚Äôs high amino acid conservation \cite{NEWMAN_2021_SARSCOV2}. This agrees with our results, as this site has an average $N_{Shenkin}$ = 32, and MES = \textminus0.18. Of the 16 positions in this site, four show high conservation across homologues and missense depletion in human: P514 ($N_{Shenkin}$ = 30, MES = \textminus0.56), D534 ($N_{Shenkin}$ = 9, MES = \textminus0.56), T552 ($N_{Shenkin}$ = 48, MES = \textminus1.87), and H554 ($N_{Shenkin}$ = 36, MES = \textminus0.85). T552 shows highest conservation across species and lowest missense enrichment (\textminus1.87) and so is most likely to have a key function in this protein family.

\subsection{Examples of potentially novel C1 cluster functional predictions}

\subsubsection{Human tenascin}

Human tenascin, TN, (\href{https://www.uniprot.org/uniprotkb/P24821/entry}{P24821}) is a hexameric extracellular matrix glycoprotein implicated in a variety of functions including cell migration, cell attachment, matrix assembly and proinflammatory cytokine synthesis \cite{BHATTACHARYYA_2022_TNC}. TN is known to interact with viruses and play a role in viral infections, e.g., HIV-1, and has been reported as a biomarker for disease severity \cite{ZULIANI_2023_TNC}. It also plays a key role in wound healing \cite{WANG_2022_TNC}, is involved in diverse cardiovascular diseases \cite{KHOMTCHOUK_2022_TNC} and in breast cancer \cite{LEPUCKI_2022_TNC}. For these reasons, there is considerable effort put into understanding better the function of TN and targeting it for therapeutic effect.

The data includes 11 structures with 11 unique ligands binding to TN, grouped in four binding sites, none annotated as functional in UniProt. One of the four binding sites is in C1, and so predicted to be of functional importance. The site is found on the Fibrinogen C-terminal domain of the protein, which functions as a molecular recognition unit that interacts with either proteins or carbohydrates (\autoref{fig:c1_example_3}). This site shows high conservation across species ($N_{Shenkin}$ = 15), as well as missense-depletion in human (MES = \textminus0.33). Accordingly, this site is likely to be of relevance for TN function. Within the 15 positions forming the site, V2012 ($N_{Shenkin}$ = 5, MES = \textminus1.0), G2046 ($N_{Shenkin}$ = 0, MES = \textminus0.67), F2047 ($N_{Shenkin}$ = 0, MES = \textminus0.67), W2055 ($N_{Shenkin}$ = 0, MES = \textminus0.54) and G2057 ($N_{Shenkin}$ = 0, MES = \textminus0.83) are the most critical interacting residues and highly conserved across homologues.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/FIG7_c1_example_3.png}
    \caption[Binding Site 0 of human tenascin]{\textbf{Binding Site 0 of human tenascin.} Human tenascin, TN, (\href{https://www.uniprot.org/uniprotkb/P24821/entry}{P24821}) binding to 8 ligands in BS0. (Cartoon PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5r60}{5R60} \cite{PDB_5R60}). A, B, and P subdomains as defined by Yee, \textit{et al.} \cite{YEE_1997_FIBRINOGEN} are coloured in blue, grey, and green respectively.}
    \label{fig:c1_example_3}
\end{figure}

\subsubsection{Human 5-aminolevulinate synthase}

\textit{ALAS2} is a gene located on the X chromosome that codes for the human mitochondrial erythroid-specific 5-aminolevulinate synthase, ALAS-E, (\href{https://www.uniprot.org/uniprotkb/P22557/entry}{P22557}). This dimeric enzyme carries out the first and rate-limiting step of the haem synthesis pathway: the pyridoxal 5‚Äô-phosphate (PLP)-dependent condensation of succinyl-CoA and glycine to form aminolaevulinic acid \cite{AKHTAR_1976_PORPHYRIN}. Across eukaryotes, these enzymes have developed extensions surrounding the catalytic core on both the N and C-termini \cite{MUNAKATA_1993_AMINOLEVULINATE}. The N-terminal extensions include the mitochondrial targeting sequence \cite{SRIVASTAVA_1988_AMINOLEVULINATE}, whereas the C-terminal extension (C-ext) plays an autoinhibitory role by regulating substrate binding and product release \cite{BAILEY_2020_AMINOLEVULINATE}. Mutations affecting C-ext can result in gain-of-function, such as X-linked protoporphyria \cite{WHATLEY_2008_AMINOLEVULINATE}, as well as loss-of-function disorders, e.g., X-linked sideroblastic anaemia \cite{DUCAMP_2011_SIDEROBLASTIC}. Accordingly, ALAS-E is a potential therapeutic target for the treatment of such diseases.

Thirty-three unique ligands across 25 ALAS-E structures were grouped in ten binding sites, only one of which was annotated as functional in UniProt. Three sites were classified as C1. Two are known to be on the interface between subunits, form key interactions to maintain the assembly and are close to the PLP binding site \cite{BAILEY_2020_AMINOLEVULINATE}. However, one (BS1) is not mentioned in the literature. This site is located on a deep pocket at the N-terminal region of the protein structure (\autoref{fig:c1_example_4}). Residues in this site are strongly conserved and depleted in missense variation: $N_{Shenkin}$ = 29, MES = \textminus0.13. Together, this suggests the site has a functional role in the protein, perhaps as an allosteric regulator, or through interaction with a partner such as succinate-CoA ligase, SCS-$\alpha$ (\href{https://www.uniprot.org/uniprotkb/Q96I99/entry}{Q96I99}) \cite{FURUYAMA_2000_SIDEROBLASTIC}. Out of the 16 residues forming the site, K381 ($N_{Shenkin}$ = 38, MES = \textminus0.94) is the most missense-depleted position in the site and should be considered for lead optimisation of a fragment binding to this site.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/FIG7_c1_example_4.png}
    \caption[Binding Site 1 of human ALAS-E]{\textbf{Binding Site 1 of human ALAS-E.} Human erythroid-specific mitochondrial 5-amino\-levulinate synthase, ALAS-E, (\href{https://www.uniprot.org/uniprotkb/P22557/entry}{P22557}) binding to 7 ligands in BS1. (Cartoon PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5qr0}{5QR0} \cite{PDB_5QR0}). Subunits A, B, C-terminal extensions A, B, as well as PLP cofactors are coloured in grey, beige, green, orange, and purple, respectively. Ligand binding residues in red, and ligands in yellow.}
    \label{fig:c1_example_4}
\end{figure}

\vspace{-13pt} % Adjust this value as needed
%\vspace{-7pt} % Adjust this value as needed

\section{Discussion}

On this chapter, a method has been presented to identify binding sites from fragment screening data and group the sites into four robust clusters by an RSA profile metric. 29/46 sites in Cluster 1 have functional support from the literature (UniProt 17/46 -- \autoref{tab:literature_c1_sites}; literature search 12/46 -- \autoref{tab:novel_c1_sites}). Seventeen further sites have similar profiles, but no evidence of their functional significance was found in the literature. Two examples from this set are shown that have compelling support from conservation and missense depletion scores of functional significance. Furthermore, all sites are listed in \autoref{tab:novel_c1_sites} as a resource for further experimentation on these proteins.

As a case study, this method was applied to the SARS-CoV-2 main protease, MPro (\href{https://www.uniprot.org/uniprotkb/P0DTD1/entry}{P0DTD1}). Twenty-five sites were defined from 511 structures, from which 8 were classed as C1, 12 as C2, 3 as C3 and only 2 as C4. Of the 8 C1 sites, one corresponds to the active site and three to allosteric sites 1, 2, and 3 \cite{DASGUPTA_2022_ALLOSTERIC} respectively. A further C1 site is at the dimer interface and known to be a potential allosteric site \cite{DOUANGAMATH_2020_SARSCOV2} (\autoref{fig:MPro_showcase}). The remaining three C1 sites may be important, but each binds only a single ligand and their function is currently unclear.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_FRAGSYS/PNG/SUPP_FIG4_MPro_C1_sites.png}
    \caption[SARS-CoV-2 MPro fragment screening]{\textbf{SARS-CoV-2 MPro fragment screening.} \textbf{(A)} Twenty-five defined ligand binding sites on the SARS-CoV-2 main protease, MPro (\href{https://www.uniprot.org/uniprotkb/P0DTD1/entry}{P0DTD1}) from 971 ligands from 511 structures; \textbf{(B)} Five of the 9 C1 sites included the known MPro active site, and four known potential allosteric sites \cite{DOUANGAMATH_2020_SARSCOV2, DASGUPTA_2022_ALLOSTERIC}. Surface PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/5R7Y}{5R7Y} \cite{DOUANGAMATH_2020_SARSCOV2}.}
    \label{fig:MPro_showcase}
\end{figure}

This chapter focuses on a small set of proteins heavily studied by fragment screening methods. However, the method described here can be applied to classify any ligand binding site or predicted site. Accordingly, future work will seek to classify all known ligand binding sites in the PDBe and provide tools to predict the likely functional class of sites predicted by tools such as P2Rank \cite{KRIVAK_2018_P2RANK}, or GRaSP \cite{SANTANA_2020_GRaSP, SANTANA_2022_GRaSP} from AlphaFold2 \cite{JUMPER_2021_ALPHAFOLD, VARADI_2022_ALPHAFOLDDB} or other 3D structure models.

It is natural to focus on sites that are most likely to be of functional significance and so possible targets to modulate function. However, binding sites identified here that are predicted to be least likely to have function may also be interesting as good locations for tagging proteins for degradation \cite{BEKES_2022_PROTACS}, phosphorylation \cite{SIRIWARDENA_2020_PHOSPHO}, dephosphorylation \cite{SIMPSON_2023_ADPROM}, or other modulation \cite{HEITEL_2023_PTMS, PENG_2023_PTMS}.

\begin{landscape}
\begin{longtable}{|M{22mm}|M{17mm}|M{20mm}|M{12mm}|M{12mm}|M{10mm}|M{10mm}|M{48mm}|M{48mm}|}
\hline
\textbf{UniProt ID} & \textbf{RSA} (\%)  & \textbf{N\textsubscript{Shenkin}} & \textbf{MES}   & \textbf{p}    & \textbf{\# aas} & \textbf{\# ligs} & \textbf{UniProt residue numbers}                                                                                       & \textbf{Literature support}                                                   \\ \hline
\endfirsthead
\multicolumn{9}{c}%
{{\bfseries Table \thetable} (continued)} \\
\hline
\textbf{UniProt ID} & \textbf{RSA} (\%)  &\textbf{N\textsubscript{Shenkin}} & \textbf{MES}   & \textbf{p}    & \textbf{\# aas} & \textbf{\# ligs} & \textbf{UniProt residue numbers}                                                                                       & \textbf{Literature support}
%
\endhead
%
\href{https://www.uniprot.org/uniprotkb/Q32ZE1/entry}{Q32ZE1}     & 17.4 & 38.4     & \textminus0.21 & 0.02 & 10          & 1          & 1762, 1763, 1765, 1766, 1769, 1791, 1991, 2034, 2035, 2038                                              & RNA binding \cite{DURGAM_2022_ZIKA_ATP}, RNA exit site \cite{MOTTIN_2017_ZIKA_HELICASE}, D3 site \cite{RAUBENOLT_2021_ZIKA_ALLOSTERIC}         \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q9Y2J2/entry}{Q9Y2J2}     & 14.6 & 38.2     & +0.01  & 0.84 & 15          & 1          & 117, 118, 119, 203, 206, 207, 210, 231, 232, 235, 236, 253, 282, 283, 286                               & GPC binding \cite{HAN_2000_CYTOSKELETON}                                                 \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q9Y2J2/entry}{Q9Y2J2}     & 13.4 & 43.3     & +0.02  & 0.7  & 21          & 4          & 154, 161, 162, 163, 164, 185, 186, 189, 208, 212, 217, 295, 297, 298, 299, 300, 301, 315, 375, 376, 379 & Calmodulin binding \cite{HAN_2000_CYTOSKELETON}                                          \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q8WS26/entry}{Q8WS26}     & 16.2 & 28.9     & \textminus0.22 & 0.26 & 19          & 2          & 105, 106, 107, 108, 109, 112, 151, 154, 155, 158, 159, 162, 170, 171, 173, 174, 175, 176, 179           & IPP, DMAPP binding \cite{MUNZKER_2020_FARNESYL, GABELLI_2006_FARNESYL}                                      \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q8WS26/entry}{Q8WS26}     & 22.1 & 31       & +0.18  & 0.58 & 8           & 2          & 308, 312, 315, 316, 320, 324, 384, 423                                                                  & IPP binding \cite{GABELLI_2006_FARNESYL}                                                 \\ \hline
\href{https://www.uniprot.org/uniprotkb/P18031/entry}{P18031}     & 20.8 & 33.9     & +0.05  & 0.48 & 14          & 1          & 1, 2, 3, 4, 6, 10, 19, 242, 243, 244, 245, 246, 247, 271                                                & Conformational change \cite{KEEDY_2018_PTP1B}, Cluster II \cite{CUI_2017_ALLOSTERIC}                   \\ \hline
\href{https://www.uniprot.org/uniprotkb/P47811/entry}{P47811}     & 17.1 & 55       & +0.08  & 0    & 19          & 10         & 191, 192, 197, 198, 232, 236, 242, 246, 249, 250, 251, 252, 255, 259, 291, 292, 293, 294, 296           & MAP insert motif, Trp197 pocket \cite{FRANCIS_2013_P38A, NICHOLS_2020_P38A} \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q6B0I6/entry}{Q6B0I6}     & 15.8 & 41.8     & +0.12  & 0.43 & 12          & 5          & 193, 224, 225, 227, 228, 239, 240, 241, 242, 243, 277, 279                                              & Cryptic binding site \cite{PEARCE_2017_CRYPTIC}                                        \\ \hline
\href{https://www.uniprot.org/uniprotkb/P0DTD1/entry}{P0DTD1}     & 12.9 & 34.3     & \textminus0.13 & 0.45 & 12          & 2          & 5501, 5503, 5809, 5810, 5811, 5838, 5839, 5840, 5841, 5856, 5858, 5878                                  & RNA binding \cite{NEWMAN_2021_SARSCOV2} \\ \hline
\href{https://www.uniprot.org/uniprotkb/P0DTD1/entry}{P0DTD1}     & 22.3 & 51.5     & \textminus0.04 & 0.87 & 9           & 1          & 5806, 5809, 5810, 5811, 5839, 5874, 5876, 5878, 5879                                                    & RNA binding \cite{NEWMAN_2021_SARSCOV2} \\ \hline
\href{https://www.uniprot.org/uniprotkb/P22557/entry}{P22557}     & 16   & 47.8     & \textminus0.09 & 0.61 & 16          & 10         & 148, 152, 155, 267, 268, 271, 272, 409, 413, 506, 570, 572, 573, 574, 575, 576                          & Dimerisation interface \cite{BAILEY_2020_AMINOLEVULINATE} \\ \hline
\href{https://www.uniprot.org/uniprotkb/P22557/entry}{P22557}     & 12.7 & 53.1     & +0.08  & 0.61 & 7           & 2          & 271, 293, 294, 295, 296, 297, 575                                                                       & Conformational change, PLP binding, succinyl-CoA inhibition \cite{BAILEY_2020_AMINOLEVULINATE} \\ \hline
\caption[Literature supported C1 sites]{\textbf{Literature supported C1 sites.} These are 12 C1 sites with no functional annotations in UniProt, therefore labelled as \textit{unknown function}, for which literature has been found that support their functional relevance. UniProt ID indicates the protein UniProt accession. RSA is the median site RSA (\%). $N_{Shenkin}$ is the average normalised Shenkin score for the site. MES is the average missense enrichment score for the site. $p$ is the $p$-value associated to this site MES. \# aas is the number of residues forming the site. \# ligs is the number of ligands binding to the site. UniProt residue numbers is a list of the UniProt residue numbers of the residues forming the site. Literature support contains a brief description of the literature-reported site function and references.}
\label{tab:literature_c1_sites}\\
\end{longtable}
\end{landscape}

\begin{landscape}
\begin{longtable}{|M{22mm}|M{17mm}|M{18mm}|M{13mm}|M{13mm}|M{10mm}|M{10mm}|M{96mm}|}
\hline
\textbf{UniProt ID} & \textbf{RSA} (\%)  & \textbf{N\textsubscript{Shenkin}} & \textbf{MES}   & \textbf{p}    & \textbf{\# aas} & \textbf{\# ligs} & \textbf{UniProt residue numbers}                                                                                                                                    \\ \hline
\endfirsthead
\multicolumn{8}{c}%
{{\bfseries Table \thetable} (continued)} \\
\hline
\textbf{UniProt ID} & \textbf{RSA} (\%)  & \textbf{N\textsubscript{Shenkin}} & \textbf{MES}   & \textbf{p}    & \textbf{\# aas} & \textbf{\# ligs} & \textbf{UniProt residue numbers}                                                                                      
%
\endhead
%
\href{https://www.uniprot.org/uniprotkb/Q5T0W9/entry}{Q5T0W9}     & 22.4 & 36.2     & \textminus0.24 & 0.08 & 12          & 10         & 149, 150, 151, 177, 233, 234, 235, 236, 270, 273, 274, 277                                         \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q5T0W9/entry}{Q5T0W9}     & 9.7  & 38.6     & \textminus0.05 & 0.79 & 12          & 2          & 125, 126, 127, 129, 229, 255, 256, 257, 272, 275, 276, 279                                         \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q8WVM7/entry}{Q8WVM7}     & 19.8 & 57.7     & \textminus0.23 & 0.62 & 5           & 1          & 285, 288, 322, 325, 326                                                                            \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q15047/entry}{Q15047}     & 18.1 & 12.4     & +0.08  & 0.78 & 18          & 2          & 295, 296, 297, 298, 300, 301, 302, 324, 328, 329, 330, 332, 333, 357, 389, 392, 393, 394           \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q8WS26/entry}{Q8WS26}     & 19.5 & 57.3     & \textminus0.11 & 0.57 & 21          & 26         & 84, 87, 88, 89, 90, 214, 217, 218, 221, 222, 225, 268, 269, 273, 277, 281, 285, 290, 295, 299, 303 \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q9UGL1/entry}{Q9UGL1}     & 28.7 & 31.3     & \textminus0.09 & 0.66 & 10          & 1          & 53, 57, 506, 582, 583, 606, 607, 609, 610, 613                                                     \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q9UGL1/entry}{Q9UGL1}     & 16.6 & 34       & \textminus0.01 & 1    & 12          & 3          & 658, 659, 662, 663, 666, 667, 670, 701, 736, 737, 738, 741                                         \\ \hline
\href{https://www.uniprot.org/uniprotkb/P15379/entry}{P15379}     & 18.3 & 19.4     & +0.09  & 0.63 & 11          & 1          & 23, 24, 40, 41, 50, 146, 148, 162, 163, 164, 165                                                   \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q9UJM8/entry}{Q9UJM8}     & 24.3 & 42.8     & \textminus0.11 & 0.86 & 6           & 1          & 5, 11, 323, 327, 328, 331                                                                          \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q6B0I6/entry}{Q6B0I6}     & 21.9 & 36.6     & \textminus0.15 & 0.68 & 4           & 1          & 50, 209, 265, 285                                                                                  \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q6B0I6/entry}{Q6B0I6}     & 12.2 & 26       & \textminus0.06 & 0.84 & 7           & 1          & 44, 199, 275, 276, 297, 300, 303                                                                   \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q9UKK9/entry}{Q9UKK9}     & 9.8  & 29.6     & \textminus0.05 & 0.73 & 15          & 1          & 65, 66, 67, 69, 75, 77, 124, 125, 145, 146, 147, 175, 200, 205, 206                                \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q92835/entry}{Q92835}     & 16.5 & 33.7     & \textminus0.05 & 0.78 & 19          & 46         & 615, 616, 617, 618, 620, 621, 622, 624, 625, 630, 631, 632, 633, 634, 635, 636, 637, 638, 674      \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q92835/entry}{Q92835}     & 12.2 & 39.4     & +0.02  & 0.92 & 12          & 1          & 560, 561, 562, 570, 571, 572, 573, 574, 578, 817, 839, 840                                         \\ \hline
\href{https://www.uniprot.org/uniprotkb/Q96HY7/entry}{Q96HY7}     & 11.6 & 38.5     & +0.07  & 0.75 & 14          & 1          & 57, 58, 60, 61, 64, 105, 106, 107, 121, 122, 125, 126, 147, 151                                    \\ \hline
\href{https://www.uniprot.org/uniprotkb/P22557/entry}{P22557}     & 17.5 & 40.6     & +0.04  & 0.72 & 16          & 7          & 143, 145, 146, 149, 348, 349, 350, 351, 352, 353, 380, 381, 383, 402, 403, 406                     \\ \hline
\href{https://www.uniprot.org/uniprotkb/P24821/entry}{P24821}     & 14.2 & 24.4     & \textminus0.29 & 0    & 15          & 8          & 2010, 2011, 2012, 2025, 2045, 2046, 2047, 2048, 2049, 2050, 2054, 2055, 2056, 2057, 2060           \\ \hline

\caption[Novel C1 sites]{\textbf{Novel C1 sites.} These are 17 C1 sites with no functional annotations in UniProt, therefore labelled as \textit{unknown function}, without any literature support. These sites represent therefore novel predicted functional sites. UniProt ID indicates the protein UniProt accession. RSA (\%) is the median site RSA. $N_{Shenkin}$ is the average normalised Shenkin score for the site. MES is the average missense enrichment score for the site. $p$ is the $p$-value associated to this site MES. \# aas is the number of residues forming the site. \# ligs is the number of ligands binding to the site. UniProt residue numbers is a list of the UniProt residue numbers of the residues forming the site.}
\label{tab:novel_c1_sites}\\
\end{longtable}
\end{landscape}
