\chapter{Comparative evaluation of methods for the prediction of protein-ligand binding sites}

\section*{Preface}

XXX.

\section*{Publications}

This chapter will include the results published in here \textit{"Comparative evaluation of methods for the prediction of protein-ligand binding sites"}.

\section*{Author contributions}

G.J.B. and J.S.U., conceived, designed, and developed the research. J.S.U. analysed the data. J.S.U. developed the software. J.S.U. and G.J.B. wrote, reviewed and edited the manuscript. G.J.B. secured funding and supervised.

\section{Methods}

\subsection{Comparison of datasets}

Training and test datasets were downloaded for all machine learning based methods reviewed in this work. Datasets were compared to our reference set, LIGYSIS, in terms of number of sites per protein, ligand-interacting chains, chain lengths, size of the sites (number of amino acids), ligand composition, size and diversity. Ligand diversity was quantified by Shannon Entropy \cite{SHANNON_1948_ENTROPY} (Equation \ref{eq:entropy_shannon}) where pi represents the proportion of each ligand $i$ of the $R$ ligands observed in the dataset. Ligand data was extracted from the Chemical Component Dictionary (CCD) \cite{WESTBROOK_2015_CCD}. An overlap (\%) was calculated for each dataset as the proportion of LIGYSIS binding sites that were covered by at least one ligand in a test dataset. A simplistic approach was adopted by calculating the intersection of ligand IDs between LIGYSIS and each dataset. Ligand IDs were defined as a string of PDB ID + ``\_'' + ligand ID, e.g., ``6GXT\_GTP'' corresponds to the guanosine-5’-triphosphate of the PDB entry with ID: 6GX7 \cite{CAMPANACCI_2019_TUBULIN}.

\begin{equation}
H' = - \sum_{i=1}^{R} p_i \ln(p_i)
\label{eq:entropy_shannon}
\end{equation}
\myequations{Shannon Entropy}

\subsection{Training datasets}

VN-EGNN trains on a subset \cite{KANDEL_2021_PURESNET} of the sc-PDB (v2017) \cite{PAUL_2004_SCPDB, KELLENBERGER_2006_SCPDB, MESLAMANI_2011_SCPDB, DESAPHY_2015_SCPDB} (sc-PDB\textsubscript{SUB}). sc-PDB is a comprehensive database of pharmacological ligand-protein complexes. The database is comprised by proteins in complex with buried, biologically relevant synthetic or natural ligands deposited in the PDB. sc-PDB contains unique non-repeating protein-ligand pairs, which means that only one ligand is considered per PDB structure entry. Smith \textit{et al.} \cite{SMITH_2024_GrASP} enriched this dataset with 9,000 extra ligands resulting in a version of sc-PDB which we call sc-PDB\textsubscript{RICH}, which GrASP trained on. Unfortunately, this dataset is not publicly accessible and therefore not considered in our analysis. DeepPocket used the full sc-PDB set to train on, sc-PDB\textsubscript{FULL}. IF-SitePred uses a sequence identity-filtered version of the non-redundant subset of the binding mother of all databases (MOAD) \cite{HU_2005_BMOAD, BENSON_2008_BMOAD, AHMED_2015_BMOAD, SMITH_2019_BMOAD}, which considers only protein family leaders. The binding MOAD, here referred to as bMOAD\textsubscript{SUB}, is a large collection of crystal structures with clearly identified biologically relevant ligands with binding data extracted from the literature. Finally, P2Rank used the CHEN11 dataset to train, which aimed to cover all SCOP \cite{HUBBARD_1997_SCOP, HUBBARD_1998_SCOP, LOCONTE_2000_SCOP} families of ligand binding proteins in a non-redundant manner \cite{CHEN_2011_ASSESSMENT} and the JOINED dataset for validation. CHEN11 not only considers the ligands in each structure but is enriched with ligands binding to homologous structures. JOINED is a combined dataset formed by other smaller datasets: ASTEX \cite{HARTSHORN_2007_ASTEX}, UB48 \cite{HUANG_2006_BU48}, DT198 \cite{ZHANG_2011_METAPOCKET} and MP210 \cite{HUANG_2009_METAPOCKET}, which represent diverse collections of protein-ligand complexes, including bound/unbound states, drug-target complexes and other ligand site predictor benchmark sets.

\subsection{Test datasets}

The majority of ligand binding site predictors published since 2018 have been using two datasets that were first presented by Krivák \textit{et al.} \cite{KRIVAK_2018_P2RANK}: COACH420 and HOLO4K, or subsets of them. COACH420 is comprised by a set of 420 single-chain structures binding a mix of drug-like molecules and naturally occurring ligands which is disjunct with the CHEN11 and JOINED datasets. COACH420 is a modified version of the original COACH test set \cite{ROY_2012_COFACTOR, YANG_2013_COFACTOR}. HOLO4K is a larger set, $N \approx$ 4,000, based on the list by Schmidtke \textit{et al.} \cite{SCHMIDTKE_2010_FPOCKET}, which includes a mix of single- and multi-chain complexes, also disjunct with P2Rank training (CHEN11) and validation (JOINED) datasets. VN-EGNN, DeepPocket and GrASP use the Mlig and Mlig+ subsets of the COACH and HOLO4K datasets, which comprise strictly biologically relevant ligands as defined by the binding MOAD. IF-SitePred is tested on the HOLO4K-AlphaFold2 Paired (HAP) and HAP-small sets. HAP is a subset of the HOLO4K dataset which presents high quality models in the AlphaFold database \cite{VARADI_2022_ALPHAFOLDDB}. HAP-small is a smaller subset of HAP that only contains proteins with sequence identity lower than 25\% to proteins in the P2Rank training set. VN-EGNN uses the refined version of PDBbind (v2020), referred here as PDBbind\textsubscript{REF}, \cite{WANG_2004_PDBBIND, WANG_2005_PDBBIND, CHENG_2009_PDBBIND, LI_2014_PDBBIND, LIU_2015_PDBBIND, LIU_2017_PDBBIND} as a third test set. Like binding MOAD, the PDBbind database provides a comprehensive collection of experimentally measured binding affinity data for macromolecular complexes. Specifically, the refined set includes those protein-ligand complexes for which binding data was obtained with the literature and met certain experimental quality thresholds. Lastly, SC6K is a dataset presented by Aggarwal \textit{et al.} \cite{AGGARWAL_2022_DEEPPOCKET} containing 6,000 protein-ligand pairs from PDB entries submitted from 01/01/2018 – 28/02/2020. 

\subsection{Protein chain alignment}

For each protein chain, atomic coordinates were translated to be centred at the origin, $O = (0, 0, 0)$, and rotated using a rotation matrix, $R$. The two principal components of the coordinate space $pc_{1}$ and $pc_{2}$ were obtained using principal component analysis (PCA) \cite{HOTELLING_1933_PCA}. A third component, $pc_{\perp}$, was obtained with the cross-product of the other two, to ensure orthogonality. A rotation matrix $P$ is constructed from these vectors (Equation \ref{eq:pca_components}). By placing the main component $pc_{1}$ on the second row of $P$, we ensure the $Y$ axis will be the major axis, representing the height of the protein chain. The second largest axis will be the $X$ axis, representing the width of the protein, and lastly the depth will be represented by the smaller magnitude of the $Z$ axis. The final rotation matrix $R$ is obtained by multiplying $P$ by the negative identity matrix $NI$ (Equations \ref{eq:NI_matrix} and \ref{eq:R_matrix}). This was done to maintain the left-handedness of the protein chains whilst ensuring a consistent alignment on the major axes.

\begin{equation}
pc_{\perp} = pc_{1} \times pc_{2} \quad \rightarrow \quad P = \begin{bmatrix}
pc_{2} \\
pc_{1} \\
pc_{\perp}
\end{bmatrix}
\label{eq:pca_components}
\end{equation}
\myequations{Principal Components Vector}

\begin{equation}
NI = -1 \cdot I_3 = -1 \cdot \begin{bmatrix}
1 & 0 & 0 \\
0 & 1 & 0 \\
0 & 0 & 1 
\end{bmatrix} = \begin{bmatrix}
-1 & 0 & 0 \\
0 & -1 & 0 \\
0 & 0 & -1 
\end{bmatrix}
\label{eq:NI_matrix}
\end{equation}
\myequations{Negative Identity Matrix}

\begin{equation}
R = P \cdot NI
\label{eq:R_matrix}
\end{equation}
\myequations{Rotation Matrix}

\subsection{Protein chain characterisation}

For a protein chain with $N$ amino acid residues, the centre of mass, $CM$, was calculated by averaging the coordinates, $r_{i}$, of all atoms (Equation \ref{eq:centre_of_mass}), and from it, the radius of gyration, $R_{g}$, was derived (Equation \ref{eq:radius_of_gyration}) \cite{FIXMAN_1962_ROG}. As the protein chains are already aligned on the axis and centred on $O = (0, 0, 0)$ the dimensions of the protein chain can be obtained as the magnitude of the PCA components or \textit{eigenvectors}, i.e., the \textit{eigenvalues}. The dimensions represent width, height, and depth for the $X$, $Y$ and $Z$ axes, respectively.

\begin{equation}
CM = \frac{1}{n} \sum_{i=1}^{n} r_i \rightarrow CM = O = (0,0,0)
\label{eq:centre_of_mass}
\end{equation}
\myequations{Centre of Mass}

\begin{equation}
R_g = \sqrt{\frac{1}{n} \sum_{i=1}^{n} (r_i - CM)^2} = \sqrt{\frac{1}{n} \sum_{i=1}^{n} (r_i - O)^2} \rightarrow R_g = \sqrt{\frac{1}{n} \sum_{i=1}^{n} r_i^2}
\label{eq:radius_of_gyration}
\end{equation}
\myequations{Radius of Gyration}

\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.98\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG11_PROTEIN_SHAPE_APPROACH_OPT.pdf}
    \caption[Protein chain shape and size classification approach]{\textbf{Protein chain shape and size classification approach.} \textbf{(A)} The volume of the sphere enclosing the protein chain as well as the protein chain volumes are calculated, and their ratio obtained ($VR$). Globular proteins present more spherical shapes and therefore occupy a higher portion of the sphere volume, resulting in higher volume ratios. Non-globular, elongated or fibrous proteins on the other hand do not and present lower volume ratios. After extensive visual examination, a threshold was established at $VR$ = 0.08, and so proteins classified in these two groups. Proteins were classified as ``tiny'' if their chain was $\leq$ 100 amino acids; \textbf{(B)} Eight examples of each protein chain group to illustrate the outcome of the approach.}
    \label{fig:protein_class_approach}
\end{figure}

Protein chain volumes were calculated using ProteinVolume \cite{CHEN_2015_PROTEINVOLUME}. A sphere enclosing the protein and centred on the protein centre of mass was obtained. The radius of this sphere is the maximum Euclidean distance between the protein atoms and the CM (Equation \ref{eq:radius_protein}). The volume of the sphere is calculated using Equation \ref{eq:volume_sphere}. Proteins were classified into four different groups based on their shape and size. Protein chains with $\leq$ 100 amino acids were classified as ``tiny''. Regarding the shape, protein chains were classified into ``elongated'' if their protein to sphere volume ratio $\leq$ 0.08 ($VR$) (\autoref{eq:volume_ratio}), i.e., the protein volume contains no more than 8\% of the sphere volume. This threshold was derived empirically by the visual inspection of all 3,448 protein chains on the LIGYSIS set. Otherwise, proteins were considered globular (\autoref{fig:protein_class_approach}). In this manner, protein chains were classified into \textit{globular} ($N$ = 2,104; 61\%), \textit{elongated} ($N$ = 670; 19\%), \textit{elongated tiny} ($N$= 341; 10\%) and \textit{globular tiny} ($N$ = 333; 10\%).

\begin{equation}
R = \max \| r_i - CM \|
\label{eq:radius_protein}
\end{equation}
\myequations{Protein Radius}

\begin{equation}
Volume_{Sphere} = \frac{4}{3} \pi R^3
\label{eq:volume_sphere}
\end{equation}
\myequations{Sphere Volume}

\begin{equation}
VR = \frac{Volume_{Protein}}{Volume_{Sphere}}
\label{eq:volume_ratio}
\end{equation}
\myequations{Volume Ratio}

\FloatBarrier

\subsection{Ligand binding site prediction}

For each segment in the LIGYSIS dataset, the representative chain as defined in the PDBe-KB was selected. Structures were cleaned using the \textit{clean\_pdb.py} script \cite{JUBB_2019_PDBTOOLS}. Eleven different ligand binding site prediction tools were employed to predict on the 3,448 representative chains: VN-EGNN \cite{SESTAK_2024_VNEGNN}, IF-SitePred \cite{CARBERY_2024_IFSP}, GrASP \cite{SMITH_2024_GrASP}, PUResNet \cite{KANDEL_2021_PURESNET, KANDEL_2024_PURESNET}, DeepPocket \cite{AGGARWAL_2022_DEEPPOCKET}, P2Rank \cite{KRIVAK_2015_P2RANK, KRIVAK_2018_P2RANK}, P2Rank\textsubscript{CONS} \cite{JENDELE_2019_PRANKWEB, JAKUBEC_2022_PRANKWEB}, fpocket \cite{GUILLOUX_2009_FPOCKET, SCHMIDTKE_2010_FPOCKET2}, PocketFinder\textsuperscript{+} \cite{AN_2005_POCKETFINDER}, Ligsite\textsuperscript{+} \cite{HENDLICH_1997_LIGSITE}, and Surfnet\textsuperscript{+} \cite{LASKOWSKI_1995_SURFNET}. Conservation scores for P2Rank were obtained from PrankWeb \url{https://prankweb.cz/}. Re-implementations of Capra \textit{et al.} \cite{CAPRA_2009_CONCAVITY} were used for PocketFinder, Ligsite and Surfnet, indicated by the ``+'' superscript. VN-EGNN, IF-SitePred, PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+} and Surfnet\textsuperscript{+} do not provide a list of residues for each pocket, but a list of centroids and their scores for the first two, and a list of grid points for each predicted pocket for the last three. For VN-EGNN, residues within 6\AA{} of the virtual nodes were considered pocket residues. For 429 predicted pockets ($\approx$3\%) no residues were found within this threshold. For IF-SitePred, residues within 6\AA{} of the clustered cloud points that resulted on a predicted pocket centroid were considered as pocket residues. Pocket residues were obtained in a similar manner for PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+} and Surfnet\textsuperscript{+}, by taking those residues within 6\AA{} of the pocket grid points. When running DeepPocket, the $-r$ threshold was removed and so all fpocket candidates were passed to the CNN-based segmentation module for pocket shape estimation. fpocket predictions re-scored by DeepPocket will be referred as DeepPocket\textsubscript{RESC}, whereas pockets extracted by the segmentation module of DeepPocket will be referred as DeepPocket\textsubscript{SEG}.

Seven of the considered methods provide residue \textit{ligandability} scores. P2Rank and P2Rank\textsubscript{CONS} report calibrated probabilities of residues being ligand-binding. Similarly, GrASP predicts the likelihood for any given heavy atom to be part of a binding site. A residue-level score was obtained for GrASP predictions by taking the maximum score of the residue atoms. For IF-SitePred, a residue ligandability score $LS$ can be obtained by averaging the 40 independently predicted probabilities of a residue being ligand-binding (Equation \ref{eq:IFSP_score}). Though calculated in a different way, these three scores range 0-1 and represent the likelihood of a residue binding a ligand and can therefore be compared. PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+}, and Surfnet\textsuperscript{+} also provide residue scores which maximum value can be $>$ 1.

\begin{equation}
LS = \frac{1}{40} \sum_{i=1}^{40} p_i
\label{eq:IFSP_score}
\end{equation}
\myequations{IF-SitePred ligandability score}

The other three methods, i.e., VN-EGNN, PUResNet, DeepPocket, and fpocket do not report residue-level scores. However, binary labels represent whether a residue is part of a pocket (1) or not (0), in the same manner as all other methods.

Throughout this work the terms ``site'' and ``pocket'' are be used indistinctly. Across all figures, tables and legends, methods are sorted in chronological order.

\subsection{Binding site characterisation}

Radius of gyration, $R_{g}$, was calculated for pockets as it was done for whole protein chains (Equation \ref{eq:radius_of_gyration}). Distance between pockets was calculated as the Euclidean distance between their centroids and overlap between pocket residues with the Jaccard Index ($JI$), or intersection over union (IOU) (Equation \ref{eq:jaccard_index}) \cite{JACCARD_1901_INDEX, JACCARD_1912_INDEX}. POVME 2.0 was employed for pocket volume calculation \cite{DURRANT_2011_POVME, DURRANT_2014_POVME2, WAGNER_2017_POVME3}. A single inclusion region was used for each pocket. This region is defined by the smallest rectangular prism containing all pocket atoms. The prism is centred on the pocket centroid and its dimensions are determined by the distance between the two farthest atomic coordinates on each axis. No exclusion regions were used. Points outside the convex hull were deleted. A contiguous-points region was defined as a 5\AA{}-radius sphere on the pocket centroid (\autoref{fig:protein_volume_approach}).

\begin{equation}
JI(A, B) = \frac{|A \cap B|}{|A \cup B|}
\label{eq:jaccard_index}
\end{equation}
\myequations{Jaccard Index}

\begin{figure}[h]
    \centering
    \includegraphics[width=0.98\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG12_POCKET_VOLUME_APPROACH_OPT.pdf}
    \caption[Pocket volume calculation algorithm]{\textbf{Pocket volume calculation algorithm.} \textbf{(A)} PUResNet predicted pocket for PDB: 4PX2 (Jordan, S.R., Chmait, S., 2015). Pocket residues are coloured in blue and have their side chains displayed; \textbf{(B)} An inclusion region is determined using the coordinates of the pocket residue atoms; \textbf{(C)} POVME 2.0 calculates the shape of the pocket defined by the residues and contained within the inclusion region; \textbf{(D)} The pocket shape is defined by a series of unit volume (1\AA{}\textsuperscript{3}) spheres. The volume of the pocket is calculated as the addition of the sphere volumes or the number of spheres within the pocket. Structure visualisation with PyMOL v2.5.2 \cite{SCHRODINGER_2015_PYMOL}.}
    \label{fig:protein_volume_approach}
\end{figure}

\subsubsection{Determination of DCC threshold}

Most methods employ distance to any atom of the ligand (DCA) and a threshold of 4\AA{} to consider a prediction as correct. Because of the way the LIGYSIS dataset has been curated, it is easier to use DCC, as our binding sites result of the clustering of multiple ligands, and not just a single ligand binding a protein. Despite DCC and DCA being different metrics, the same threshold of D = 4\AA{} is used for both when benchmarking methods \cite{AGGARWAL_2022_DEEPPOCKET, SESTAK_2024_VNEGNN, KANDEL_2021_PURESNET}. Supplementary Figure 10A shows the relation between DCC, and pocket residue overlap for the best pocket predictions, i.e., minimum inter-centroid Euclidean distance, for each observed pocket for each method. Across all methods, there are more than 15,000 predicted pockets with a DCC $>$ 4\AA{} and a residue overlap $\geq$ 50\%. Setting the DCC threshold at 4\AA{} would result in the wrong labelling of these predictions as ``false positives''. For this reason, we endeavoured into empirically establishing a more meaningful DCC threshold through the thorough visual inspection of predicted-observed pocket pairs. Supplementary Figure 10B suggests this threshold might be somewhere in between 10-15\AA{}, where the proportion of pockets with $I_{rel} \geq$ 0.5 decreases until reaching 0.

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/SUPP/PDF/SUPP_FIG10_IREL_vs_DCC_OPT.pdf}
    \caption[$I_{rel}$ \textit{vs} DCC]{\textbf{$I_{rel}$ \textit{vs} DCC.} \textbf{(A)} Hexagonal binned plot of $I_{rel}$ (Y) \textit{vs} DCC (X). Data points are grouped into hexagonal bins, and these are coloured by the number of data points within each bin using the \textit{viridis} colour palette. Colour bar axis is in log scale. Black dashed lines indicate the literature consensus DCC = 4\AA{} threshold and an arbitrary $I_{rel}$ threshold of 0.5, i.e., coverage of half of the observed residues by the predicted pocket.  Red lines delimit the likely location of a potentially more informative DCC thresholds; \textbf{(B)} Cumulative proportion of predicted pockets with $I_{rel} \geq$ 0.5 for each DCC 1\AA{} interval. The commonly used threshold of DCC = 4\AA{} would label $>$15,000 predictions with $I_{rel} \geq$ 0.5 as false. Error bars indicate 95\% CI of the proportion.}
    \label{fig:irel_vs_dcc}
\end{figure}

We set a hard threshold at D = 20\AA{} and decided that based purely on distance, pockets with DCC $>$ 20 would not be considered as correct predictions. For each DCC interval of 1\AA{}, the pocket with the highest and lowest $I_{rel}$ were inspected (\autoref{fig:DCC_determination}). This initial visual inspection supported the hypothesis that a more meaningful DCC threshold is between 10-14\AA{}. For the next step, only predicted-observed pocket pairs with minimal overlap ($I_{rel} <$ 0.25) were considered. Starting at DCC = 10\AA{}, and using unit intervals, the 100 farthest pockets were inspected for each interval, and the proportion of correct predictions was calculated as the number of pockets labelled as ``correct'' upon visual inspection divided by 100, i.e., \%. For D = 10\AA{}, 94\% of pockets were correct (\autoref{fig:DCC_10_examples}), 86\% for D = 11\AA{} (\autoref{fig:DCC_11_examples}), 85\% for D = 12\AA{} (\autoref{fig:DCC_12_examples}) and 66\% for D = 13\AA{}. Due to the considerable drop of correct pockets at D = 13\AA{}, the final distance threshold was set at D = 12\AA{}. Accordingly, predictions were considered as true positives if DCC $\leq$ 12\AA{} (\autoref{eq:DCC_threshold}).

\begin{equation}
\textit{True Positive} \iff (\text{DCC} \leq 12\text{\AA{}})
\label{eq:DCC_threshold}
\end{equation}
\myequations{Empirically determined DCC threshold}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/SUPP/PDF/SUPP_FIG11_DETERMINING_DCC_THRESHOLD_2_OPT.pdf}
    
    \caption[Determination of DCC threshold]{\textbf{Determination of DCC threshold.} Highest and lowest-residue overlap predictions for each 2\AA{} DCC unit interval. Observed LIGYSIS sites are coloured in green, predicted pockets in other colours. $D$ represents DCC and $I_{rel}$ the relative intersection between predicted and observed pocket residues, i.e., proportion of observed site residues covered by predicted pocket residues. ``YES'' or ``NO'' labels indicate whether we considered a given prediction as correct upon visual inspection. ``?'' at DCC = 12\AA{} illustrates    }
    \label{fig:DCC_determination}
\end{figure}

\begin{figure}[ht]
\ContinuedFloat
\caption*{(continued) the inflection point between 10-12\AA{}, where it is no longer clear whether predicted pockets within this DCC interval and $I_{rel} \approx$ 0 agree with the observed pockets. To facilitate the visualisation of the observed pocket, this one is coloured after the predicted one. Otherwise, for cases where $I_{rel}$ = 1 only the predicted pocket would be shown. Despite 1\AA{} intervals were inspected, only representatives of 2\AA{} intervals are shown here for simplicity.}
\end{figure}
\FloatBarrier

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/SUPP/PDF/SUPP_FIG12_EXAMPLES_DCC_10_OPT.pdf}
    \caption[Predicted-observed pocket pairs at DCC = 10\AA{} and $I_{rel} <$ 0.25]{\textbf{Predicted-observed pocket pairs at DCC = 10\AA{} and $I_{rel} <$ 0.25} Out of the 100 examples visually inspected, 94 were considered as correct predictions on the bases that the predicted and observed pockets are adjacent, i.e., their surface area is in contact, and it is therefore easy to imagine a ligand that would bind to this region. LIGYSIS observed sites are coloured in green, and predicted pockets in other colours.}
    \label{fig:DCC_10_examples}
\end{figure}
\FloatBarrier

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/SUPP/PDF/SUPP_FIG13_EXAMPLES_DCC_11_OPT.pdf}
    \caption[Predicted-observed pocket pairs at DCC = 11\AA{} and $I_{rel} <$ 0.25]{\textbf{Predicted-observed pocket pairs at DCC = 11\AA{} and $I_{rel} <$ 0.25} Out of the 100 examples visually inspected, 86 were considered as correct predictions. LIGYSIS observed sites are coloured in green, and predicted pockets in other colours.}
    \label{fig:DCC_11_examples}
\end{figure}
\FloatBarrier

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/SUPP/PDF/SUPP_FIG14_EXAMPLES_DCC_12_OPT.pdf}
    \caption[Predicted-observed pocket pairs at DCC = 12\AA{} and $I_{rel} <$ 0.25]{\textbf{Predicted-observed pocket pairs at DCC = 12\AA{} and $I_{rel} <$ 0.25} Out of the 100 examples visually inspected, 85 were considered as correct predictions. LIGYSIS observed sites are coloured in green, and predicted pockets in other colours.}
    \label{fig:DCC_12_examples}
\end{figure}
\FloatBarrier

\subsection{Prediction evaluation}

After predicting on the 3,448 chains of the LIGYSIS dataset, only chains where all residues across all predicted sites presented UniProt residue mapping were kept. This resulted in a final set of 2,775 protein chains which was used for the performance assessment of the methods.

The performance of ligand binding site prediction methods can be evaluated at two different levels: \textit{residue} level, and \textit{pocket} level. Prediction at the residue level involves the discrimination of those residues that are likely to interact with a ligand, whereas the aim of pocket-level prediction is to define distinct regions on a protein, i.e., pockets where a ligand is likely to bind. This region can either be defined by a (pocket) centroid, a group of cloud/grid points, a set of pocket residues, or a combination of these. Some methods are residue-centric, and first predict at the residue-level, use a threshold to select high-probability ligand-binding residues, and then cluster them into pockets. \textit{Residue}-centric methods include IF-SitePred, or GrASP. Other methods (\textit{pocket}-centric) directly predict the location or shape of the pocket, without the need of predicting at the residue level first. Some of these methods can use their pocket-level prediction to report residue ligandability scores, e.g., P2Rank\textsubscript{CONS}, P2Rank, PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+} or Surfnet\textsuperscript{+}, and others, such as VN-EGNN, PUResNet, DeepPocket, or fpocket do not report residue ligandability scores.

\subsubsection{Residue-level predictions}

GrASP, P2RANK\textsubscript{CONS}, P2Rank, PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+} and Surfnet\textsuperscript{+} all offer residue ligandability scores. Additionally, a ligandability score was derived for IF-SitePred using Equation \ref{eq:IFSP_score}. Given a ligandability threshold $t_{LS}$, a residue with a ligandability score $LS_{i}$ is classified as ``positive'' if $LS_{i} > t_{LS}$. Conversely, the residue is classed as ``negative'' if $LS_{i} \leq t_{LS}$. Further stratification results from comparing the predictions to the LIGYSIS reference dataset.

\begin{itemize}
\item True Positive (TP): residue classified as positive that binds a ligand according to the reference.
\item False Positive (FP): residue classified as positive that does not bind a ligand in the reference.
\item True Negative (TN): residue classified as negative that does not bind a ligand.
\item False Negative (FN): residue classified as negative but is known to bind a ligand according to the data.
\end{itemize}

With these four classes, true positive rate (TPR) (Equation \ref{eq:TPR}, false positive rate (FPR) (Equation \ref{eq:FPR}), precision (Equation \ref{eq:precision}) and recall (Equation \ref{eq:recall}) can be calculated and receiver operating characteristic (ROC), precision-recall (PR) curves plotted. Area under the curve (AUC) for ROC and average precision (AP) for PR are reported. Baselines for these are 0.5 and the proportion of true positive labels, respectively.

\begin{equation}
TPR = \frac{TP}{TP + FN}
\label{eq:TPR}
\end{equation}
\myequations{True Positive Rate}

\begin{equation}
FPR = \frac{FP}{FP + TN}
\label{eq:FPR}
\end{equation}
\myequations{False Positive Rate}

\begin{equation}
Precision = \frac{TP}{TP + FP}
\label{eq:precision}
\end{equation}
\myequations{Precision}

\begin{equation}
\text{Recall} = \frac{\text{TP}}{\text{TP} + \text{FN}}
\label{eq:recall}
\end{equation}
\myequations{Recall}

Pocket binary labels (0: no pocket residue; 1: pocket residue) can determine TP, FP, TN and FN for each residue in a protein chain $P$. An F1 score is computed across all residues of $P$, which combines precision and recall into a unified metric, capturing the accuracy and completeness of predictions at the residue level (Equation \ref{eq:F1_score}). The Matthews Correlation Coefficient (MCC) \cite{MATTHEWS_1975_MCC} (Equation \ref{eq:MCC}) was also calculated. The median F1 score and MCC across the dataset proteins is reported for each method.

\begin{equation}
F1 = \frac{2 \times \text{Precision} \times \text{Recall}}{\text{Precision} + \text{Recall}}
\label{eq:F1_score}
\end{equation}
\myequations{F1 Score}

\begin{equation}
MCC = \frac{TP \times TN - FP \times FN}{\sqrt{(TP + FP)(TP + FN)(TN + FP)(TN + FN)}}
\label{eq:MCC}
\end{equation}
\myequations{Matthews Correlation Coefficient}

\subsubsection{Pocket-level predictions}

There are no \textit{negatives} predicted at the pocket level of ligand binding site prediction, only \textit{positives}. A positive is a predicted pocket, which will be true (TP) or false (FP) depending on whether it is observed in the reference data. False negatives are those pockets observed in the reference data that are not predicted. They are the pockets the method fails to predict, and therefore, are not scored. A true negative would be a ``non-pocket'' that is not predicted. This can't be quantified easily and even if it was, it would not be scored by the method, as it is not predicted. For this reason, in this context, neither TPR, nor FPR can be calculated. Consequently, ROC/AUC can't be utilised to assess ligand binding site prediction at the pocket level. False negatives are known, but not scored, and therefore PR/AUC is not an option either. What can be calculated is the recall given a certain criterion. In this case, because of the nature of the LIGYSIS dataset, where defined sites result from the clustering of multiple ligands, the distance between the predicted pocket centroid to the observed binding site (DCC) was chosen as a criterion.

For each observed binding site in our reference dataset, the ``best'' prediction for each method is chosen. This is defined as the prediction with the minimum Euclidean distance to the observed pocket centroid or DCC. Once the observed-predicted pairs were obtained, only those with DCC $\leq$ 12 \AA{} were considered as correct predictions. A threshold of 12\AA{} was chosen as 4\AA{} is too strict a threshold when using DCC (See Supplementary Note 6 and Figures 10-14). A threshold of 4\AA{} works well for the distance to closest ligand atom (DCA) but does not for DCC. The top-$N$ and $N$+2 ranking predictions were considered to calculate success rate, or recall (Equation \ref{eq:success_rate}), and maximum recall was calculated by considering all predictions, regardless of their score or rank. $N$ represents the number of observed sites for a given protein.

\begin{equation}
\text{Success rate (recall)} = \frac{\text{observed sites with predicted site DCC} \leq 12 \text{Å}}{\text{observed sites}}
\label{eq:success_rate}
\end{equation}
\myequations{Success Rate}

Additionally, instead of conventional ROC, ROC100 \cite{WEBBER_2003_ROC100, SCOTT_2007_ROC100} can be used to measure the predictive performance of the methods. To do this, for each method, all predictions across dataset proteins were ranked based on the pocket score and cumulative true positives were plotted against cumulative false positives until 100 false positives are reached. In a similar way, a precision curve can be calculated by taking the top-$N$, in this case $N$ = 1,000, predictions, which measures how precision changes as more predictions with lower scores are considered. This is indicative of how informative pocket scores are.

\begin{figure}[htbp!]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG13_VOLUME_OVERLAP_APPROACH_OPT.pdf}
    \caption[Relative Volume Overlap ($RVO$) calculation]{\textbf{Relative Volume Overlap ($RVO$) calculation.} \textbf{(A)} Example of two very accurate predictions by PUResNet and P2Rank on PDB: 4PX2 (Jordan, S.R., Chmait, S., 2015). Pocket volumes are calculated with POVME 2.0 and represented by coloured surfaces. These volumes result from the addition of unit volume spheres on a grid. To obtain the $RVO$, the intersection of these spheres between predicted and observed site is divided by the number of observed pocket spheres. Both predictions cover the entirety of the observed pocket volume; \textbf{(B)} GrASP and VN-EGNN predictions of a site on PDB: 2ZOX \cite{NOGUCHI_2008_STRUCTURE}. Th volumes of these predicted sites overlap less with the observed site: $RVO$ = 0.67 for GrASP and $RVO$ = 0.11 for VN-EGNN.}
    \label{fig:protein_RVO_approach}
\end{figure}

\FloatBarrier

To measure the similarity in shape and residue membership between the predicted and observed pockets, relative residue overlap ($RRO$) and relative volume overlap ($RVO$) were employed. For an observed-predicted pocket pair, RRO represents the proportion of observed ligand-binding residues ($R_{o}$) that are covered by the predicted pocket residues ($R_{p}$) (Equation \ref{eq:RRO}). The POVME output was utilised for the calculation of $RVO$ (Figure \ref{fig:protein_RVO_approach}). POVME defines the volume of a pocket as a series of equidistantly spaced spheres of unit volume. As predictions by the different methods were on the same coordinate reference, these pocket volume spheres were already aligned, and the volume overlap was calculated simply as the proportion of spheres in the observed pocket ($V_{o}$) that overlap with the predicted pocket spheres ($V_{p}$) (Equation \ref{eq:RVO}).

\begin{equation}
RRO = \frac{|R_p \cap R_o|}{R_o}
\label{eq:RRO}
\end{equation}
\myequations{Relative Residue Overlap}

\begin{equation}
RVO = \frac{|V_p \cap V_o|}{V_o}
\label{eq:RVO}
\end{equation}
\myequations{Relative Volume Overlap}

\subsection{Statistics and reproducibility}

VN-EGNN was installed and run locally from \url{https://github.com/ml-jku/vnegnn}. Likewise, for IF-SitePred: \url{https://github.com/annacarbery/binding-sites}. GrASP was obtained from \url{https://github.com/tiwarylab/GrASP} and predictions generated using their Google Colab Notebook. PUResNet predictions were obtained through the PUResNet v2.0 web server: \url{https://nsclbio.jbnu.ac.kr/tools/jmol}. DeepPocket was installed and executed locally: \url{https://github.com/devalab/DeepPocket}. P2Rank v2.4.2 was used to run all predictions as well as PRANK re-scoring: \url{https://github.com/rdk/p2rank}. fpocket v4.0 was installed via Conda: \url{https://anaconda.org/conda-forge/fpocket}. For PocketFinder, Ligsite and Surfnet, the ConCavity v0.1 ``+'' re-implementations were employed: \url{https://compbio.cs.princeton.edu/concavity/}.

Other recent methods including  EquiPocket \cite{YAN_2024_EQUIPOCKET}, RefinePocket \cite{LIU_2023_REFINEPOCKET}, GLPocket \cite{LI_2023_GLPOCKET}, NodeCoder \cite{ABDOLLAHI_2023_NODECODER}, SiteRadar \cite{EVTEEV_2023_SITERADAR}, RecurPocket \cite{LI_2022_RECURPOCKET}, PointSite \cite{YAN_2022_POINTSITE}, DeepSurf \cite{MYOLNAS_2021_DEEPSURF}, GRaSP \cite{SANTANA_2020_GRaSP}, BiteNet \cite{KOZLOVSKII_2020_BITENET}, Kalasanty \cite{STEPNIEWSKA_2020_KALASANTY} or DeepSite \cite{JIMENEZ_2017_DEEPSITE} were not included in this analysis due to technical reasons. Open-source methods with publicly accessible code, clear installation instructions, well defined dependencies, and accessible command line interfaces were prioritised in this work.

ChimeraX v1.7.1 \cite{PETTERSEN_2021_CHIMERAX} was used for structural visualisation in all figures unless otherwise stated, in which case PyMOL v2.5.2 was employed \cite{SCHRODINGER_2015_PYMOL}.

\subsection{Data availability}

The main results tables and files necessary to replicate the analysis described in this paper can be found here: \url{https://doi.org/10.5281/zenodo.13121414} \cite{UTGES_2024_LBSCOMP_ZENODO}.

\subsection{Code availability}

Software developed to carry out this analysis is found in this GitHub repository: \url{https://github.com/bartongroup/LBS-comparison} \cite{UTGES_2024_LBSCOMP_REPO}.

\section{Results}

\subsection{The LIGYSIS dataset}

The human subset of the LIGYSIS dataset, LIGYSIS\textsubscript{HUMAN}. comprises protein-ligand complexes for 3,448 human proteins. For each protein, biologically relevant protein-ligand interactions, in accordance with BioLiP \cite{YANG_2013_BIOLIP}, are considered across the PISA-defined \cite{KRISSINEL_2007_PISA} biological assemblies of the multiple entries deposited in the PDBe \cite{ARMSTRONG_2020_PDBE}. Ligands are clustered using their protein interaction fingerprint to identify ligand binding sites as described in Utgés \textit{et al.} \cite{UTGES_2024_FRAGSYS, UTGES_2024_FRAGSYS_ZENODO}. The full LIGYSIS dataset includes $\approx$30,000 proteins with known ligand-bound complexes. Here, we focus on the human subset of LIGYSIS as a manageable set to run all prediction methods on and refer to this as LIGYSIS for brevity.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG1_REDUNDANT_PLIS_SPLIT_1_OPT.pdf}
    \caption[Redundancy in protein-ligand interfaces (I)]{\textbf{Redundancy in protein-ligand interfaces (I).} For PDB: 1QJY, the asymmetric unit comprises three copies of a homo-pentamer, whereas the biologically functional assembly is a single pentamer. A BMSC-0010 ligand molecule binds to each copy, except for one, of each of the three pentamers. This results in the same protein-ligand interface repeated 14 times, i.e., 14x redundancy. Dashed rectangles indicate the asymmetric and biological units.}
    \label{fig:redundant_plis_1}
\end{figure}

The LIGYSIS dataset differs from previous train and test sets for ligand binding sites by considering biological units, aggregating multiple structures of the same protein, and removing redundant protein-ligand interfaces. Biological units, or assemblies, as opposed to asymmetric units are considered in LIGYSIS in a consistent manner. This is important for any analysis that delves into molecular interactions at the residue or atomistic level. The crystal asymmetric unit often does not correspond to the biological unit and can result in artificial crystal contacts or redundant protein-ligand interfaces. An example of this illustrated in Figure \ref{fig:redundant_plis_1} is PDB: 1JQY \cite{PICKENS_2002_ANCHOR}, present in the HOLO4K dataset, where the asymmetric unit is formed by three copies of a homo-pentamer, whereas the biological unit comprises a single pentamer. In this structure, 14 molecules of BMSC 0010 (A32) interact with 14 copies of \textit{Escherichia coli} heat-labile enterotoxin B chain (P32890). This protein-ligand interface is the same repeated 14 times.

\FloatBarrier

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG1_REDUNDANT_PLIS_SPLIT_2_OPT.pdf}
    \caption[Redundancy in protein-ligand interfaces (II)]{\textbf{Redundancy in protein-ligand interfaces (II).} For PDB: 1PPR both the asymmetric and biological units are a homo-trimer. Different molecules of the same ligands are binding to the same interfaces across the three copies of the trimer, i.e., 3x redundancy. Dashed rectangle indicate the asymmetric/biological unit.}
    \label{fig:redundant_plis_2}
\end{figure}

Protein-ligand interface redundancy can also be an issue when the asymmetric unit equals the biological assembly (Figure \ref{fig:redundant_plis_2}). In PDB: 1PPR \cite{HOFMANN_1996_CAROTENOID}, also in HOLO4K, molecules of chlorophyll A (CLA), peridinin (PID) and digalactosyl diacyl glycerol (DGD) bind to the three copies of a peridinin-chlorophyll a-binding protein 1, chloroplastic, PCP, (P80484) trimer, resulting in a redundancy of 3x. To account for this, LIGYSIS considers unique non-redundant protein-ligand interfaces by retrieving the UniProt residue numbers of the residues the ligands interact with, so 1/14 interfaces would be retrieved for PDB: 1JQY and 12/36 for PDB: 1PPR. Finally, unique ligand interactions are aggregated across different structures for the same protein and ligand sites defined.

\FloatBarrier

%\begin{figure}[htbp!]
%    \centering
%    \includegraphics[width=0.9\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG1_REDUNDANT_PLIS_OPT.pdf}
%    \caption[Redundancy in protein-ligand interfaces]{\textbf{Redundancy in protein-ligand interfaces.} %Two examples of how the type of macromolecular assembly and difference between the asymmetric and %biological unit of a protein-ligand complex results in redundant protein-ligand interfaces. %\textbf{(A)} For PDB: 1QJY, the asymmetric unit comprises three copies of a homo-pentamer, whereas the %biologically functional assembly is a single pentamer. A BMSC-0010 ligand molecule binds to each copy, %except for one, of each of the three pentamers. This results in the same protein-ligand interface %repeated 14 times, i.e., 14x redundancy; \textbf{(B)} For PDB: 1PPR both the asymmetric and biological %units are a homo-trimer. Different molecules of the same ligands are binding to the same interfaces %across the three copies of the trimer, i.e., 3x redundancy. Dashed rectangles indicate the asymmetric/%biological units.}
%    \label{fig:redundant_plis}
%\end{figure}

Figure \ref{fig:PDBbind_VS_LIGYSIS} shows the comparison between PDB: 4GQQ \cite{WILLIAMS_2012_AMYLASE}, present in the PDBbind dataset, and the LIGYSIS entry for human pancreatic alpha-amylase (P04746), which representative structure is also 4GQQ. The entry in PDBbind represents a single protein-ligand complex, whereas LIGYSIS makes use of 51 structures, 195 ligands to define 13 different ligand binding sites. LIGYSIS aggregates all unique biologically relevant protein-ligand interactions for a protein in a non-redundant manner, thus representing the most complete and integrative protein-ligand binding dataset up to date. For this reason, we propose LIGYSIS as a new benchmark dataset for the prediction of ligand binding sites and use it in this work to evaluate a set of 11 ligand binding site prediction and cavity identification tools.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG2_LIGYSIS_vs_PDBbind_OPT.pdf}
    \caption[Comparison of PDBbind and LIGYSIS]{\textbf{Comparison of PDBbind and LIGYSIS.} PDBbind is comprised by complexes between a protein and the most biologically relevant ligand in a structure. For PDB: 4GQQ, this is ethyl caffeate (0XR). LIGYSIS considers all unique biologically relevant protein-ligand interactions across all the structures for a given protein. For human pancreatic alpha-amylase (P04746), which representative structure is 4GQQ, 13 ligand binding sites are defined from 195 ligands across 51 structures. LIGYSIS provides a better representation of the ligand-binding capabilities of a protein than a single protein-ligand complex and constitutes therefore a better benchmark for ligand binding site prediction tools.}
    \label{fig:PDBbind_VS_LIGYSIS}
\end{figure}

\subsection{Comparison of datasets}

To assess the scope and limitations of the predictive methods surveyed in this work, their training and test sets were compared with LIGYSIS by number of sites per protein, number of interacting protein chains per ligand site, ligand size, ligand site size, and ligand composition. sc-PDBFULL represents the full sc-PDB dataset used for training by DeepPocket, bMOAD\textsubscript{SUB} the subset of binding MOAD used for training by IF-SitePred and PDBbind\textsubscript{REF} the reference subset of PDBbind which VN-EGNN uses for testing. Table 2 summarises the size of the datasets, which methods employ them and their overlap with the LIGYSIS set. LIGYSIS differs from all other datasets since biologically relevant ions are considered, comprising $\approx$ 40\% of the ligand sites. For additional reference, LIGYSIS\textsubscript{NI}, a subset of LIGYSIS without ions, is also included in this analysis. Figure \ref{fig:dataset_comp_1}A shows the number of binding sites per entry across datasets. sc-PDB\textsubscript{FULL}, PDBbind\textsubscript{REF} and SC6K only consider the most relevant ligand for each entry. COACH420 mostly presents single-ligand entries (72\%). bMOAD\textsubscript{SUB} (46\%) and CHEN11 (58\%) present more similar distributions to LIGYSIS, where 54\% of the protein chains present more than one binding site. This percentage decreases for LIGYSIS\textsubscript{NI} (38\%) as ion sites are removed. HOLO4K presents the highest proportion (62\%) of multi-ligand entries. Both HOLO4K and COACH420 are based on asymmetric units, and not biological assemblies. For HOLO4K, 1,811 (40\%) of structures present different numbers of chains between the asymmetric and biological units. This is even more frequent in COACH420: 234 (56\%). Moreover, multimeric complexes might present the same protein-ligand interface repeated across the copies of the complex (Figures \ref{fig:redundant_plis_1}, \ref{fig:redundant_plis_2}). Considering predictions of these interfaces as independent can lead to an overestimate of the performance of a predictor. Regarding the number of chains interacting with a given ligand (Figure \ref{fig:dataset_comp_1}B), CHEN11 and COACH420 present the smaller fraction of multimeric protein-ligand interactions: 3\% and 6\%, respectively, whereas SC6K presents the highest (44\%). The rest of the methods range between 20-30\%. There are no striking differences regarding the size of the interacting protein chains, represented by the number of residues (Figure \ref{fig:dataset_comp_1}C).

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG3_DATASET_COMPARISON_SPLIT_1.pdf}
    \caption[Comparison of datasets (I)]{\textbf{Comparison of datasets (I).} Panels A, B, C, E and F plot the frequencies of binned intervals of a discrete variable, coloured using the \textit{cividis} palette. Interval ranges were selected to facilitate data interpretation. \textbf{(A)} Number of ligand binding sites per dataset entry; \textbf{(B)} Number of ligand-interacting protein chains. This represents whether the ligand interacts with a single protein chain or more; \textbf{(C)} Length of ligand-interacting protein chains (number of amino acids); \textbf{(D)} Ligand molecule type frequency as described in the CCD; \textbf{(E)} Number of ligand atoms; \textbf{(F)} Binding site size, i.e., number of ligand-interacting protein residues. Dashed lines drawn at $p$ = 0.5.}
    \label{fig:dataset_comp_1}
\end{figure}

Figure \ref{fig:dataset_comp_1}D represents the ligand type composition of the datasets. Non-polymer ligands dominate all datasets ($>$66\%), and the proportion of peptides and nucleic acids differ across datasets, with LIGYSIS and LIGYSIS\textsubscript{NI} presenting fewer ligands of these types (1.6\% and 1\%). sc-PDB\textsubscript{FULL} and SC6K are depleted in saccharides ($<$1\%). Figure \ref{fig:dataset_comp_1}E depicts the difference in the number of atoms of the ligands in each dataset. LIGYSIS is of course different, due to its ion ligand content, however, there is no difference between LIGYSIS\textsubscript{NI} and the other datasets. Figure \ref{fig:dataset_comp_1}F conveys the difference in the number of ligand-interacting residues. LIGYSIS has the largest proportion of small sites, 1-10 residues, (56\%). This is directly related to the prominent ion component, and the frequency decreases when ions are removed (LIGYSIS\textsubscript{NI}: 36\%). CHEN11, COACH420 and HOLO4K are more similar to LIGYSIS\textsubscript{NI}, whereas sc-PDB\textsubscript{FULL}, PDBbind\textsubscript{REF} and SC6K are clearly different and present almost exclusively large sites, larger than 20 amino acids, ($>$90\%).

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG3_DATASET_COMPARISON_SPLIT_2.pdf}
    \caption[Comparison of datasets (II)]{\textbf{Comparison of datasets (II).} Five most frequent ligands per dataset. Error bars represent 95\% confidence interval of the proportion \cite{WILSON_197_PROP_CI}. Ligands of similar type are coloured in shades of the same colour: greens for ions, reds for co-factors, blues for energy-carrier molecules, yellows for sugars, grey for peptides and white for other non-polymeric ligands. Above the bars, Shannon Entropy and the proportion of all ligands in each set covered by these top-5 can be found. Both are measures of ligand diversity within each dataset. A subset of LIGYSIS with no ions (NI), LIGYSIS\textsubscript{NI}, is included in this analysis, as most training and test datasets do not consider ions. CLR: cholesterol; MAN: mannose; FUC: fucose; NAP: nicotinamide-adenine-dinucleotide phosphate; SAH: S-Adenosyl-L-homocysteine; FMN: Flavin mononucleotide; GAL: galactose; N-mer: protein peptides of N amino acids; 017: Darunavir; PLP: Vitamin B6 phosphate; BGC: glucose; XYP: xylose.}
    \label{fig:dataset_comp_2}
\end{figure}

Figure \ref{fig:dataset_comp_2} explores the ligand diversity on each dataset by showing the top-5 most frequent ligands per dataset, the proportion of the total number of ligands they represent, as well as Shannon entropy $H'$. Shannon entropy is a measure of diversity. Larger numbers indicate a more evenly spread distribution of a larger number of different molecules, whereas small numbers indicate higher frequency of a few ligands. Whilst four out of the top-5 ligands of LIGYSIS are ions (Zn\textsuperscript{+2}, Ca\textsuperscript{+2}, Mg\textsuperscript{+2}, Mn\textsuperscript{+2}) and represent 19.2\% of all ligands, its diverse composition is comparable to that of PDBbind\textsubscript{REF}. Removing ions, LIGYSIS\textsubscript{NI} becomes the most diverse dataset with $H'$ = 8.8 and its top-5 ligands only covering 5.3\% of all ligands in the set. SC6K is the least diverse with its top-5 most frequent ligands covering 33\% of all ligands. All datasets, except for LIGYSIS, LIGYSIS\textsubscript{NI}, and PDBbind\textsubscript{REF}, are dominated by co-factor ligands, such as flavin-adenine dinucleotide (FAD), nicotinamide-adenine dinucleotide (NAD), and haem (HEM) or energy carrier molecules such as adenine tri-, di- and monophosphate (ATP, ADP, AMP). Short peptides ($<$10 aas) are the most common ligands in PDBbind\textsubscript{REF} (5\%), and energy carriers represent $<$2\% of the top-5 ligands. Cholesterol, mannose and fucose are some of the most common ligands in LIGYSIS\textsubscript{NI}. For all further analysis, including characterisation of binding pockets and performance evaluation of the prediction methods, LIGYSIS, i.e., including ions, was utilised.

\subsection{Binding pocket characterisation}

After removing backbone-only chains and those with missing residue mapping to UniProt, the final LIGYSIS set which was employed for the benchmark of the methods comprises 2,775 representative chains. Not all methods predict pockets on all the chains. VN-EGNN, GrASP, fpocket, PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+}, and Surfnet\textsuperscript{+} predict in $>$99\% of the chains, P2Rank\textsubscript{CONS} on 93\%, followed by P2Rank on 86\%, PUResNet and DeepPocket\textsubscript{SEG} (85\%), and finally IF-SitePred only predicts pockets on 75\% of the chains. PUResNet, DeepPocket, P2Rank\textsubscript{CONS} and P2Rank often don't predict on smaller proteins ($<$100 amino acids) as well as non-globular or elongated proteins, representing 60-80\% of proteins with no predicted pockets. However, for IF-SitePred larger globular proteins represent $\approx$50\% of all proteins where this method fails to predict a pocket (Figure \ref{fig:missed_preds_prot_class}). Predicted residue ligandability scores for P2Rank\textsubscript{CONS}, P2Rank and IF-SitePred (which we derived in this work), were examined for the proteins with no predicted pockets. Figure \ref{fig:ifsp_missed_preds} illustrates 8 examples of proteins where residues with IF-SitePred (Equation \ref{eq:IFSP_score}) high ligandability scores cluster in space into clear binding sites that are not reported as predictions by this method. This suggests that IF-SitePred is too strict in selecting only those residues predicted as ligand-binding by all 40 models. The cloud point selection clustering approach or threshold in this method may also play a role in this.

\begin{figure}[ht!]
    \centering
    \includegraphics[width=0.50\textwidth]{figures/ch_LBS_COMP/SUPP/PDF/SUPP_FIG1_MISSED_PREDS_PROT_CLASS.pdf}
    \caption[Where methods do not predict any sites]{\textbf{Where methods do not predict any sites.} IF-SitePred does not predict any ligand binding sites on 700 of the 2,775 protein chains in the LIGYSIS set (25\%), PUResNet on 415 (15\%), DeepPocket\textsubscript{SEG} (426; 15\%), P2Rank\textsubscript{CONS} (196; 7\%) and P2Rank (373; 13\%). All methods struggle to predict on elongated proteins, regardless of their size, as well as on tiny globular proteins. Globular proteins comprise the most common group amongst the proteins with no predictions for IF-SitePred (53\%).}
    \label{fig:missed_preds_prot_class}
\end{figure}

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG4_IFSP_MISSED_PREDICTIONS_OPT.pdf}
    \caption[IF-SitePred ``missed'' predictions]{\textbf{IF-SitePred ``missed'' predictions.} Eight examples of human protein chains where IF-SitePred does not report any predicted ligand binding sites. Predictions are made on ligand-stripped chains. Ligand molecules, in orange, are superposed to illustrate how the ligandability scores recapitulate the observed binding site. These are protein representative chains and ligand molecules might not be observed in the same entry; \textbf{(A)} GDP-fucose protein O-fucosyltransferase 2, Q9Y2G5, with GFB superimposed (PDB: 4AP6) \cite{CHEN_2012_POFUT2}; \textbf{(B)} tRNA (cytosine(72)-C(5))-methyltransferase NSUN6, Q8TEA1, (PDB: 5WWT) \cite{LIU_2017_NSUN6} with superposed SFG (PDB: 5WWR); \textbf{(C)} Tubulin beta-2B chain, Q9BVA1, with G2P (PDB: 7ZCW) \cite{RAMIREZ_2023_VASH2}; \textbf{(D)} Cyclic GMP-AMP phosphodiesterase SMPDL3A, Q92484, with C5P (PDB: 5EBE) \cite{LIM_2016_SPHINGOMYELIN}; \textbf{(E)} tRNA (adenine(58)-N(1))-methyltransferase catalytic subunit, Q96FX7, with SAH (PDB: 5CCB) \cite{FINER_2015_tRNA}; \textbf{(F)} Chronophin, Q96GD0, (PDB: 5GYN) (Lee HJ, Cho HJ, Kang BS, 2017) with PLP (PDB: 2FCT) \cite{BLASIAK_2006_SYRB2}; \textbf{(G)} Mitochondrial Methylmalonic aciduria type A protein, Q8IVH4, with GDP (PDB: 8GJU) \cite{MASCARENHAS_2023_GPROTEIN}; \textbf{(H)} Renalase, Q5VYX0, (PDB: 3QJ4) with FAD \cite{MILANI_2011_NADP}. Residues are coloured based on the ligandability score calculated by averaging the probabilities predicted by each of the 40 IF-SitePred prediction models. This is a score ranging 0-1 which is indicative of the likelihood of a given residue binding a ligand. Clear pockets can be observed formed by residues with high ligandability scores (darker blue colour), which agree with the sites where ligands bind.}
    \label{fig:ifsp_missed_preds}
\end{figure}

Table 3 summarises the ligand site characterisation analysis. fpocket predicts the most sites out of all the methods, with 57,859, followed by IF-SitePred (44,948), DeepPocket\textsubscript{SEG} (21,718), VN-EGNN (13,582), P2Rank (12,412), P2Rank\textsubscript{CONS} (10,180), Surfnet\textsuperscript{+} (9,043), PocketFinder\textsuperscript{+} (8,913), Ligsite\textsuperscript{+} (6,903), GrASP (4,694) and PUResNet, which predicts fewest sites (2,621). LIGYSIS defines 6,882 binding sites from experimental data. Relative to LIGYSIS, the prediction methods have ratios of predicted/defined sites ranging from 8.4 (fpocket) to 0.4 (PUResNet) with P2Rank\textsubscript{CONS} in the middle, predicting 1.5 pockets per observed reference site. IF-SitePred, DeepPocket\textsubscript{SEG} as well as P2Rank and fpocket predict more pockets on larger protein chains, whereas the rest of methods do not (Figure \ref{fig:sites_vs_prot_size}). This effect is most clear with fpocket, which predicts 350 pockets for chain A of PDB: 7SUD \cite{LIU_2022_DNAPK}, a structure of the DNA-dependent protein kinase catalytic subunit, DNPK1, (P78527) with 3,736 amino acid residues. In contrast, VN-EGNN, which initially places $K$ = 8 virtual nodes, results in a maximum of 8 predicted pockets, regardless of protein chain size, and PUResNet predicts a single pocket in 90\% of the protein chains. 

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/SUPP/PDF/SUPP_FIG2_POVKETS_VS_PROTEIN_SIZE.pdf}
    \caption[Number of pockets \textit{vs} protein size]{\textbf{Number of pockets \textit{vs} protein size.} Number of defined (LIGYSIS) and predicted sites against protein chain size, i.e., number of amino acid residues. Number of residues has been discretised into intervals of 50 until 650, and larger intervals until the maximum, $\approx$3,800. Error bars represent one standard deviation (SD).}
    \label{fig:sites_vs_prot_size}
\end{figure}

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG5_BINDING_POCKET_FEATURES_SPLIT_1_OPT.pdf}
    \caption[Binding pocket characterisation (I)]{\textbf{Binding pocket characterisation (I).} Violin plots show the main distributions and swarm plots are used to show outliers. Data points farther than four standard deviations (SD) from the mean are considered outliers. The limit of the Y axis is the maximum non-outlier value plus a buffer value. This way, only the most extreme outliers are hidden, which maximises visual interpretation of the data whilst minimising the number of data points not shown. Within the violin plots are box plots representing the underlying distribution. Line represents the median, box contains the interquartile range (IQR), and whiskers extend to 1.5 $\times$ IQR. \textbf{(A)} Number of pockets per protein; \textbf{(B)} Pocket radius of gyration $R_{g}$ (\AA{}).}
    \label{fig:pocket_features_1}
\end{figure}

Figures \ref{fig:pocket_features_1}, \ref{fig:pocket_features_2} represents how the eleven ligand site predictors compare to each other as well to LIGYSIS, which defines ligand sites from experimentally determined biologically relevant protein-ligand complexes. Figure \ref{fig:pocket_features_2}A shows how VN-EGNN, GrASP and PUResNet differ from the other methods with a maximum of 7, 12 and 4 predicted pockets, respectively. PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+} and Surfnet\textsuperscript{+} present narrow distributions like LIGYSIS and with medians of 1-3 pockets per protein. P2Rank\textsubscript{CONS} and P2Rank also present a median of 3 pockets per protein but display wider distributions as they can predict up to 60, and 80 pockets per protein, respectively. Overall, P2Rank\textsubscript{CONS} predicts fewer pockets than P2Rank. IF-SitePred, DeepPocket\textsubscript{SEG} and fpocket follow, with a median of 20, 6 and 17 pockets. The difference in number of pockets between DeepPocket\textsubscript{SEG} and DeepPocket\textsubscript{RESC} or fpocket is because 60\% of fpocket candidates are not extracted by the CNN segmentation module implemented in DeepPocket.

%\begin{figure}[htbp!]
%    \centering
%    \includegraphics[width=0.87\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/%FIG5_BINDING_POCKET_FEATURES_OPT.pdf}
%    \caption[Binding pocket characterisation]{\textbf{Binding pocket characterisation.} Violin plots %show the main distributions and swarm plots are used to show outliers. Data points farther than four %standard deviations (SD) from the mean are considered outliers. The limit of the Y axis is the maximum %non-outlier value plus a buffer value. This way, only the most extreme outliers are hidden, which %maximises visual interpretation of the data whilst minimising the number of data points not shown. %Within the violin plots are box plots representing the underlying distribution. Line represents the %median, box contains the interquartile range (IQR), and whiskers extend to 1.5 $\times$ IQR. %\textbf{(A)} Number of pockets per protein; \textbf{(B)} Pocket radius of gyration $R_{g}$ (\AA{}); %\textbf{(C)} Minimum inter-pocket centroid distance (MCD) (\AA{}). This is a measure of how close %predicted pockets are to each other within a protein; \textbf{(D)} Maximum inter-pocket residue overlap %(MRO). Residue overlap was calculated as Jaccard Index. This is a measure of how much the pockets %overlap in terms of binding residues.}
%    \label{fig:pocket_features}
%\end{figure}

Figure \ref{fig:pocket_features_1}B shows the distribution of pocket radius of gyration, $R_{g}$. VN-EGNN and IF-SitePred differ from the rest of methods with narrow distributions and medians around 6\AA{}. These two methods do not report pocket residues. Instead, they were obtained using a distance threshold of 6\AA{} from the centroid, for VN-EGNN, and cloud points, for IF-SitePred. This is reflected by examining the percentage of pockets with $R_{g} >$ 10\AA{} which is 0\% and 0.1\% for VN-EGNN and IF-SitePred. This is a striking difference compared to the other methods: 1.8\% (fpocket), 4.8\% (P2Rank), 5.7\% (DeepPocket\textsubscript{SEG}), 6.4\% (GrASP), 6.5\% (P2Rank\textsubscript{CONS}) PUResNet (11.6\%), 12.6\% (LIGYSIS), 16\% (Surfnet\textsuperscript{+}), 21.4\% (PocketFinder\textsuperscript{+}) and 33.5\% (Ligsite\textsuperscript{+}). The latter three predict the sites with largest median $R_{g} \approx$ 9\AA{}. VN-EGNN, GrASP, PUResNet and DeepPocket\textsubscript{SEG} predict sites with $R_{g}$ = 0\AA{}. This is rather infrequent (7.8\% GrASP) and $<$3\% for the other three. These examples correspond to singletons, i.e., pockets formed by only one amino acid.

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG5_BINDING_POCKET_FEATURES_SPLIT_2_OPT.pdf}
    \caption[Binding pocket characterisation (II)]{\textbf{Binding pocket characterisation (II).} Violin plots show the main distributions and swarm plots are used to show outliers. Data points farther than four standard deviations (SD) from the mean are considered outliers. The limit of the Y axis is the maximum non-outlier value plus a buffer value. This way, only the most extreme outliers are hidden, which maximises visual interpretation of the data whilst minimising the number of data points not shown. Within the violin plots are box plots representing the underlying distribution. Line represents the median, box contains the interquartile range (IQR), and whiskers extend to 1.5 $\times$ IQR. \textbf{(A)} Minimum inter-pocket centroid distance (MCD) (\AA{}). This is a measure of how close predicted pockets are to each other within a protein; \textbf{(B)} Maximum inter-pocket residue overlap (MRO). Residue overlap was calculated as Jaccard Index. This is a measure of how much the pockets overlap in terms of binding residues.}
    \label{fig:pocket_features_2}
\end{figure}

Figure \ref{fig:pocket_features_2}A illustrates how close predicted sites are to each other within a protein chain. Pairwise distances between the centroids of all ligand site pairs for a protein are calculated, and for each site, the minimum distance is taken. Sites predicted by VN-EGNN, IF-SitePred and DeepPocketSEG are very close to each other, with median distances ($\tilde{d}$) of 1.1, 3.4 and 4.6\AA{}, respectively. fpocket follows with $\tilde{d}$ = 9.7\AA{}. The rest of the methods present median distances ranging between 13-18\AA{} (LIGYSIS, P2RankCONS, P2Rank, PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+}, Surfnet\textsuperscript{+}), and finally GrASP ($\tilde{d}$ = 21.7\AA{}) and PUResNet ($\tilde{d}$ = 27\AA{}). Both versions of P2Rank present the most similar distribution to what is observed on LIGYSIS.

Figure \ref{fig:pocket_features_2}B depicts the overlap existing between residues that form the predicted pockets within a protein. All pairwise overlaps, i.e., Jaccard Index, are calculated between pockets in a chain, and for each pocket, the maximum is taken. This is a measure of how much the predicted pockets overlap. This is directly related to how close pockets are, and so VN-EGNN, IF-SitePred and DeepPocket\textsubscript{SEG} present very high overlaps $\tilde{o}$ = 0.85, $\tilde{o}$ = 0.55 and $\tilde{o}$ = 0.4, respectively. fpocket follows with $\tilde{o}$ = 0.15, Ligsite\textsuperscript{+} ($\tilde{o}$ = 0.09), Surfnet\textsuperscript{+} ($\tilde{o}$ = 0.07), P2Rank\textsubscript{CONS}, P2Rank and PocketFinder\textsuperscript{+} ($\tilde{o}$ = 0.05), and finally LIGYSIS, GrASP and PUResNet with $\tilde{o}$ = 0. GrASP is the only method of the 12 presented here that clusters atoms directly, and as a result, overlap between pockets is minimised. Other methods cluster ligands (LIGYSIS), cloud points (IF-SitePred), SAS points (P2Ranks) voxels (PUResNet, DeepPocket), alpha spheres (fpocket), or grid points (PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+}, Surfnet\textsuperscript{+}) but not residues, resulting consequently in higher overlapping.

Proximity in space between predicted sites as well as residue overlap are indicators of redundant ligand binding site predictions, i.e., duplicate predictions of a unique observed ligand site. This is the case for VN-EGNN, IF-SitePred and DeepPocket\textsubscript{SEG}. This phenomenon could negatively impact the precision and recall of these methods. Accordingly, correcting for redundancy should have a significant impact on the performance of these methods. In contrast, GrASP and PUResNet which predict a small number of pockets show low proximity and overlap of predicted sites and so redundancy is not an issue.

\subsection{Evaluation of predictive performance}

\subsubsection{Pocket level evaluation}

The ideal ligand binding site predictor would have a high precision, i.e., most of the predictions it makes are correct, whilst maintaining a high recall, i.e., recapitulating most of the observed sites. Moreover, the ideal predictor returns predictions that are non-redundant, i.e., it does not predict the same pocket multiple times. Additionally, pockets are ranked in a systematic manner according to a strong and meaningful pocket scoring scheme which captures well the nature of existing ligand binding sites and therefore ranks the predicted pockets from more likely (high score, top) to least likely (low score, bottom). A good predictor would also perform well at the residue level. This means it is able of capturing the likelihood for a residue to bind a ligand. This can be done by means of a residue ligandability score, which additionally might highlight key residues, the more ligandable within a binding site. Ligand site prediction methods were benchmarked with these criteria in mind.

Figure 7A illustrates the recall curve for top-$N$+2 pockets for each method, where $N$ is the number of observed sites for a target protein. Reported recall is obtained using DCC = 12\AA{}. DeepPocket\textsubscript{RESC} yields the highest recall, 58.1\%, closely followed by, P2Rank\textsubscript{CONS} (53.9\%) and P2Rank (51.9\%). The rest of the methods present recall $<$50\% with PUResNet, VN-EGNN and IF-SitePred with lowest recall of 41.1\%, 40.9\% and 25.7\%, respectively (Table 4). Figure 7B, shows the recall curve considering different top-$N$+X predictions. Most methods reach a plateau by top-$N$+5, as they do not predict that many pockets. However, methods that predict more pockets per protein, such as IF-SitePred, fpocket, DeepPocket\textsubscript{SEG} and DeepPocket\textsubscript{RESC}, which take fpocket predictions as a base, increase their recall as more predictions are considered. fpocket and DeepPocket\textsubscript{RESC} reach a maximum recall of $\approx$90\% when all predictions are considered, regardless of rank. Other methods present maximum recall of $\approx$40-50\%. Figure 7C illustrates the recall curve if residue overlap was used instead of DCC as a criterion. In this case, Ligsite\textsuperscript{+}, PocketFinder\textsuperscript{+}, and Surfnet\textsuperscript{+} come on top with recall $\approx$45\% at $I_{rel} \geq$ 0.5. This is explained by their prediction of massive cavities, that while often fully contain or overlap with the observed pocket, do not meet the DCC criterion, as their centroids are farther than 12\AA{} from the observed site.

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG7_POCKET_LEVEL_BENCHMARK_OG_OPT.pdf}
    \caption[Ligand binding site prediction benchmark at the pocket level]{\textbf{Ligand binding site prediction benchmark at the pocket level.} \textbf{(A)} Recall, percentage of observed sites that are correctly predicted by a method according to a DCC = 12\AA{} threshold; \textbf{(B)} Recall using DCC = 12\AA{} but considering increasing rank thresholds, i.e., top-$N$, $N$+1, $N$+2, etc. \textit{ALL} represents the maximum recall of a method, obtained by considering all predictions, regardless of their rank or score; \textbf{(C)} Recall curve using $I_{rel}$ as a criterion; \textbf{(D)} ROC100 curve  (cumulative \#TP against cumulative FP until 100 FP are reached); \textbf{(E)} Precision curve for the top-1,000 predictions of each method across the LIGYSIS dataset, Precision\textsubscript{1K}. Error bars represent 95\% CI of the recall (A-C) and precision (E), which are 100 $\times$ proportion. Numbers at the right of the panels indicate groups or blocks of methods that perform similarly for each metric. Asterisks (*) indicate outlier methods, or methods that perform very differently than the rest.}
    \label{fig:pocket_level_benchmark_OG}
\end{figure}

Figure 7D represents the cumulative number of TP against FP when predictions across the proteins in the reference dataset are sorted by score. This shows how effective the scoring scheme of each method is in ranking their predictions to reflect the nature of ligand binding sites. At 100 FP the \#TP fall into three different blocks and one outlier: Ligsite\textsuperscript{+}, fpocket, Surfnet\textsuperscript{+} and PocketFinder\textsuperscript{+} are at the bottom with \#TP $\ni$ (60, 120). Secondly, DeepPocket\textsubscript{SEG}, DeepPocket\textsubscript{RESC}, P2Rank, and PUResNet follow with \#TP $\ni$ (530, 670). GrASP, IF-SitePred and P2Rank\textsubscript{CONS}, present a high \#TP ranging 900-1,000 at 100 FP. Finally, VN-EGNN sits at the top with 1,301 TP. However, this number is not representative, as the \#TP are inflated due to the redundancy in the predictions of VN-EGNN. This is the same for IF-SitePred and DeepPocket\textsubscript{SEG}. Redundant correct predictions of the same pocket will count as multiple TPs, whereas they should only count as 1 TP. Newer methods, e.g., GrASP, P2Rank\textsubscript{CONS}, and VN-EGNN, and IF-SitePred, despite redundancy in their prediction for the latter two, are better at ranking their predicted pockets, presenting up to 1,000 more TP for 100 FP than earlier methods. This means their scoring schemes are significantly better at capturing the essence of a ligand binding site. Including evolutionary conservation in P2Rank (P2Rank\textsubscript{CONS}) predictions results in an increase of +346 TP relative to default P2Rank, indicating that the fewer predicted pockets, and their scores are a more faithful representation of the observed LIGYSIS dataset.

Figure 7E provides insight into the precision of the methods by examining how this metric changes as more predictions are considered. In the same manner as for Figure 7D, predictions across proteins in the LIGYSIS dataset are sorted and cumulative precision is plotted for the top-1,000 scoring predictions. Methods group into two clear blocks. Newer (machine learning-based) methods VN-EGNN, GrASP, IF-SitePred, P2Rank\textsubscript{CONS}, DeepPocket\textsubscript{SEG}, DeepPocket\textsubscript{RESC} and PUResNet present a Precision\textsubscript{1K} of 80-95\%. Earlier (geometry/energy-based) methods Ligsite\textsuperscript{+}, fpocket, PocketFinder\textsuperscript{+} and Surfnet\textsuperscript{+} present lower Precision\textsubscript{1K} of 40-50\%. DeepPocketSEG takes fpocket (geometry-based) predictions as a starting point and achieves much higher \#TP\textsubscript{100FP} as well as Precision\textsubscript{1K}. This is further evidence that performance can be boosted with a solid scoring scheme.

Table 4 summarises these results and shows the median relative residue overlap ($RRO$) and relative volume overlap ($RVO$), which are qualitative measures of how well predicted sites align with observed ones in shape. VN-EGNN and IF-SitePred present the smallest $RRO$ and $RVO$, but it is important to note that these methods do not report pocket residues and so residues were taken within 6Å of their centroid, or pocket spheres. PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+}, and Surfnet\textsuperscript{+} present unusually high $RRO$ and $RVO$. This is a consequence of the massive size of their predicted cavities, that rather than overlap with the observed site, fully contain and are much larger than it. This might not be convenient in the context of pocket finding for drug discovery where more clearly defined drug-like sites might be of interest. GrASP, PUResNet and DeepPocket\textsubscript{SEG} present high values of $RRO \approx$ 60\% and $RVO \approx $70\% whilst presenting a size distribution more like LIGYSIS (Figure 5B) and provide the best representation of the observed sites in terms of shape similarity.

\subsubsection{Residue level evaluation}

Ligand binding site prediction tools can also be evaluated at the residue level. $F1$ Score as well as Matthews correlation coefficient ($MCC$) were utilised to assess prediction performance at the residue level. For each protein chain, $F1$ and $MCC$ were obtained, distributions graphed, and medians reported (Table 5). IF-SitePred does not report a residue ligandability score beyond a binary label (0, 1). In this work, a score was computed by utilising the scores returned by the 40 prediction models of IF-SitePred. These scores range 0-1 and can be averaged as probabilities (Equation \ref{eq:IFSP_score}). This will now be referenced as IF-SitePred ligandability score.

Figure 8A-B illustrate the distributions of the $F1$ score and $MCC$ for each method on the 2,775 protein chains of the final LIGYSIS set. Both metrics agree that PUResNet ($F1$ = 0.41, $MCC$ = 0.39), GrASP ($F1$ = 0.39, $MCC$ = 0.33) and P2Rank\textsubscript{CONS} ($F1$ = 0.36, $MCC$ = 0.30) are the top-3 performing methods in this task of binary classification into pocket (1) and non-pocket residues (0). fpocket presents the lowest $F1$ = 0.23 and $MCC$ = 0.12 since it predicts many unobserved pockets (residues) that will count as FP here.

\begin{figure}[ht!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG8_RESIDUE_LEVEL_BENCHMARK_OPT.pdf}
    \caption[Ligand binding site prediction benchmark at the residue level]{\textbf{Ligand binding site prediction benchmark at the residue level.} \textbf{(A)} F1 score distributions; \textbf{(B)} MCC distributions. In both panels, each data point in the distribution corresponds to the score obtained from all residues in a protein chain; (C) ROC curve for methods that report a residue score. Dashed line represents the baseline, 1 FP for each TP, i.e., diagonal and AUC = 0.5; (D) PR curve. Dashed line represents the baseline, i.e., proportion of observed binding residues (true positives) = 0.1; (E) Distribution of residue ligandability scores for IF-SitePred, GrASP, P2Rank\textsubscript{CONS} and P2Rank. PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+} and Surfnet\textsuperscript{+} are not included as their scores do not range 0-1, and a small number of scores can reach values $>$ 25.}
    \label{fig:residue_level_benchmark}
\end{figure}

For IF-SitePred, GrASP, P2Rank\textsubscript{CONS}, P2Rank, PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+}, and Surfnet\textsuperscript{+}, which report a residue level score (beyond a binary label), ROC and PR curves were plotted (Figure 8C) and area under the curve (AUC) and average precision (AP) reported. IF-SitePred, with the ligandability score calculated in this work (Eq. 10), presents the highest AUC = 0.80, closely followed by P2Rank\textsubscript{CONS} (AUC = 0.78) and P2Rank (AUC= 0.75). Surfnet\textsuperscript{+} presents the lowest AUC = 0.71. Figure 8D shows the PR curves, which agree with ROC AUC and highlight IF-SitePred (with \autoref{eq:IFSP_score} scoring) as the method with the highest average precision = 0.42, followed by P2Rank\textsubscript{CONS} (AP = 0.41) and Surfnet\textsuperscript{+} the lowest with (AP = 0.25). Figure 8E shows IF-SitePred presenting a different residue ligandability score distribution to GrASP, P2Rank\textsubscript{CONS}, and P2Rank. The IF-SitePred ligandability score, resulting from averaging the scores from the 40 IF-SitePred models, is the most ``generous'' with $\approx$20\% of the residues presenting a score > 0.5, in contrast with GrASP which residue scoring is very strict (\% ($LS \geq$ 0.5) = 0.75\%) and P2Ranks ($\approx$3\%). This difference, combined with the ROC and PR curves further supports the use of the IF-SitePred ligandability score proposed in this work to define the predicted binding sites in this method. It also suggests that GrASP might benefit from a less strict residue level scoring scheme. PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+} and Surfnet\textsuperscript{+} were not included in this analysis as their scores do not range 0-1 and very high scores ($>$25) can be obtained.

\section{Discussion}

In this work we have carried out the most comprehensive comparative analysis of ligand binding site prediction methods to date, spanning three decades of methods development. First, predictions from the different methods as well as observed sites from our reference dataset, LIGYSIS, were compared in terms of the number of proteins methods predict on, the number of predicted sites per protein, their size, distance and overlap between the sites. This analysis provides insight into how the different methods work and hints at potential limitations or room for improvement, e.g., the prediction of a fixed number of sites per protein, or considerable proximity and overlap between the predictions.

Recall (\% of sites that are correctly predicted) is more informative than precision (\% of predictions that are correct). In a context of discovery where the true ligand binding sites of the target are unknown, it is more useful to have multiple predictions that might or might not correspond to real sites (lower precision), rather than a single or few predictions that are very precise but are missing other likely sites. Most methods do well in predicting the most obvious (orthosteric) site. This site, however, might not be available for therapeutic targeting and it is convenient to have other predicted sites that could modulate function acting as allosteric sites. Precision, though a metric that provides valuable insight, and covered in this work, must always be contextualised with recall. Our results show that the most precise methods do not correspond to higher recalling methods. A method predicting the most obvious site, that could be identified by eye, might be 90\% precise, but present a lower recall, e.g., 30\%. This matter is of even more importance if the ``success rate'' is defined as the precision of the top scoring 1-3 predictions, which is not a very representative performance assessment metric.

For this reason, we recommend that developers of methods not only share the code of the method, but also of the benchmarking analysis. Moreover, the definition of success rate must be standardised as recall, as some methods use recall, and others precision, both under the same term of ``success rate''. This can be confusing when comparing the results from different analyses. Alternatively, success rate should be clearly defined, so readers can fully understand the implications of the metric employed on a benchmark.

It is clear from our results that a DCC threshold of 4\AA{Å} is too conservative, and a more flexible DCC threshold of 10-12\AA{} should be used for comparable performance with DCA = 4\AA{}. According to our results and our reference set, predictions with DCC 4-12\AA{} overlap or are adjacent to observed sites and should be considered as correct predictions. The reason for this is the inherent noise in the ground truth, observed data or reference dataset, i.e., a ligand binding to a cavity might not be representative of all ligands that could bind to it. For most proteins, not all existing ligand sites are characterised and as different ligands can bind to the same region, it is unrealistic to use such a small DCC threshold. Our results show several examples of predictions of an observed cavity where DCC $>$ 4\AA{}.

The usefulness of residue-level metrics as F1 score or MCC is limited, as methods that precisely and correctly predict the clearest sites, such as PUResNet (high precision, low recall) perform better on these metrics, whilst methods that predict more pockets (lower precision, higher recall) such as fpocket and re-scored versions, will obtain worse results. Pocket-level metrics, particularly recall, are more representative of the ability to predict ligand binding sites.

Whilst datasets like PDBbind, binding MOAD or the brand new PLINDER \cite{DURAIRAJ_2024_PLINDER} (will) prove extremely useful to train, validate and test deep learning models tackling problems such as rigid body docking \cite{STARK_2022_EQUIBIND}, flexible pocket docking \cite{QIAO_2024_DGN}, or pocket-conditioned ligand generation \cite{SCHNEUING_2023_DIFFUSION}, they might not be ideal as a test set for ligand binding site prediction. LIGYSIS analyses all unique, biologically relevant protein-ligand interfaces, including ions, across the biological assembly from multiple experimentally determined structures of a given protein. It then clusters these ligands based on their interactions with the protein, resulting in the observed binding sites. Beyond considering biological assemblies and unique protein-ligand interfaces, the greatest innovation in LIGYSIS is exploiting the extensive structural data on the PDBe-KB to aggregate ligand-binding interactions across different structures of the same protein, thus better capturing the ligand-binding capabilities than just taking a single structure of a protein-ligand complex. In doing so, LIGYSIS represents the most complete and non-redundant protein-ligand complex dataset to date. 

Aggregating protein-ligand interactions across structures of the same protein is likely to be beneficial not only for testing, but also when training these methods. Most current methods train on datasets where a protein is represented by a single structure interacting with a single ligand. For example, in 100\% of sc-PDB and 50\% of entries for binding MOAD training sets. Methods consider as ligand binding, and therefore TP, those residues within a certain distance of the ligand and TN all other residues. In doing so, residues of the same protein that bind ligands on other structures, but not the one present on these sets, will be incorrectly labelled as ``non-ligand-binding'' (FN). This mislabelling of residues could lead to a lower prediction performance. This issue is to a certain extent approached by P2Rank and GrASP, which enriched their training datasets by including ligands from other chains, or homologous structures. This noise in the training dataset might be more prevalent for DeepPocket, PUResNet and VN-EGNN, which seem to rely fully on 1:1 interactions. The usage of LIGYSIS, or any other data set that aggregates ligand interactions across structures, might alleviate this issue and hints at potential room for improvement in the field of ligand binding site prediction.

\section{Conclusions}

The conclusions resulting from our analysis are as follows:

\begin{itemize}

\item LIGYSIS aggregates non-redundant biologically relevant protein-ligand interactions across multiple structures for a protein and sets a new test set standard for the benchmark of ligand binding site prediction tools. 

\item Ligand binding site prediction methods differ significantly in the number of predicted sites, their size, proximity and overlap, which offers insight into how the methods work.

\item Recall is a more informative measure of the performance of a ligand site prediction tool, rather than precision and so it must be reported. Precision, though a useful metric, should always be contextualised with recall.

\item ll authors of ligand site prediction tools should use recall as “success rate” for consistency. Benchmarking code should also be shared by the authors for the sake of reproducibility.

\item Pocket-level metrics (recall, precision) are a more adequate representation of the capabilities of ligand site prediction methods than residue-level metrics (F1, MCC).

\item A DCC threshold of 4\AA{} is too conservative, and to obtain comparable results between DCA and DCC recall, a threshold of DCC of 10-12\AA{} should be employed.

\item Re-scoring of fpocket predictions by DeepPocket\textsubscript{RESC} present the highest recall (60\%) among the methods reviewed in this analysis.

\item Methods that systematically predict a low number of pockets, e.g., VN-EGNN, GrASP or PUResNet, are very precise ($>$90\%), however their recall is low, and might not be as useful in a discovery context.

\end{itemize}
