\chapter{Comparative evaluation of methods for the prediction of protein-ligand binding sites}

\section*{Preface}

XXX.

\section*{Publications}

This chapter will include the results published in here \textit{"Comparative evaluation of methods for the prediction of protein-ligand binding sites"}.

\section*{Author contributions}

G.J.B. and J.S.U., conceived, designed, and developed the research. J.S.U. analysed the data. J.S.U. developed the software. J.S.U. and G.J.B. wrote, reviewed and edited the manuscript. G.J.B. secured funding and supervised.

\section{Methods}

\subsection{Comparison of datasets}

Training and test datasets were downloaded for all machine learning based methods reviewed in this work. Datasets were compared to our reference set, LIGYSIS, in terms of number of sites per protein, ligand-interacting chains, chain lengths, size of the sites (number of amino acids), ligand composition, size and diversity. Ligand diversity was quantified by Shannon’s Entropy \cite{SHANNON_1948_ENTROPY} (Equation \ref{eq:entropy_shannon}) where pi represents the proportion of each ligand $i$ of the $R$ ligands observed in the dataset. Ligand data was extracted from the Chemical Component Dictionary (CCD) \cite{WESTBROOK_2015_CCD}. An overlap (\%) was calculated for each dataset as the proportion of LIGYSIS binding sites that were covered by at least one ligand in a test dataset. A simplistic approach was adopted by calculating the intersection of ligand IDs between LIGYSIS and each dataset. Ligand IDs were defined as a string of PDB ID + ``\_'' + ligand ID, e.g., ``6GXT\_GTP'' corresponds to the guanosine-5’-triphosphate of the PDB entry with ID: 6GX7 \cite{CAMPANACCI_2019_TUBULIN}.

\begin{equation}
H' = - \sum_{i=1}^{R} p_i \ln(p_i)
\label{eq:entropy_shannon}
\end{equation}

\subsection{Training datasets}

VN-EGNN trains on a subset \cite{KANDEL_2021_PURESNET} of the sc-PDB (v2017) \cite{PAUL_2004_SCPDB, KELLENBERGER_2006_SCPDB, MESLAMANI_2011_SCPDB, DESAPHY_2015_SCPDB} (sc-PDB\textsubscript{SUB}). sc-PDB is a comprehensive database of pharmacological ligand-protein complexes. The database is comprised by proteins in complex with buried, biologically relevant synthetic or natural ligands deposited in the PDB. sc-PDB contains unique non-repeating protein-ligand pairs, which means that only one ligand is considered per PDB structure entry. Smith \textit{et al.} \cite{SMITH_2024_GrASP} enriched this dataset with 9,000 extra ligands resulting in a version of sc-PDB which we call sc-PDB\textsubscript{RICH}, which GrASP trained on. Unfortunately, this dataset is not publicly accessible and therefore not considered in our analysis. DeepPocket used the full sc-PDB set to train on, sc-PDB\textsubscript{FULL}. IF-SitePred uses a sequence identity-filtered version of the non-redundant subset of the binding mother of all databases (MOAD) \cite{HU_2005_BMOAD, BENSON_2008_BMOAD, AHMED_2015_BMOAD, SMITH_2019_BMOAD}, which considers only protein family leaders. The binding MOAD, here referred to as bMOAD\textsubscript{SUB}, is a large collection of crystal structures with clearly identified biologically relevant ligands with binding data extracted from the literature. Finally, P2Rank used the CHEN11 dataset to train, which aimed to cover all SCOP \cite{HUBBARD_1997_SCOP, HUBBARD_1998_SCOP, LOCONTE_2000_SCOP} families of ligand binding proteins in a non-redundant manner \cite{CHEN_2011_ASSESSMENT} and the JOINED dataset for validation. CHEN11 not only considers the ligands in each structure but is enriched with ligands binding to homologous structures. JOINED is a combined dataset formed by other smaller datasets: ASTEX \cite{HARTSHORN_2007_ASTEX}, UB48 \cite{HUANG_2006_BU48}, DT198 \cite{ZHANG_2011_METAPOCKET} and MP210 \cite{HUANG_2009_METAPOCKET}, which represent diverse collections of protein-ligand complexes, including bound/unbound states, drug-target complexes and other ligand site predictor benchmark sets.

\subsection{Test datasets}

The majority of ligand binding site predictors published since 2018 have been using two datasets that were first presented by Krivák \textit{et al.} \cite{KRIVAK_2018_P2RANK}: COACH420 and HOLO4K, or subsets of them. COACH420 is comprised by a set of 420 single-chain structures binding a mix of drug-like molecules and naturally occurring ligands which is disjunct with the CHEN11 and JOINED datasets. COACH420 is a modified version of the original COACH test set \cite{ROY_2012_COFACTOR, YANG_2013_COFACTOR}. HOLO4K is a larger set, $N \approx$ 4,000, based on the list by Schmidtke \textit{et al.} \cite{SCHMIDTKE_2010_FPOCKET}, which includes a mix of single- and multi-chain complexes, also disjunct with P2Rank training (CHEN11) and validation (JOINED) datasets. VN-EGNN, DeepPocket and GrASP use the Mlig and Mlig+ subsets of the COACH and HOLO4K datasets, which comprise strictly biologically relevant ligands as defined by the binding MOAD. IF-SitePred is tested on the HOLO4K-AlphaFold2 Paired (HAP) and HAP-small sets. HAP is a subset of the HOLO4K dataset which presents high quality models in the AlphaFold database \cite{VARADI_2022_ALPHAFOLDDB}. HAP-small is a smaller subset of HAP that only contains proteins with sequence identity lower than 25\% to proteins in the P2Rank training set. VN-EGNN uses the refined version of PDBbind (v2020), referred here as PDBbind\textsubscript{REF}, \cite{WANG_2004_PDBBIND, WANG_2005_PDBBIND, CHENG_2009_PDBBIND, LI_2014_PDBBIND, LIU_2015_PDBBIND, LIU_2017_PDBBIND} as a third test set. Like binding MOAD, the PDBbind database provides a comprehensive collection of experimentally measured binding affinity data for macromolecular complexes. Specifically, the refined set includes those protein-ligand complexes for which binding data was obtained with the literature and met certain experimental quality thresholds. Lastly, SC6K is a dataset presented by Aggarwal \textit{et al.} \cite{AGGARWAL_2022_DEEPPOCKET} containing 6,000 protein-ligand pairs from PDB entries submitted from 01/01/2018 – 28/02/2020. 

\subsection{Protein chain alignment}

For each protein chain, atomic coordinates were translated to be centred at the origin, $O = (0, 0, 0)$, and rotated using a rotation matrix, $R$. The two principal components of the coordinate space $pc_{1}$ and $pc_{2}$ were obtained using principal component analysis (PCA) \cite{HOTELLING_1933_PCA}. A third component, $pc_{\perp}$, was obtained with the cross-product of the other two, to ensure orthogonality. A rotation matrix $P$ is constructed from these vectors (Equation \ref{eq:pca_components}). By placing the main component $pc_{1}$ on the second row of $P$, we ensure the $Y$ axis will be the major axis, representing the height of the protein chain. The second largest axis will be the $X$ axis, representing the width of the protein, and lastly the depth will be represented by the smaller magnitude of the $Z$ axis. The final rotation matrix $R$ is obtained by multiplying $P$ by the negative identity matrix $NI$ (Equation \ref{eq:NI_R_matrices}). This was done to maintain the left-handedness of the protein chains whilst ensuring a consistent alignment on the major axes.

\begin{equation}
pc_{\perp} = pc_{1} \times pc_{2} \quad \rightarrow \quad P = \begin{bmatrix}
pc_{2} \\
pc_{1} \\
pc_{\perp}
\end{bmatrix}
\label{eq:pca_components}
\end{equation}

\begin{equation}
NI = -1 \cdot I_3 = -1 \cdot \begin{bmatrix}
1 & 0 & 0 \\
0 & 1 & 0 \\
0 & 0 & 1 
\end{bmatrix} = \begin{bmatrix}
-1 & 0 & 0 \\
0 & -1 & 0 \\
0 & 0 & -1 
\end{bmatrix} \quad \rightarrow \quad R = P \cdot NI
\label{eq:NI_R_matrices}
\end{equation}

\subsection{Protein chain characterisation}

For a protein chain with $N$ amino acid residues, the centre of mass, $CM$, was calculated by averaging the coordinates, $r_{i}$, of all atoms (Equation \ref{eq:centre_of_mass}), and from it, the radius of gyration, $R_{g}$, was derived (Equation \ref{eq:radius_of_gyration}) \cite{FIXMAN_1962_ROG}. As the protein chains are already aligned on the axis and centred on $O = (0, 0, 0)$ the dimensions of the protein chain can be obtained as the magnitude of the PCA components or \textit{eigenvectors}, i.e., the \textit{eigenvalues}. The dimensions represent width, height, and depth for the $X$, $Y$ and $Z$ axes, respectively.

\begin{equation}
CM = \frac{1}{n} \sum_{i=1}^{n} r_i \rightarrow CM = O = (0,0,0)
\label{eq:centre_of_mass}
\end{equation}

\begin{equation}
R_g = \sqrt{\frac{1}{n} \sum_{i=1}^{n} (r_i - CM)^2} = \sqrt{\frac{1}{n} \sum_{i=1}^{n} (r_i - O)^2} \rightarrow R_g = \sqrt{\frac{1}{n} \sum_{i=1}^{n} r_i^2}
\label{eq:radius_of_gyration}
\end{equation}

Protein chain volumes were calculated using ProteinVolume \cite{CHEN_2015_PROTEINVOLUME}. A sphere enclosing the protein and centred on the protein centre of mass was obtained. The radius of this sphere is the maximum Euclidean distance between the protein atoms and the CM (Equation \ref{eq:radius_protein}). The volume of the sphere is calculated using Equation \ref{eq:volume_sphere}. Proteins were classified into four different groups based on their shape and size. Protein chains with $\leq$ 100 amino acids were classified as ``tiny''. Regarding the shape, protein chains were classified into ``elongated'' if their protein to sphere volume ratio $\leq$ 0.08 ($VR$) (Equation \ref{eq:volume_ratio}), i.e., the protein volume contains no more than 8\% of the sphere volume. This threshold was derived empirically by the visual inspection of all 3,448 protein chains on the LIGYSIS set. Otherwise, proteins were considered globular (Figure 11). In this manner, protein chains were classified into \textit{globular} ($N$ = 2,104; 61\%), \textit{elongated} ($N$ = 670; 19\%), \textit{elongated tiny} ($N$= 341; 10\%) and \textit{globular tiny} ($N$ = 333; 10\%).

\begin{equation}
R = \max \| r_i - CM \|
\label{eq:radius_protein}
\end{equation}

\begin{equation}
Volume_{Sphere} = \frac{4}{3} \pi R^3
\label{eq:volume_sphere}
\end{equation}

\begin{equation}
VR = \frac{Volume_{Protein}}{Volume_{Sphere}}
\label{eq:volume_ratio}
\end{equation}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.98\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG11_PROTEIN_SHAPE_APPROACH_OPT.pdf}
    \caption[Protein chain shape and size classification approach]{\textbf{Protein chain shape and size classification approach.} \textbf{(A)} The volume of the sphere enclosing the protein chain as well as the protein chain volumes are calculated, and their ratio obtained ($VR$). Globular proteins present more spherical shapes and therefore occupy a higher portion of the sphere volume, resulting in higher volume ratios. Non-globular, elongated or fibrous proteins on the other hand do not and present lower volume ratios. After extensive visual examination, a threshold was established at $VR$ = 0.08, and so proteins classified in these two groups. Proteins were classified as ``tiny'' if their chain was $\leq$ 100 amino acids; \textbf{(B)} Eight examples of each protein chain group to illustrate the outcome of the approach.}
    \label{fig:protein_class_approach}
\end{figure}

\subsection{Ligand binding site prediction}

For each segment in the LIGYSIS dataset, the representative chain as defined in the PDBe-KB was selected. Structures were cleaned using the \textit{clean\_pdb.py} script \cite{JUBB_2019_PDBTOOLS}. Eleven different ligand binding site prediction tools were employed to predict on the 3,448 representative chains: VN-EGNN \cite{SESTAK_2024_VNEGNN}, IF-SitePred \cite{CARBERY_2024_IFSP}, GrASP \cite{SMITH_2024_GrASP}, PUResNet \cite{KANDEL_2021_PURESNET, KANDEL_2024_PURESNET}, DeepPocket \cite{AGGARWAL_2022_DEEPPOCKET}, P2Rank \cite{KRIVAK_2015_P2RANK, KRIVAK_2018_P2RANK}, P2Rank\textsubscript{CONS} \cite{JENDELE_2019_PRANKWEB, JAKUBEC_2022_PRANKWEB}, fpocket \cite{GUILLOUX_2009_FPOCKET, SCHMIDTKE_2010_FPOCKET2}, PocketFinder\textsuperscript{+} \cite{AN_2005_POCKETFINDER}, Ligsite\textsuperscript{+} \cite{HENDLICH_1997_LIGSITE}, and Surfnet\textsuperscript{+} \cite{LASKOWSKI_1995_SURFNET}. Conservation scores for P2Rank were obtained from PrankWeb https://prankweb.cz/. Re-implementations of Capra \textit{et al.} \cite{CAPRA_2009_CONCAVITY} were used for PocketFinder, Ligsite and Surfnet, indicated by the ``+'' superscript. VN-EGNN, IF-SitePred, PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+} and Surfnet\textsuperscript{+} do not provide a list of residues for each pocket, but a list of centroids and their scores for the first two, and a list of grid points for each predicted pocket for the last three. For VN-EGNN, residues within 6\AA{} of the virtual nodes were considered pocket residues. For 429 predicted pockets ($\approx$3\%) no residues were found within this threshold. For IF-SitePred, residues within 6\AA{} of the clustered cloud points that resulted on a predicted pocket centroid were considered as pocket residues. Pocket residues were obtained in a similar manner for PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+} and Surfnet\textsuperscript{+}, by taking those residues within 6\AA{} of the pocket grid points. When running DeepPocket, the $-r$ threshold was removed and so all fpocket candidates were passed to the CNN-based segmentation module for pocket shape estimation. fpocket predictions re-scored by DeepPocket will be referred as DeepPocket\textsubscript{RESC}, whereas pockets extracted by the segmentation module of DeepPocket will be referred as DeepPocket\textsubscript{SEG}.

Seven of the considered methods provide residue \textit{ligandability} scores. P2Rank and P2Rank\textsubscript{CONS} report calibrated probabilities of residues being ligand-binding. Similarly, GrASP predicts the likelihood for any given heavy atom to be part of a binding site. A residue-level score was obtained for GrASP predictions by taking the maximum score of the residue atoms. For IF-SitePred, a residue ligandability score $LS$ can be obtained by averaging the 40 independently predicted probabilities of a residue being ligand-binding (Equation \ref{eq:IFSP_score}). Though calculated in a different way, these three scores range 0-1 and represent the likelihood of a residue binding a ligand and can therefore be compared. PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+}, and Surfnet\textsuperscript{+} also provide residue scores which maximum value can be $>$ 1.

\begin{equation}
LS = \frac{1}{40} \sum_{i=1}^{40} p_i
\label{eq:IFSP_score}
\end{equation}

The other three methods, i.e., VN-EGNN, PUResNet, DeepPocket, and fpocket do not report residue-level scores. However, binary labels represent whether a residue is part of a pocket (1) or not (0), in the same manner as all other methods.

Throughout this work the terms ``site'' and ``pocket'' are be used indistinctly. Across all figures, tables and legends, methods are sorted in chronological order.

\subsection{Binding site characterisation}

Radius of gyration, $R_{g}$, was calculated for pockets as it was done for whole protein chains (Equation \ref{eq:radius_of_gyration}). Distance between pockets was calculated as the Euclidean distance between their centroids and overlap between pocket residues with the Jaccard Index ($JI$), or intersection over union (IOU) (Equation \ref{eq:jaccard_index}) \cite{JACCARD_1901_INDEX, JACCARD_1912_INDEX}. POVME 2.0 was employed for pocket volume calculation \cite{DURRANT_2011_POVME, DURRANT_2014_POVME2, WAGNER_2017_POVME3}. A single inclusion region was used for each pocket. This region is defined by the smallest rectangular prism containing all pocket atoms. The prism is centred on the pocket centroid and its dimensions are determined by the distance between the two farthest atomic coordinates on each axis. No exclusion regions were used. Points outside the convex hull were deleted. A contiguous-points region was defined as a 5\AA{}-radius sphere on the pocket centroid (Figure 12).

\begin{equation}
JI(A, B) = \frac{|A \cap B|}{|A \cup B|}
\label{eq:jaccard_index}
\end{equation}

\begin{figure}[h]
    \centering
    \includegraphics[width=0.98\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG12_POCKET_VOLUME_APPROACH_OPT.pdf}
    \caption[Pocket volume calculation algorithm]{\textbf{Pocket volume calculation algorithm.} \textbf{(A)} PUResNet predicted pocket for PDB: 4PX2 (Jordan, S.R., Chmait, S., 2015). Pocket residues are coloured in blue and have their side chains displayed; \textbf{(B)} An inclusion region is determined using the coordinates of the pocket residue atoms; \textbf{(C)} POVME 2.0 calculates the shape of the pocket defined by the residues and contained within the inclusion region; \textbf{(D)} The pocket shape is defined by a series of unit volume (1\AA{}\textsuperscript{3}) spheres. The volume of the pocket is calculated as the addition of the sphere volumes or the number of spheres within the pocket. Structure visualisation with PyMOL v2.5.2 \cite{SCHRODINGER_2015_PYMOL}.}
    \label{fig:protein_volume_approach}
\end{figure}

\subsection{Prediction evaluation}

After predicting on the 3,448 chains of the LIGYSIS dataset, only chains where all residues across all predicted sites presented UniProt residue mapping were kept. This resulted in a final set of 2,775 protein chains which was used for the performance assessment of the methods.

The performance of ligand binding site prediction methods can be evaluated at two different levels: \textit{residue} level, and \textit{pocket} level. Prediction at the residue level involves the discrimination of those residues that are likely to interact with a ligand, whereas the aim of pocket-level prediction is to define distinct regions on a protein, i.e., pockets where a ligand is likely to bind. This region can either be defined by a (pocket) centroid, a group of cloud/grid points, a set of pocket residues, or a combination of these. Some methods are residue-centric, and first predict at the residue-level, use a threshold to select high-probability ligand-binding residues, and then cluster them into pockets. \textit{Residue}-centric methods include IF-SitePred, or GrASP. Other methods (\textit{pocket}-centric) directly predict the location or shape of the pocket, without the need of predicting at the residue level first. Some of these methods can use their pocket-level prediction to report residue ligandability scores, e.g., P2Rank\textsubscript{CONS}, P2Rank, PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+} or Surfnet\textsuperscript{+}, and others, such as VN-EGNN, PUResNet, DeepPocket, or fpocket do not report residue ligandability scores.

\subsubsection{Residue-level predictions}

GrASP, P2RANK\textsubscript{CONS}, P2Rank, PocketFinder\textsuperscript{+}, Ligsite\textsuperscript{+} and Surfnet\textsuperscript{+} all offer residue ligandability scores. Additionally, a ligandability score was derived for IF-SitePred using Equation \ref{eq:IFSP_score}. Given a ligandability threshold $t_{LS}$, a residue with a ligandability score $LS_{i}$ is classified as ``positive'' if $LS_{i} > t_{LS}$. Conversely, the residue is classed as ``negative'' if $LS_{i} \leq t_{LS}$. Further stratification results from comparing the predictions to the LIGYSIS reference dataset.

\begin{itemize}
\item True Positive (TP): residue classified as positive that binds a ligand according to the reference.
\item False Positive (FP): residue classified as positive that does not bind a ligand in the reference.
\item True Negative (TN): residue classified as negative that does not bind a ligand.
\item False Negative (FN): residue classified as negative but is known to bind a ligand according to the data.
\end{itemize}

With these four classes, true positive rate (TPR) (Equation \ref{eq:TPR}, false positive rate (FPR) (Equation \ref{eq:FPR}), precision (Equation \ref{eq:precision}) and recall (Equation \ref{eq:recall}) can be calculated and receiver operating characteristic (ROC), precision-recall (PR) curves plotted. Area under the curve (AUC) for ROC and average precision (AP) for PR are reported. Baselines for these are 0.5 and the proportion of true positive labels, respectively.

\begin{equation}
TPR = \frac{TP}{TP + FN}
\label{eq:TPR}
\end{equation}

\begin{equation}
FPR = \frac{FP}{FP + TN}
\label{eq:FPR}
\end{equation}

\begin{equation}
Precision = \frac{TP}{TP + FP}
\label{eq:precision}
\end{equation}

\begin{equation}
\text{Recall} = \frac{\text{TP}}{\text{TP} + \text{FN}}
\label{eq:recall}
\end{equation}

Pocket binary labels (0: no pocket residue; 1: pocket residue) can determine TP, FP, TN and FN for each residue in a protein chain $P$. An F1 score is computed across all residues of $P$, which combines precision and recall into a unified metric, capturing the accuracy and completeness of predictions at the residue level (Equation \ref{eq:F1_score}). The Matthews Correlation Coefficient (MCC) \cite{MATTHEWS_1975_MCC} (Equation \ref{eq:MCC}) was also calculated. The median F1 score and MCC across the dataset proteins is reported for each method.

\begin{equation}
F1 = \frac{2 \times \text{Precision} \times \text{Recall}}{\text{Precision} + \text{Recall}}
\label{eq:F1_score}
\end{equation}

\begin{equation}
MCC = \frac{TP \times TN - FP \times FN}{\sqrt{(TP + FP)(TP + FN)(TN + FP)(TN + FN)}}
\label{eq:MCC}
\end{equation}

\subsubsection{Pocket-level predictions}

There are no \textit{negatives} predicted at the pocket level of ligand binding site prediction, only \textit{positives}. A positive is a predicted pocket, which will be true (TP) or false (FP) depending on whether it is observed in the reference data. False negatives are those pockets observed in the reference data that are not predicted. They are the pockets the method fails to predict, and therefore, are not scored. A true negative would be a ``non-pocket'' that is not predicted. This can't be quantified easily and even if it was, it would not be scored by the method, as it is not predicted. For this reason, in this context, neither TPR, nor FPR can be calculated. Consequently, ROC/AUC can't be utilised to assess ligand binding site prediction at the pocket level. False negatives are known, but not scored, and therefore PR/AUC is not an option either. What can be calculated is the recall given a certain criterion. In this case, because of the nature of the LIGYSIS dataset, where defined sites result from the clustering of multiple ligands, the distance between the predicted pocket centroid to the observed binding site (DCC) was chosen as a criterion.

For each observed binding site in our reference dataset, the ``best'' prediction for each method is chosen. This is defined as the prediction with the minimum Euclidean distance to the observed pocket centroid or DCC. Once the observed-predicted pairs were obtained, only those with DCC $\leq$ 12 \AA{} were considered as correct predictions. A threshold of 12\AA{} was chosen as 4\AA{} is too strict a threshold when using DCC (See Supplementary Note 6 and Figures 10-14). A threshold of 4\AA{} works well for the distance to closest ligand atom (DCA) but does not for DCC. The top-$N$ and $N$+2 ranking predictions were considered to calculate success rate, or recall (Equation \ref{eq:success_rate}), and maximum recall was calculated by considering all predictions, regardless of their score or rank. $N$ represents the number of observed sites for a given protein.

\begin{equation}
\text{Success rate (recall)} = \frac{\text{observed sites with predicted site DCC} \leq 12 \text{Å}}{\text{observed sites}}
\label{eq:success_rate}
\end{equation}

Additionally, instead of conventional ROC, ROC100 \cite{WEBBER_2003_ROC100, SCOTT_2007_ROC100} can be used to measure the predictive performance of the methods. To do this, for each method, all predictions across dataset proteins were ranked based on the pocket score and cumulative true positives were plotted against cumulative false positives until 100 false positives are reached. In a similar way, a precision curve can be calculated by taking the top-$N$, in this case $N$ = 1,000, predictions, which measures how precision changes as more predictions with lower scores are considered. This is indicative of how informative pocket scores are.

To measure the similarity in shape and residue membership between the predicted and observed pockets, relative residue overlap ($RRO$) and relative volume overlap ($RVO$) were employed. For an observed-predicted pocket pair, RRO represents the proportion of observed ligand-binding residues ($R_{o}$) that are covered by the predicted pocket residues ($R_{p}$) (Equation \ref{eq:RRO}). The POVME output was utilised for the calculation of $RVO$ (Figure 13). POVME defines the volume of a pocket as a series of equidistantly spaced spheres of unit volume. As predictions by the different methods were on the same coordinate reference, these pocket volume spheres were already aligned, and the volume overlap was calculated simply as the proportion of spheres in the observed pocket ($V_{o}$) that overlap with the predicted pocket spheres ($V_{p}$) (Equation \ref{eq:RVO}).

\begin{equation}
RRO = \frac{|R_p \cap R_o|}{R_o}
\label{eq:RRO}
\end{equation}

\begin{equation}
RVO = \frac{|V_p \cap V_o|}{V_o}
\label{eq:RVO}
\end{equation}

\begin{figure}[htbp!]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/ch_LBS_COMP/MAIN/PDF/FIG13_VOLUME_OVERLAP_APPROACH_OPT.pdf}
    \caption[Relative Volume Overlap ($RVO$) calculation]{\textbf{Relative Volume Overlap ($RVO$) calculation.} \textbf{(A)} Example of two very accurate predictions by PUResNet and P2Rank on PDB: 4PX2 (Jordan, S.R., Chmait, S., 2015). Pocket volumes are calculated with POVME 2.0 and represented by coloured surfaces. These volumes result from the addition of unit volume spheres on a grid. To obtain the $RVO$, the intersection of these spheres between predicted and observed site is divided by the number of observed pocket spheres. Both predictions cover the entirety of the observed pocket volume; \textbf{(B)} GrASP and VN-EGNN predictions of a site on PDB: 2ZOX \cite{NOGUCHI_2008_STRUCTURE}. Th volumes of these predicted sites overlap less with the observed site: $RVO$ = 0.67 for GrASP and $RVO$ = 0.11 for VN-EGNN.}
    \label{fig:protein_volume_approach}
\end{figure}

\subsection{Statistics and reproducibility}

VN-EGNN was installed and run locally from https://github.com/ml-jku/vnegnn. Likewise, for IF-SitePred: https://github.com/annacarbery/binding-sites. GrASP was obtained from https://github.com/tiwarylab/GrASP and predictions generated using their Google Colab Notebook. PUResNet predictions were obtained through the PUResNet v2.0 web server: https://nsclbio.jbnu.ac.kr/tools/jmol. DeepPocket was installed and executed locally: https://github.com/devalab/DeepPocket. P2Rank v2.4.2 was used to run all predictions as well as PRANK re-scoring: https://github.com/rdk/p2rank. fpocket v4.0 was installed via Conda: https://anaconda.org/conda-forge/fpocket. For PocketFinder, Ligsite and Surfnet, the ConCavity v0.1 “+” re-implementations were employed: https://compbio.cs.princeton.edu/concavity/.

Other recent methods including SiteRadar [49], NodeCoder [48], GLPocket [50], RefinePocket [137], EquiPocket [51], RecurPocket [46], PointSite [45], DeepSurf [44], Kalasanty [42], GRaSP [40], BiteNet [41] or DeepSite [38] were not included in this analysis due to technical reasons. Open-source methods with publicly accessible code, clear installation instructions, well defined dependencies, and accessible command line interfaces were prioritised in this work.

ChimeraX v1.7.1 [138] was used for structural visualisation in all figures unless otherwise stated, in which case PyMOL v2.5.2 was employed [59].


\subsection{Data availability}

The main results tables and files necessary to replicate the analysis described in this paper can be found here: https://doi.org/10.5281/zenodo.13121414 .

\subsection{Code availability}

Software developed to carry out this analysis is found in this GitHub repository: https://github.com/bartongroup/LBS-comparison.



\section{Results}