\chapter{LIGYSIS-web: a resource for the analysis of protein-ligand binding sites}
\label{chap:LIGYSIS_WEB}

\section*{Preface}

This Chapter refines and extends the ligand binding site definition and chatacterisation approaches employed in \autoref{chap:FRAGSYS} to the entire PDB. LIGYSIS is a ligand site analysis dataset comprising biologically relevant protein-ligand interactions from 30,000 proteins with experimentally determined structures across species. The LIGYSIS web server is presented, LIGYSIS-web, a resource to explore the LIGYSIS database of ligand binding sites as well as to analyse custom user structure set and visualise them in a interactive dynamic way.

\section*{Publications}

Utgés, J.S., MacGowan, S.M., Barton, G.J. LIGYSIS-web: a resource for the analysis of protein-ligand binding sites. (\textit{Manuscript in preparation})

\section*{Author contributions}

J.S.U., S.A.M. and G.J.B. conceived, designed, and developed the research. J.S.U. analysed the data. J.S.U. and S.A.M. developed the software. J.S.U., S.A.M and G.J.B. wrote, reviewed and edited the manuscript. G.J.B. secured funding and supervised.

\section{Introduction}

Ligands play a critical role in protein function acting as natural co-factors, substrates, inhibitors and drugs in disease therapy. Identifying the protein regions where these molecules bind, understanding the mode in which they do so and characterising that interface is therefore key to understanding and modulating protein function. The UniProt knowledgebase (UPKB) catalogues 248 million protein sequences \cite{UNIPROT_2018_UNIPROT, UNIPROT_2023_UNIPROT}. While structure models for most of these proteins are available through resources such as the AlphaFold Database \cite{JUMPER_2021_ALPHAFOLD, VARADI_2022_ALPHAFOLDDB, ABRAMSON_2024_ALPHAFOLD3} and other providers \cite{GUEX_2009_SWISSMODEL, BEIENERT_2016_SWISSMODEL, WATERHOUSE_2018_SWISSMODEL}, only a small fraction present residue-level functional annotations in UniProt – 55 thousand (0.02\% of UPKB) or include biologically relevant ligands co-crystallised in the Protein Data Bank Europe (PDBe) \cite{BERMAN_2003_PDB, wwPDB_2019_PDB} – 29 thousand (0.01\%). The significant expense and time required for experimental validation underscores an urgent need for computational methods to characterise ligand sites  systematically and highlight residues likely to be relevant to protein function.

In \autoref{chap:FRAGSYS} a novel method was proposed for defining ligand binding sites from multiple protein-ligand complexes derived from fragment screening. Fragment screening is a widely used technique for identifying lead molecules in drug discovery \cite{CONGREVE_2003_RO3, REES_2004_FBLD, MURRAY_2009_FBDD}. Unlike traditional approaches, which rely on spatial ligand positioning \cite{SHIN_2005_PDBLIGAND, KOZAKOV_2005_CLUSTERING, WASS_2010_3DLIGANDSITE, MCGREIG_2022_3DLIGANDSITE}, Utgés, \textit{et al.} \cite{UTGES_2024_FRAGSYS} used protein-ligand interactions to group ligands into binding sites. Sites were categorised by solvent accessibility into four clusters, each showing differential enrichment in known functional sites. These clusters provide a basis for ranking sites by likelihood of functional significance. This method was subsequently applied to the entire PDBe, resulting in the LIGYSIS dataset, which Utgés and Barton \cite{UTGES_2024_LBSCOMP} employed to benchmark protein-ligand binding site prediction methods. This will be discussed in detail in \autoref{chap:LBS_COMP} and \autoref{chap:LBS_IMPROV}.

This Chapter introduces the ligand site analysis web server \textit{LIGYSIS-web}, a resource for the analysis of protein-ligand binding sites \autoref{fig:LIGYSIS_WEB}. LIGYSIS-web hosts the LIGYSIS dataset, comprising 64,782 ligand binding sites, defined from 435,038 biologically relevant ligands, across 25,003 proteins with protein-ligand complexes deposited in the PDBe. Additionally, users can submit their own structures to the LIGYSIS web server for analysis, visualise the results in a dynamic and interactive manner and download the results for further analysis.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LIGYSIS_WEB/PNG/LIGYSIS_WEB_GRAPHICAL_ABSTRACT.png}
    \caption[LIGYSIS-web]{\textbf{LIGYSIS-web.} LIGYSIS-web is a resource for the analysis of protein-ligand binding sites. Users can query the pre-computed LIGYSIS database of 64,782 protein-ligand binding sites across 25,003 proteins in UniProt, or submit their own set of protein-ligand complexes for analysis. Results can then be explored dynamically in the LIGYSIS web server, a web application. These results include protein-ligand contacts information, evolutionary divergence scores from multiple sequence alignments of homologous proteins, human missense enrichment scores and solvent accessibility providing an integrative view of the likelihood of function of the defined binding sites and individual residues that form them.}
    \label{fig:LIGYSIS_WEB}
\end{figure}

\section{Methods}

\subsection{Derivation of the LIGYSIS dataset}

There are 248 million proteins in the UniProtKB database \cite{UNIPROT_2020_UNIPROT}, $\approx$65,000 of which (0.02\%) of which have at least one experimentally determined three-dimensional structure deposited in the Protein Data Bank (PDB) \cite{ARMSTRONG_2020_PDBE}. 28,997 of these (45\%; 0.01\% of UPKB) proteins present at least one structure in complex with a biologically relevant ligand as defined by BioLiP \cite{YANG_2013_BIOLIP}, represented by 29,657 different \textit{structural segments}, as defined on the PDBe-KB. These segments represent a UniProt sequence region with one or more structurally overlapping chains  \cite{ELLAWAY_2024_CONFORMATIONS}. A protein might present multiple domains which are usually represented by different structural segments. Protein chains mapping to a given UniProt accession identifier were obtained from the PDBe aggregated API endpoint: \url{https://www.ebi.ac.uk/pdbe/graph-api/uniprot/superposition/} \cite{PDBEKB_2019_PDBEKB}. Transformation matrices to superpose protein chains for a structural segment were downloaded from the PDBe FTP site at \url{http://ftp.ebi.ac.uk/pub/databases/pdbe-kb/superposition} \cite{PDBE_2022_PDBEKB} and used to structurally align all chains mapping to each segment.

Preferred biological assemblies, as defined by PISA \cite{KRISSINEL_2007_PISA}, were downloaded from PDBe via ProIntVar \cite{MACGOWAN_2020_DRSASP}. Protein-ligand contacts were determined with pdbe-arpeggio \cite{JUBB_2017_ARPEGGIO}. \autoref{fig:LIGYSIS_method} illustrates the ligand site definition approach used to obtain the new dataset presented here: LIGYSIS. This approach is an extension of the one used in \autoref{chap:FRAGSYS}. For a pair of ligands, $L_A$, $L_B$, fingerprints $A$, $B$ are defined as sets containing the UniProt residue numbers of the amino acids interacting with each ligand. PDB residues were cross-referenced to UniProt using the SIFTS mapping present in the macromolecular crystallographic information file (mmCIF) files, located under the \texttt{\_atom\_site.pdbx\_si\discretionary{-}{}{}fts\_xref\_db} fields \cite{VELANKAR_2012_SIFTS, DANA_2018_SIFTS}. Relative intersection, $I_{rel}$, (\autoref{eq:relative_intersection}) is a similarity metric that quantifies how similar these fingerprints are \cite{UTGES_2024_FRAGSYS}. Subtracting $I_{rel}$ from 1 gives a distance, $D$ (\autoref{eq:Irel_distance}), which takes the value of 0 when $A$ and $B$ share all the binding residues and 1 when they share none. For a given protein segment, interacting with $M$ biologically meaningful ligands across $N$ chains, ligand fingerprints were clustered using average linkage with SciPy \cite{VIRTANEN_2020_SCIPY} and ligand sites obtained by cutting the tree at $D$ = 0.5. Note that pdbe-arpeggio currently does not support multiple occupancy atoms, and so ligands with atom occupancy $\neq$ 1 were not considered in our analysis. This affects 6256 structures (6\% of LIGYSIS dataset). 64,782 sites were defined from 435,038 ligands across 26,260 structural segments. These segments mapped to 25,003 different proteins, which were represented by 104,456 structures.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LIGYSIS_WEB/PNG/FIG1_LIGYSIS_APPROACH.png}
    \caption[LIGYSIS ligand binding site definition algorithm]{\textbf{LIGYSIS ligand binding site definition algorithm.} For a given protein-ligand interaction complex \textbf{(A)}, a ligand binding fingerprint \textbf{(B)} is obtained as the set of unique UniProt sequence residue numbers interacting with the ligand as defined by pdbe-arpeggio; \textbf{(C)} Fingerprints from different ligands binding to the same protein can be compared and a distance calculated; \textbf{(D)} This distance can be used to perform hierarchical clustering, which groups the different ligands in distinct clusters binding to the same region of the protein or binding site \textbf{(E)}. Example is human arginase-2, mitochondrial (\href{https://www.uniprot.org/uniprotkb/P78540/entry}{P78540}) represented by PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/4ixv}{4IXV} \cite{GOLEBIOWSKI_2013_ARGINASE} with \href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/XA1}{XA1} bound.}
    \label{fig:LIGYSIS_method}
\end{figure}

\vspace{-12pt} % Adjust this value as needed
\vspace{-12pt} % Adjust this value as needed

\begin{equation}
D = 1 - I_{rel}
\label{eq:Irel_distance}
\end{equation}
\myequations{Ligand fingerprint distance}

\vspace{-12pt} % Adjust this value as needed
\vspace{-12pt} % Adjust this value as needed
\vspace{-12pt} % Adjust this value as needed
\vspace{-12pt} % Adjust this value as needed

\subsection{Alignments and variants}

For each UniProt accession, the canonical sequence was used to perform a homologue sequence search in SwissProt \cite{BOUTET_2016_UNIPROT}. jackHMMER \cite{EDDY_1995_HMMER} was employed with three iterations to generate a multiple sequence alignment (MSA). Amino acid divergence was quantified with the normalised Shenkin divergence score \cite{SHENKIN_1991_SCORE} $N_{Shenkin}$ \cite{UTGES_2021_ANKS}. Genetic missense variants mapping to human sequences in the MSA were retrieved from gnomAD \cite{KARCZEWSKI_2020_GNOMAD} using VarAlign \cite{MACGOWAN_2017_VARIANTS}. Missense enrichment scores (MES), i.e., odds ratio (OR), were calculated for alignment columns \cite{MACGOWAN_2024_VARIANTS} and 95\% confidence intervals and $p$-values used to evaluate their significance \cite{SZUMILAS_2010_ODDSRATIOS}.

\subsection{RSA-based clustering ans score}

Accessible surface area was calculated by DSSP \cite{KABSCH_1983_DSSP} via ProIntVar \cite{MACGOWAN_2020_DRSASP} and normalised \cite{TIEN_2013_RSA} to relative solvent accessibility (RSA). A Keras \cite{CHOLLET_2015_KERAS} multilayer perceptron (MLP) \cite{CYBENKO_1989_MLP} described in \autoref{chap:FRAGSYS} \cite{UTGES_2024_FRAGSYS} was employed to predict site RSA-based cluster labels: C1 – C4. These clusters are differentially enriched in functional sites annotated in UniProt \cite{NIGHTINGALE_2017_API} (OR\textsubscript{C1} $\approx$ 28 $\times$ OR\textsubscript{C4}).

For a binding site $i$, a functional score $FS$ is calculated with \autoref{eq:func_score}. In this equation, $p_{i_{j}}$ represents the probability of site $i$ belonging in Cluster $j$ and $f_j$ denotes the proportion of annotated functional sites within Cluster $j$. The probabilities $p_{i_{j}}$ are derived from the vector $P$ (\autoref{eq:probability_vector}), which is returned by the MLP and provides the probabilities of a site belonging to each class. The vector $F$ (\autoref{eq:func_prob_vector}) contains the proportions of functional sites in each cluster, which were determined through hierarchical clustering and functional classification of the human subset of the LIGYSIS dataset, composed of 13,000 sites across 3500 proteins. Both the functional label and score serve as metrics that indicate the likelihood of a binding site being functional and can be used to rank binding sites within a protein.

\begin{equation}
FS_i = \sum_{j=1}^{4} p_{i_{j}} f_j
\label{eq:func_score}
\end{equation}
\myequations{Functional score}

\vspace{-12pt} % Adjust this value as needed
\vspace{-12pt} % Adjust this value as needed

\begin{equation}
P = [\,p_{c_1},\,p_{c_2},\,p_{c_3},\,p_{c_4}\,]
\label{eq:probability_vector}
\end{equation}
\myequations{RSA cluster probability vector}

\vspace{-12pt} % Adjust this value as needed
\vspace{-12pt} % Adjust this value as needed

\begin{equation}
F = [\,f_{1},\,f_{2},\,f_{3},\,f_{4}\,] 
    = [\,0.52,\,0.18,\,0.05,\,0.04\,]
\label{eq:func_prob_vector}
\end{equation}
\myequations{Functional enrichment vector}

\vspace{-12pt} % Adjust this value as needed
\vspace{-12pt} % Adjust this value as needed

\subsection{LIGYSIS customised pipeline}

The LIGYSIS customised pipeline is used for the analysis of user-submitted structures to the server. The customised pipeline does not rely on the PDBe-KB for mapping chains to a protein, nor transformation matrices. Instead, STAMP \cite{RUSSELL_1992_STAMP} is used to superpose the uploaded structures, which can be in PDB (\textit{.ent}, \textit{.pdb}) or mmCIF \textit{.cif}  format. However, all structures in the set must be in the same format, e.g., either all PDB or all mmCIF. Structures must present the same number of protein chains, either all monomers, dimers, trimers, etc. The LIGYSIS web server currently supports only homomeric protein-ligand complexes, i.e., complexes between $N$ copies of a single protein sequence and any number of ligands. For structures mapping to a protein in UniProt, it is required to submit the corresponding UniProt identifier, so residues across different structures can all be mapped to the same reference. If the submitted structures represent a protein not in UniProt, this field can be left blank. The residue numbering employed will be the structure one, so structures are expected to present the same numbering scheme.

%\subsection{Site function classification}

%UniProt features were retrieved using the UniProt proteins API \cite{NIGHTINGALE_2017_API} on the features endpoint: \url{https://www.ebi.ac.uk/proteins/api/features/}. \textbf{XXX} of the \textbf{XXX} (92.5\%) proteins presented UniProt features. Only features of the category ``DOMAINS\_AND\_SITES'' and the types ``BINDING'', ``SITE'', and ``ACT\_SITE'' were used for functional assignment of a protein, which resulted in \textbf{XXX} (\textbf{XXX\%}) proteins with such annotations. Binding sites presenting at least a functionally annotated residue were classified as known function (KF) ($N$ = \textbf{XXX}), and those with no annotations as unknown function (UF) ($N$ = \textbf{XXX}).

\subsection{Data Availability}

The code for the LIGYSIS-web Python Flask application can be found in our GitHub repository: \url{https://github.com/bartongroup/LIGYSIS-web} \cite{UTGES_2024_LIGYSIS_WEB_ZENODO}. The code for the LIGYSIS pipeline, employed to generate the LIGYSIS dataset, which LIGYSIS-web explores can be found here: \url{https://github.com/bartongroup/LIGYSIS} \cite{UTGES_2024_LIGYSIS_ZENODO}. The code for the LIGYSIS pipeline adapted to handle user-submitted jobs can be found in our repository: \url{https://github.com/bartongroup/LIGYSIS-custom} \cite{UTGES_2024_LIGYSIS_CUSTOM_ZENODO}. Source code for Slivka-bio can be found on the project repository:  \url{https://github.com/bartongroup/slivka-bio} \cite{MACGOWAN_2020_DRSASP}. 


\section{LIGYSIS-web}

\subsection{LIGYSIS-web results page}

The LIGYSIS web server can be used in two modes: (1) to explore the LIGYSIS dataset and (2) to submit a set of structures for analysis to the LIGYSIS pipeline. The LIGYSIS dataset comprises $\approx$25,000 proteins with deposited structures of proteins bound to BioLiP biologically relevant ligands on the PDBe. Results can be explored by searching for a UniProt accession, entry, or name of the protein of interest. Results for user-submitted jobs can be accessed through the provided link when submitting the job, or through the jobs table at the user session tab.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LIGYSIS_WEB/PNG/LIGYSIS_WEB_RESULTS_PAGE_P00517.png}
    \caption[LIGYSIS-web results page]{\textbf{LIGYSIS-web results page.} The results page is divided in three panels: \textit{Binding Sites} (left), \textit{Structure} (centre), and \textit{Binding Residues} (right) panels. In this example, 103 ligands across 84 structures have been clustered into 8 different ligand binding sites for bovine cAMP-dependent protein kinase catalytic subunit alpha, PKA C-alpha, (\href{https://www.uniprot.org/uniprotkb/P00517/entry}{P00517}). This is the only segment of PKA C-alpha and covers its whole sequence (1-351). Chart.js scatter points and table rows represent binding sites and their average features on the Binding Sites Panel, whereas they represent individual amino acid residues on the Binding Residues Panel. Both panels interact with the 3Dmol.js central structure viewer through hover and click events. Below this viewer, function buttons can be found to hide/show slab controls, surfaces, labels, ligands, water molecules and protein-ligand contacts. Structures and contact data can also be downloaded from these buttons. Surfaces, ligands and water molecules are displayed in this screenshot. Superposition representative PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/1svh}{1SVH} \cite{BREITENLECHNER_2004_PKB}, chain: A.}
    \label{fig:LIGYSIS_web_results_page}
\end{figure}

\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.5\textwidth]{figures/ch_LIGYSIS_WEB/PNG/BINDING_SITES_PANEL_P00517.png}
    \caption[LIGYSIS-web results page Binding Sites Panel]{\textbf{LIGYSIS-web results page Binding Sites Panel.} UniProt accession identifier and entry name for cAMP-dependent protein kinase catalytic subunit alpha (\href{https://www.uniprot.org/uniprotkb/P00517/entry}{P00517}) (top). Below, Chart.js scatter of MES (log\textsubscript{10} scale) \textit{vs} divergence score for the 8 binding sites defined for PKA C-alpha. Dashed line indicates neutrality, i.e., missense variation within the site is no different than at the rest of the protein. Binding Site 2 (cyan) is an example of this. In contrast, Binding site 1 (pastel red), the known active site of the kinase, is depleted in missense variation (MES = 0.81). Additionally, it is buried (RSA = 12\%), conserved across homologues (DS = 27/100) and presents a high functional score (FS = 0.52). Binding sites with low divergence and missense enrichment scores (lower-left quadrant) are most likely to be functional, whereas those at the top-right are least likely to have biological effect when bound a ligand, e.g., BS6 (purple). Binding sites table is sorted by functional score (FS). Binding Site 0 is clicked (green highlight) and Binding Site 5 is being hovered on (yellow highlight).}
    \label{fig:LIGYSIS_web_BINDING_SITES_PANEL}
\end{figure}

\autoref{fig:LIGYSIS_web_results_page} illustrates the LIGYSIS-web main results page. This page is divided into three panels, from left to right: \textit{Binding Sites}, \textit{Structure}, and \textit{Binding Residues} panels. The results page is the same when exploring a LIGYSIS entry and user job results. At the top of the panels, general information about the target protein can be found, from left to right: UniProt accession identifier (link to UniProt), entry, protein names and number of chains, ligands and defined binding sites for the target protein.

\subsubsection{Binding Sites panel}

\autoref{fig:LIGYSIS_web_BINDING_SITES_PANEL} depicts the Binding Sites panel of the results page. This panel is formed by a dynamic Chart.js canvas and a table. Both are displaying the mean binding site features, calculated from averaging the features of the residues forming the site. These features are Relative Solvent Accessibility (RSA), Normalised Shenkin Divergence Score (DS), Missense Enrichment Score (MES), the size of the site, i.e., number of amino acid residues, the RSA-derived Cluster label and its associated Functional Score (FS). The table rows can be sorted by any of these variables and the axes of the chart dynamically changed. MES uses logarithmic scale, as it is an odds ratio. The chart as well as the table are linked to the structure viewer by hover and click events. Hovering on a data point/row will temporarily display the side chains of the site residues, whereas clicking on it would fix them in the view until another site is clicked or the current one unclicked. Refer to \autoref{fig:LIGYSIS_web_BINDING_SITES_PANEL} legend for more details on how to interpret this graph.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.5\textwidth]{figures/ch_LIGYSIS_WEB/PNG/STRUCTURE_PANEL_P00517.png}
    \caption[LIGYSIS-web results page Structure Panel]{\textbf{LIGYSIS-web results page Structure panel.} Structure Panel of the LIGYSIS-web results page. Preferred biological assembly of PKA C-alpha (\href{https://www.uniprot.org/uniprotkb/P00517/entry}{P00517}) PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/1Q24}{1Q24} \cite{GAEL_2003_AKT} with displayed interactions with adenosine tri-phosphate (\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/ATP}{ATP}). Water molecules and ligand-binding residue labels are displayed. Interaction label between NZ atom of LYS73 chain: A and O2B of \href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/ATP}{ATP} 400 chain: D is displayed on hover. This is a hydrogen bond and the distance between the atoms is of 3.16 \AA{}. Residues in contact with multiple ligand atoms, as LYS73, which forms hydrogen bond and ionic interactions with three atoms, or ASP185, which interacts with four \href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/ATP}{ATP} atoms, are likely to be more relevant for the binding mode than residues with a single contribution to the binding interface, as ALA71. Non-carbon atoms are coloured based on the Jmol colouring scheme \cite{JMOL}.}
    \label{fig:LIGYSIS_web_STRUCTURE_PANEL}
\end{figure}

\subsubsection{Structure panel}

\autoref{fig:LIGYSIS_web_STRUCTURE_PANEL} depicts the central Structure panel of the results page. This panel is divided in three parts: the main 3Dmol.js viewer, in the centre, segment and structure selectors above it, and the structure buttons below it. The Segment selector  is a drop-down menu showing the segment identifier and its protein sequence coordinates. The Structure selector is a drop-up menu that allows the user to swap between the Superposition view (default) and individual biological assemblies. The Superposition view consists of a white cartoon representation of a segment representative chain and the superposed ligands coloured by binding site. These are hidden to start with and can be displayed by clicking on the ``LIGAND'' button and hidden if clicked again. The ``SURF'' button will display the protein chain surface, which will be white for non-ligand binding residues and coloured for binding residues. The same applies to water molecules and the ``HOH'' button. To see the labels of a clicked binding site, click on the ``LABEL'' button, and click again to hide them. The ``CONTACT'' button is disabled for the superposition view but can be clicked when exploring an assembly or structure. The relevant ligands, as well as the residues interacting with them will be coloured based on their site, and dashed cylinders depicting the protein-ligand interactions calculated by pdbe-arpeggio will be displayed. They are coloured based on the Arpeggio colour scheme, e.g., green for hydrophobic, red for polar, or yellow for ionic. The width of these cylinders is representative of the distance between the atoms. Thicker cylinders denote a clash between the Van der Waals (VdW) radii of the atoms (closer) whilst thinner cylinders indicate interaction between the VdW radii of the atoms. The view can also be sliced between two planes to focus on a region of interest. The slab or slice controls are displayed with the ``SLICE'' button. Hiding them does not reset the slab, just hides the controls. Click on button again to keep changing the slab. Clicking on the circular arrow generates a spin animation on the Y axis. Clicking on any ligand atom on the structure viewer links to the new PDBe-KB Ligand Pages \cite{CHOUDHARY_2024_PDBETOOLS} providing enhanced ligand annotations and a holistic view of small molecules for their biological context.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=0.6\textwidth]{figures/ch_LIGYSIS_WEB/PNG/BINDING_RESIDUES_PANEL_P00517.png}
    \caption[LIGYSIS-web results page Binding Residues panel]{\textbf{LIGYSIS-web results page Binding Residues panel.} Missense Enrichment Score (MES) \textit{vs} Normalised Shenkin Divergence Score (DS) for the 51 residues of Binding Site 0 of PKA C-alpha (\href{https://www.uniprot.org/uniprotkb/P00517/entry}{P00517}). This site corresponds to the ATP binding site of PKA C-alpha. Most residues on this site present MES $<$ 1 (below dashed line), indicating depletion in human missense variation resulting from selective pressure or constraint. Missense-depleted residues that are also conserved are most likely to be functional and cause an effect on the protein function if targeted. An example is Lysine 169, which is buried (RSA = 6\%), conserved across homologues (DS = 10) and depleted in missense variation (MES = 0.44, $p \approx$ 0). This residue is known to be functionally relevant as it interacts with \href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/ATP}{ATP} which agrees with the results displayed on the table. Residues are sorted by missense enrichment on the table (lowest at the top) and LYS169 which is the most depleted is being hovered on (yellow highlight).}
    \label{fig:LIGYSIS_web_BINDING_RESIDUES_PANEL}
\end{figure}

\subsubsection{Binding Residues panel}

The Binding Residues Panel (\autoref{fig:LIGYSIS_web_BINDING_RESIDUES_PANEL}) is similar to the Binding Sites Panel and comprises a chart and a table. However, data points and rows correspond to residues forming the selected site. The table displays the UniProt residue number (UPResNum), the column in the multiple sequence alignment (MSACol), the NShenkin divergence score (DS), MES, associated p-value, amino acid name (AA), RSA and secondary structure (SS). SS corresponds to the original 8-state DSSP classification: 3\textsubscript{10} helix (G), $\alpha$-helix (H), $\pi$-helix (I), $\beta$-sheet (E), $\beta$-bridge (B), helix turn (T), bend (S) and coil (C) \cite{KABSCH_1983_DSSP}. Variables on the chart axes can also be changed, and table rows sorted by column. Only hover events are supported for this panel, clicking chart points or table rows has no effect. Refer to \autoref{fig:LIGYSIS_web_BINDING_RESIDUES_PANEL} legend for more details on how to interpret this graph.

\subsection{Data export}

A ``Download Table'' button can be found on the Binding Site and Binding Residues panels. Clicking on this button will download the binding site or binding site residues tables, respectively, in .csv format. The multiple sequence alignment from which the divergence scores are derived can also be downloaded in Stockholm format (\textit{.sto}) by clicking on the ``Download MSA'' button. Additionally, (pdbe-arpeggio) protein-ligand contacts can be downloaded in tabular format by clicking on the ``Download Current Assembly Contacts'' and ``Download ALL Assemblies Contacts'' for the assembly currently being explored, or all of them. The first option is not available when exploring the Superposition view, since contacts are calculated on individual assemblies, and the second one will download a zipped folder of CSV (\textit{.csv}) files.

The ligand superposition view, i.e., representative chain with bound ligands across structures can be saved to ChimeraX (\textit{.cxc}) or PyMOL (\textit{.pml}) script files by clicking on ``Download Superposition'' and then selecting the preferred viewer. The same can be done for individual (``Download Current Assembly'') or all assemblies (``Download ALL Assemblies''). \autorefpanel{fig:supp_suport}{ A-C} exemplifies this with the ligand superposition for human PKA C-alpha and \autorefpanel{fig:supp_suport}{ D-F} illustrates contacts between PKA C-alpha and \href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/M77}{M77} on PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/1q8w}{1Q8W} \cite{BREITENLECHNER_2003_PKA}. Screenshots of both graphs as well as 3Dmol.js view can be saved to PNG (\textit{.png}) files by clicking on the camera icon on the corresponding panel.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LIGYSIS_WEB/PNG/supp_contacts_support.png}
    \caption[LIGYSIS-web supports ChimeraX and PyMOL]{\textbf{LIGYSIS-web supports ChimeraX and PyMOL.} LIGYSIS-web employs 3Dmol.js for visualisation. Additionally, it supports download of both superposition view as well as individual assemblies in the common structure viewers ChimeraX and PyMOL. In this superposition example \textbf{(A-C)}, PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/1svh}{1SVH} \cite{BREITENLECHNER_2004_PKB}, chain: A is the representative chain for cAMP-dependent protein kinase catalytic subunit alpha (\href{https://www.uniprot.org/uniprotkb/P00517/entry}{P00517}), for which 8 binding sites are defined from 103 ligands across 84 structures. Ligands are coloured by their binding site; PKA C-alpha interacting with \href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/M77}{M77} on PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/1q8w}{1Q8W} visualised with 3Dmol.js \textbf{(D)}, ChimeraX \textbf{(E)} and PyMOL \textbf{(F)}. Non-carbon atoms are coloured based on the Jmol colouring scheme \cite{JMOL} in \textbf{(D-F)}.}
    \label{fig:supp_suport}
\end{figure}

\subsection{Server architecture}

The LIGYSIS web server is implemented using the Python Flask framework \cite{GRINBERG_2018_FLASK}, with dynamic content rendered via Jinja \cite{JINJA} templates and data transferred through Flask routes. The frontend, built with JavaScript \cite{JAVASCRIPT}, features a 3Dmol.js \cite{REGO_2014_3DMOL, SESHADRI_2020_3DMOL} structure viewer for molecular visualisation, integrated with Chart.js \cite{CHARTJS} graphs and interactive tables. Styling is handled using Bootstrap \cite{BOOTSTRAP} and plain CSS \cite{HTML_CSS}. Asynchronous updates are facilitated by jQuery \cite{jQUERY} and AJAX \cite{AJAX} requests, enabling dynamic data exchange between the client and server.

User job submission is handled through Slivka-bio v0.8.3 \cite{MACGOWAN_2020_DRSASP, WAROWNY_2021_SLIVKA, SLIVKA_BIO} and jobs run on the School of Life Sciences, University of Dundee HPC infrastructure. Once the job execution has finished, result files are served to the client and displayed in the same way as the LIGYSIS database entries.

\section{LIGYSIS-web analysis of bovine rhodopsin}

G protein-coupled receptors (GPCR) comprise the largest protein receptor family in the human genome, with $\approx$800 members \cite{LIU_2024_GPCRs}. GPCRs have a regulatory role in most physiological processes including the visual, gustatory and smell sense, immune and nervous system activity as well as in disease \cite{LATORRACA_2017_GPCRs}. Consequently, GPCRs represent  an important target in drug therapy, covering $\approx$35\% of drugs approved by the Food and Drug Administration (FDA) \cite{HAUSER_2017_GPCRS, INSEL_2019_GPCRs}. GPCRs share a conserved seven-transmembrane helix fold connected by three intra- and three extra-cellular loops \cite{ZHANG_2024_GPCRs} and bind to a variety of endogenous ligands including peptides, ions, lipids or neurotransmitters \cite{ZARZYCKA_2019_GPCRs, MANNES_2022_GPCRs}. Once activated by external stimuli, e.g., ligand binding, GPCRs employ G-proteins to interact with downstream effectors thus triggering intracellular signalling cascades \cite{CHENG_2023_GPCRs}. To accommodate for the wide variety of substrates that GPCRs bind, specificity-determining positions (SDP) are required at the protein-ligand interface. These sites are unconserved across homologues and depleted in missense variation in human (UMD) \cite{MACGOWAN_2024_VARIANTS}.

\autoref{fig:ligysis_rhodopsin} illustrates how SDPs can easily be identified in LIGYSIS-web for the example of bovine rhodopsin (GPCR). \autorefpanel{fig:ligysis_rhodopsin}{ A} shows the average site missense enrichment \textit{vs} divergence for the 16 binding sites defined in LIGYSIS from 145 ligands of interest across 45 structures (\autorefpanel{fig:ligysis_rhodopsin}{ B}). Binding Site 0 corresponds to the orthosteric GPCR site and is on average unconserved (DS $>$ 50) and missense-depleted (MES $<$ 1). Four of the 38 residues within this site are significantly depleted in missense variation ($p <$ 0.05): GLU113, ALA117, SER186 and TYR288 (\autorefpanel{fig:ligysis_rhodopsin}{ C}). \autorefpanel{fig:ligysis_rhodopsin}{ D} exemplifies the specificity role of these positions by showing how different GPCRs bind to their substrates with different amino acids at these positions.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_LIGYSIS_WEB/PNG/LIGYSIS_use_case_NEW.png}
    \caption[LIGYSIS analysis of bovine rhodopsin]{\textbf{LIGYSIS analysis of bovine rhodopsin.} \textbf{(A)} Scatter of average site missense enrichment score (MES) \textit{vs}  divergence (DS) for the 16 ligand binding sites defined for bovine rhodopsin (\href{https://www.uniprot.org/uniprotkb/P06299/entry}{P06299}). Binding Site 0 is divergent (unconserved) (DS = 56) and depleted in missense variation (MES = 0.95); \textbf{(B)} Superposition view of 145 ligands of interest across 49 structures of rhodopsin visualised with 3Dmol.js;\textbf{(C)} MES \textit{vs} DS for the 38 residues forming Binding Site 0. Residues within the dashed rectangle are conserved across homologues and depleted in missense variation in human, indicating common function and constraint across members of the family. However, GLU113, ALA117, SER186 and TYR288 in bovine rhodopsin are divergent within the family yet constrained in human, suggesting a substrate specificity determining role for these residues; \textbf{(D)} Examples of multiple GPCRs binding to endogenous substrates (retinal and lysophosphatidic acid) and inverse agonist (carazolol). (I) Bovine rhodopsin (\href{https://www.uniprot.org/uniprotkb/P06299/entry}{P06299}) binding to retinal (\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/RET}{RET}) – PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/1jfp}{1JFP} \cite{YEAGLE_2001_GPCRs}. (II) Human lysophosphatidic acid receptor 1 (\href{https://www.uniprot.org/uniprotkb/Q92633/entry}{Q92633}) binding to oleoyl lysophosphatidic acid (\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/NKP}{NKP}) - PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/7TD0}{7TD0} \cite{LIU_2022_GPCRs}. (III) Turkey beta-1-adrenergic receptor (\href{https://www.uniprot.org/uniprotkb/P07700/entry}{P07700}) binding to carazolol (\href{https://www.ebi.ac.uk/pdbe-srv/pdbechem/chemicalCompound/show/CAU}{CAU}) – PDB: \href{https://www.ebi.ac.uk/pdbe/entry/pdb/2YCW}{2YCW} \cite{MOUKHAMETZIANOV_2011_GPCRs}. The same alignment position presents different amino acids across members of the family to accommodate for a variety of substrates. However, each of these residues is missense-depleted and forms ligand-specific interactions within each GPCR. Jmol colouring scheme \cite{JMOL} is applied to non-carbon atoms.}
    \label{fig:ligysis_rhodopsin}
\end{figure}

\section{Discussion}

The LIGYSIS web server is a free open resource accessible to all users without any login requirement for the analysis of protein-ligand binding sites. It hosts the LIGYSIS dataset, an integrative protein-ligand complex dataset including 65,000 biologically relevant binding sites across 25,000 proteins with structures on the PDBe. Additionally, users can upload their structures for analysis, visualisation and download. LIGYSIS defines binding sites by clustering protein-ligand interactions and characterises them by evolutionary divergence, missense variation and solvent accessibility, thus offering insight into the likelihood of function of sites as well as individual residues. These results are dynamically displayed on LIGYSIS-web, a Python Flask web application using Chart.js for dynamic graph rendering and 3Dmol.js for structure visualisation.

Future steps include implementing a search by ligand functionality, overcoming the current limitation of multi-occupancy ligand atoms, the implementation of heteromeric protein-ligand complexes or analysing predicted ligand binding sites by methods as P2Rank \cite{KRIVAK_2015_PRANK, KRIVAK_2015_P2RANK, KRIVAK_2018_P2RANK, JENDELE_2019_PRANKWEB, JAKUBEC_2022_PRANKWEB}, fpocket \cite{GUILLOUX_2009_FPOCKET, SCHMIDTKE_2010_FPOCKET2} or IF-SitePred \cite{CARBERY_2024_IFSP}.
