\chapter{Conclusions}

\section*{Preface}

This Chapter will bring together the main findings presented in the four previous Chapters of results of this Thesis and contextualise them with the current state of the art within the field. Additionally, it will also cover the direction in which this research is heading by describing the next steps that should be taken to take the work presented in this Thesis further.

\section{Introduction}

Proteins are the functional and structural blocks upon which life is built. They carry out or are involved in all biological processes that take place within cells. The amount, localisation, state and interaction with other molecules of proteins is strictly controlled by complex gene regulation networks, signalling cascades, and environment-dependent conformational changes. Any of these mechanisms failing to work correctly can result in under- or overexpression, hypo- or hypermorphism, truncated, mutated or unfolded proteins, amongst other issues, which eventually can lead to disease. Proteins do not carry their function in isolation, but usually interact with other proteins, nucleic acids, ions or small molecules. Moreover, chemical compounds can be used as drugs to modulate or inhibit protein function. Identifying or predicting the location of where these molecules bind to proteins and the way in which they do so is therefore critical to understand better and modulate protein function.

The work presented in this Thesis sheds light into the nature of ligand binding sites by developing methods to define and characterise ligand binding sites using protein sequence data from evolutionarily related homologues, missense genetic variation from the human species and structural features such as solvent accessibility \cite{UTGES_2024_FRAGSYS}. A pipeline is described to characterise all experimentally determined ligand binding sites deposited in the PDBe \cite{ARMSTRONG_2020_PDBE} resulting in the LIGYSIS dataset and a web resource presented for the exploration of 65,000 ligand binding sites across 25,000 proteins as well as for the analysis of custom user sets of protein-ligand complexes. Finally, the human component of the LIGYSIS set is employed to carry out the largest critical comparative assessment of ligand binding site prediction tools. This analysis objectively evaluates the performance of thirteen canonical methods and fifteen novel variants first introduced in this work using the LIGYSIS dataset as a benchmark and more than ten assessment metrics \cite{UTGES_2024_LBSCOMP}.

\section{Fragment screening sites analysis}

Fragment screening is a widely used technique to find lead molecules in early-stage drug discovery \cite{MURRAY_2009_FBDD}. It usually employs X-ray crystallography to obtain high-quality and throughput data about the binding mode of small molecule fragments to a target protein. \autoref{chap:FRAGSYS} describes novel methods for the definition and characterisation of 293 ligand binding sites derived from 37 publicly available fragment screening experiments \cite{UTGES_2024_FRAGSYS}. Unlike previous methods that rely on the structural superposition of protein chains or ligands \cite{SHIN_2005_PDBLIGAND, KOZAKOV_2005_CLUSTERING, WASS_2010_3DLIGANDSITE}, the method proposed in \autoref{chap:FRAGSYS} defines sites by grouping ligands based on their interactions with the protein. This approach avoids completely the structural alignment of protein chains, which can be difficult when dealing with a high number of protein chains or different conformational states for a given protein. The defined ligand sites therefore represent a location on the protein surface to which ligands with a similar interaction fingerprint bind.

Ligand binding sites were later grouped into four robust clusters (C1-C4) by their relative solvent accessibility profile. Vectorial representations of the RSA \cite{TIEN_2013_RSA} of each residue within the site were employed, as this captured better the solvent accessibility distribution of the site than a simple average of the RSA of the residues. The four clusters did not only differ in terms of accessibility, but also in their evolutionary divergence across homologues \cite{UTGES_2021_ANKS}, relative enrichment in human missense variation \cite{MACGOWAN_2024_VARIANTS}, and most importantly enrichment in known functional sites annotated in UniProt \cite{UNIPROT_2018_UNIPROT}. C1 sites were the most buried, conserved across homologues, depleted in missense variation and enriched in functional sites, whereas C4 sites were the complete opposite. Through a literature search, seventeen examples were identified of sites yet to be annotated in UniProt, but with literature evidence. These findings, further supported the hypothesis that these RSA-defined cluster labels recapitulate known functional sites and can be employed to rank the sites on a protein based on likelihood of function.

This information could be used in early-stage drug discovery to prioritise between targets and even between sites within the same target protein. Cluster 1 sites are $\approx$28 times more likely to be functional than sites with a C4 label. Consequently, targeting a C1 site is $\approx$28 more likely to have an effect on the target protein function. Considering the evolutionary divergence profile of the site can also provide insight into off-target effects, i.e., targeting a conserved site might also affect other members of the same protein family. In a similar manner, the missense enrichment profile of the site can shed light into potential pharmacogenetic effects. For example, a site with a high enrichment in genetic variation might not be desirable, as different individuals with distinct alleles might respond different to the drug targeting the site. This information could be leveraged in a different way if looking for a binding site for a molecular glue \cite{SCHREIBER_2021_GLUES}, targeted modification \cite{BREWER_2024_ATLAS, BREWER_2024_SMAD3, ZHAO_2024_TFEB} or degradation \cite{ZENGERLE_2015_BRD4, GADD_2017_PROTAC} of a protein. In this case, easily accessible sites which do not alter the protein function in any way might be of interest, and so highly accessible, divergent across homologues sites might be more attractive, i.e. C3-C4 sites.

\section{Fragment screening sites analysis}

XXX.

\section{Critical assessment of ligand binding site prediction tools}

XXX.

\section{Improving ligand binding site prediction tools}

XXX.

\section{Future steps}

XXX.

\section{Concluding remarks}

XXX.