\chapter{Introduction}

\section*{Preface}

In this Chapter, I introduce the basic concepts and methodologies necessary to understand this Thesis.

\section{The purpose of life}

``Long, long ago there was a time when nothing but mere matter existed in this world. In the teeming ooze, forms of a certain \textit{something} appeared, disappeared, and appeared again and one of them eventually survived. We know it as \textit{life}. The reason that \textit{life} ultimately survived was because it was in its nature \textit{to multiply}. \textit{Life} took new forms in order to multiply, adapting to every kind of environment, and culminating in \textit{us} today. Greater numbers, greater diversity, greater abundance. This is why we say that the purpose of \textit{life} is \textit{to multiply}.'' \cite{ISAYAMA_2021}

\section{Central dogma of molecular biology}

It is estimated based on geological \cite{SCHIDLOWSKI_1979_LIFE}, fossil \cite{SCHOPF_2007_LIFE} and phylogenetic \cite{BETTS_2018_LIFE} analyses that the origin of life in our planet Earth dates back to 3.7-4.0 billion years ago. Since then, \textit{life} has not just survived, but adapted and evolved to give raise to a gargantuan estimated biodiversity of 8.7 million eukaryotic species \cite{MORA_2011_SPECIES} and upward of 1 trillion microbial species \cite{HUG_2016_SPECIES, LOCEY_2016_SPECIES} with the vast majority of these still to be described \cite{COSTELLO_2013_SPECIES}. Despite the immense variation across species in terms of reproductive strategies, morphological, metabolic, behavioural traits or ecological niche, there is one thing \textit{all} species have in common: nucleic acids \cite{KOONIN_2011_LIFE}. All living species rely on nucleic acids, mostly deoxyribonucleic acid (DNA), except for some viruses \cite{KOONIN_2006_VIRUS} and viroids \cite{NAVARRO_2021_VIROIDS} that use ribonucleic acid (RNA), to store their genetic information. This information flows sequentially from DNA to RNA through the process of transcription and from RNA to protein through translation. This flow of molecular information is known as the \textit{Central Dogma of Molecular Biology} \cite{CRICK_1958_DOGMA, CRICK_1970_DOGMA}.

\section{The genetic code}

DNA is a polymer composed of two polynucleotide chains that coil around each other to form a double helix \cite{WATSON_1953_DNA}. These polymer chains are formed by simpler units called nucleotides. Each nucleotide presents a common scaffold formed of a deoxyribose sugar and a phosphate and a variable nitrogen-containing nucleobase. There are four different bases: adenine (A), thymine (T), cytosine (C) and guanine (G). The information stored in DNA gets transferred to RNA through the process of transcription. RNA tends to adopt a single-stranded conformation and is also formed by nucleotides. These nucleotides differ from DNA ones in that they present a ribose sugar, instead of deoxyribose, and an alternative uracil (U) nucleobase instead of thymine \cite{LEVENE_1909_NUCLEICS}.

Messenger RNA (mRNA) corresponds to the genetic sequence of a gene and is read by the ribosomal macromolecular machinery in the process of protein synthesis, or translation. In this process, a peptide chain is formed by linking amino acids in the order specified by the codons in the mRNA \cite{CRICK_1957_CODE}. A codon is a set of three nucleotides that corresponds to one of the twenty canonical amino acids. The equivalence between these codons and the amino acid they encode is known as the \textit{Genetic Code} \cite{GAMOW_1954_CODE}.

\section{Genetic variation}

Genetic variation is the difference in DNA sequence between individuals or populations of the same species. The main source of genetic variation is \textit{de novo} mutation. Mutations are changes in a genetic sequence that usually arise during DNA replication due to errors made by the imperfect replication machinery. Mutation can also occur as a result of damage to DNA, e.g., ultraviolet radiation, or during the repair process of such damage. Genetic variation can affect a single nucleotide in the sequence, i.e., a single nucleotide polymorphism (SNP), or multiple nucleotides (MNP), or larger DNA regions, even entire chromosomes, e.g., insertion, deletion, translocation, or fusion. SNPs are the only type of genetic variation relevant for the research described in this Thesis.

\subsection{Types of genetic variation}

\subsubsection{Genomic location}

Based on genomic location, genetic variation can be classified into \textit{coding} variation if it affects the mRNA that codes for the protein sequence. Alternatively, \textit{non-coding} variants are those that affect other regions that do not code for a protein product, such as introns, intergenic regions, promoters, enhancers or other regulatory elements.

\subsubsection{Effect on coding sequence}

The genetic code is \textit{degenerate} or redundant, as there are 4 $\times$ 4 $\times$ 4 = 64 codons coding for only twenty amino acids. For this reason, a change in the coding DNA sequence is not always reflected in the protein sequence. Mutations that due to the redundancy in the genetic code do not alter the protein sequence are called synonymous or silent. Nonsynonymous mutations \textit{do} change the protein sequence and can be further classified into: missense, nonsense, nonstop and frameshift mutations. Missense mutations are those that replace one of the twenty amino acids by a different one. Missense variants can be conservative, if the interchanged residues present similar physicochemical properties, e.g., leucine $\rightarrow$ isoleucine, or they can be non-conservative or radical if the exchanged amino acids are biochemically different, e.g., lysine $\rightarrow$ threonine. Nonsense mutations replace one of the twenty amino acids by one of the three STOP codons, resulting in an early termination of the peptide chain. Nonstop variants are the exact opposite and exchange the original stop codon by one of the twenty amino acids thus resulting in an abnormally elongated protein. Finally, frameshift mutations result from the insertion or deletion of nucleotides that are not a multiple of three. When this happens, the frame on which the translation machinery reads the mRNA is shifted and a completely different protein product is obtained.

While missense mutations, which simply replace one amino acid by another and can be conservative, have a limited effect on protein sequence and structure, nonsense, nonstop and frameshift mutation have more drastic consequences. Because of this, missense variants tend to be more tolerated and are observed with higher frequency in the general population \cite{COULTER_2004_MUTATIONS}.

\subsubsection{Impact on phenotype}

Genetic variants can also be classified based on the effect they have on the phenotype, or clinical significance, which usually corresponds to an effect on the concentration, structure, function or activity rate of a protein \cite{VIHINEN_2022_VARIATION}. Mutations that do not have a harmful effect on the protein are called neutral or benign. Since neutral variants have no noticeable effect on the \textit{fitness} \cite{DARWIN_1859_ORIGIN}, i.e., the ability to leave offspring, they are not under selective pressure and consequently roam around in the general population \cite{KIMURA_1968_NEUTRAL}. Conversely, pathogenic variants disrupt biological processes and eventually result in disease. Disease severity will dictate the strength with which natural selection acts upon the causing variant and therefore its frequency in the population. Mutations affecting genes needed for development and survival, or essential genes, might have lethal effects and never be observed in the population \cite{GLUECKSOHN_1963_LETHALITY}.

It is estimated that only 2\% of the more than 4 million observed human missense variants have been clinically classified as pathogenic or benign \cite{LEK_2016_EXAC}. Variants of unknown significance (VUS) therefore represent the vast majority of observed missense variants and the prediction of their effect on fitness is an important ongoing challenge in human genetics \cite{MCLAREN_2016_VEP}. Several methods exploiting different technologies have been developed over the years to tackle this challenge with SIFT \cite{KUMAR_2009_SIFT}, PolyPhen \cite{ADZHUBEI_2013_POLYPHEN} and the recent AlphaMissense \cite{CHENG_2023_ALPHAMISSENSE} being some of the most relevant ones.

\subsection{Variation is constrained}

XXX.

\section{Proteins}

Proteins are molecular machines that are involved in virtually all cellular processes including cell division, immune response or metabolism. They result from the process of translation of mRNA. Proteins are natural polymers formed of smaller monomers called amino acids linked to each other through peptide bonds.

\subsection{Amino acid structure}

There are twenty canonical amino acids that are found in all protein sequences. They receive this name because of their chemical structure, which includes both an amino and carboxylic acid functional groups. \autoref{fig:amino_acid} shows the general structure of an amino acid. A carbon atom is found in the centre which binds covalently to four different groups. This carbon is known as $\alpha$-carbon (CA) and is attached to the amino group (--NH2), the carboxyl group (--COOH), a hydrogen atom (H) and a side chain (R) that differs across the twenty amino acid residues.

\begin{figure}[htbp!]
    \centering
    \includegraphics[width=0.50\textwidth]{figures/ch_INTRO/PNG/amino_acid.png}
    \caption[Amino acid structure]{\textbf{Amino acid structure.} All twenty amino acids share this common structure formed by the $\alpha$-carbon (CA) chemically attached to the amino (NH2) and carboxyl (COOH) groups, a hydrogen atom (H), and a side chain (R). The side chain is different and defines the amino acids.}
    \label{fig:amino_acid}
\end{figure}

\subsection{Amino acid properties}

The different side chains of the amino acids confer them different physicochemical properties \cite{SNEATH_1966_PROPERTIES}. \autoref{fig:properties} illustrates the ten main properties, the relationship between them and which amino acids present them \cite{TAYLOR_1986_PROPERTIES}. The three most important properties setting amino acids apart are hydrophobicity, polarity and size. Hydrophobic residues present side chains that are less soluble in water and therefore tend to be located in the interior protein core, whereas hydrophilic residues are present on the surface. Polar residue side chains contain electronegative atoms like nitrogen (N) or oxygen (O) that favour interaction with water and other polar molecules. Size is also relevant as there is a big difference in volume between the amino acids ranging from 60 \AA{}\textsuperscript{3} (Glycine) to $>$200 \AA{}\textsuperscript{3} (Tryptophan). Within these three main categories, other subsets can be found as aliphatic (open chain of carbon atoms), aromatic (closed carbon chain), positively and negatively charged, tiny or proline. Proline has its own category because of its unique cyclical side chain which links back to the backbone \cite{ZVELEBIL_1987_SSPRED}.

These physicochemical properties of amino acids are crucial to understand the arrangement of protein atoms in three-dimensional (3D) space and their conservation across evolutionarily related proteins is the basis for traditional sequence analysis and protein structure prediction \cite{CHOTHIA_1986_DIVERGENCE}.

\begin{figure}[htb!]
    \centering
    \includegraphics[width=\textwidth]{figures/ch_INTRO/PNG/properties.png}
    \caption[Amino acid properties]{\textbf{Amino acid properties.} Taylor Venn diagram illustrating the different physicochemical properties of the twenty proteinogenic amino acids. Adapted from Livingstone and Barton \cite{LIVINGSTONE_1993_MSA}.}
    \label{fig:properties}
\end{figure}

\subsection{Substitution matrices}

Similar or identical protein sequences carrying out related functions and displaying a comparable 3D structure can be found within a genome (\textit{paralogous} sequences) and across species (\textit{ortholoous} sequences). These proteins are evolutionarily related, i.e., \textit{homologous}, and their origin can be traced back in time to a common ancestor. The comparative analysis of such related sequences provides insight into the evolutionary history of a given set of related sequences, or family \cite{BARTON_1990_MSA}. Amino acid substitution matrices can be calculated by quantifying the differences between closely related sequences. These matrices indicate the likelihood of observing transitions at a given protein position between the different amino acids. Transitions between amino acids with similar physicochemical properties, e.g., aspartate $\rightarrow$ glutamate, are less likely to alter the protein structure and are therefore observed with higher frequency. The Point Accepted Mutation (PAM) \cite{DAYHOFF_1978_PAM} and Block Substitution Matrix (BLOSUM) \cite{HENIKOFF_1992_BLOSUM} are some of the more relevant substitution matrices and serve as a scoring function for the construction of alignment of multiple sequences (MSA) that are related in evolution \cite{BARTON_1987_MSA}.

\subsection{Multiple sequence alignment}

In an MSA, multiple sequences (\textit{rows}) are piled up and aligned in a way that positions across proteins that are thought to be homologous, i.e., share common ancestry, are located on the same column \cite{NEEDLEMAN_1970_MSA}. Through time, sequences diverge and might suffer point mutations, insertions or deletions. To accommodate for this, aligners introduce \textit{gaps} (--) \cite{SMITH_1981_MSA}. By looking at the distribution of amino acid residues across columns in the alignment patterns of amino acid conservation can be observed when residues or their physicochemical properties are invariant across sequences or rows. Columns that present little or no variation are called \textit{conserved} whereas columns that present a wider variety of amino acids with different properties are called \textit{unconserved} or \textit{divergent} \cite{LIVINGSTONE_1993_CONS}. Many methods for the alignment of multiple sequence have been developed over the years with Clustal \cite{HIGGINS_1988_CLUSTAL, HIGGINS_1992_CLUSTALV, THOMPSON_1994_CLUSTALW, JEANMOUGIN_1998_CLUSTALX, SIEVERS_2011_CLUSTALO}, MAFFT \cite{KATOH_2002_MAFFT, KATOH_2008_MAFFT, KATOH_2013_MAFFT} or MUSCLE \cite{EDGAR_2004_MUSCLE, EDGAR_2022_MUSCLE5} being some of the most widely used.

\subsection{Amino acid conservation}

Amino acid conservation observed in MSAs is evidence of evolutionary constraint. Throughout evolution, conserved positions have remained fixed due to their functional or structural relevance while divergent positions accumulate substitutions resulting in variability in amino acid residues across proteins within the same family \cite{ZUCKERKANDL_1965_DIVERGENCE}. There is not an obvious way to quantify amino acid conservation. Because of this, several scores exploring different approaches have been developed through the years \cite{VALDAR_2002_SCORES}. Some of these scores consider amino acids as symbols and use their relative frequencies \cite{WU_1970_SCORE, JORES_1990_SCORE, LOCKLESS_1999_SCORE}, or entropy \cite{SANDER_1991_SCORE, SHENKIN_1991_SCORE, GERSTEIN_1995_SCORE} to score conservation. Others focus on their stereochemical properties \cite{TAYLOR_1986_PROPERTIES, ZVELEBIL_1987_PREDICTION}, use mutation data \cite{KARLIN_1996_SCORE, THOMPSON_1997_SCORE, LANDGRAF_1999_SCORE, PILPEL_1999_SCORE, ARMON_2001_SCORE, VALDAR_2001_SCORE} or combine amino acid properties and symbol entropy \cite{WILLIAMSON_1995_SCORE, MIRNY_1999_SCORE}.

The score developed by Shenkin \textit{et al.} \cite{SHENKIN_1991_SCORE} is based on Shannon's Entropy ($S$) which is calculated with \autoref{eq:entropy_shannon2} \cite{SHANNON_1948_ENTROPY}. The proportion within an alignment column of each amino acid $i$ of the $K$ = 20 naturally occurring amino acids is denoted by $p_i$. The Shenkin Score, $V_{Shenkin}$, described in \autoref{eq:shenkin}, measures divergence and increases as amino acid variability within a column grows. In a fully conserved position, where all amino acids are the same, entropy is minimum ($S$ = 0) and so is divergence ($V_{Shenkin}$ = 6). Conversely, in a fully divergent position, where all amino acids are equally represented, entropy is maximum ($S \approx$ 4.32) and so is divergence ($V_{Shenkin}$ = 120). Utg√©s \textit{et al.} \cite{UTGES_2021_ANKS} defined a version of this score, $N_{Shenkin}$ (\autoref{eq:shenkin_norm}) which normalises the original score by the minimum and maximum scores within the alignment and ranges 0-100. This normalised score is employed in this Thesis to measure amino acid divergence.

\begin{equation}
S = - \sum_{i=1}^{K} p_i \log_2(p_i)
\label{eq:entropy_shannon2}
\end{equation}
\myequations{Shannon's Entropy}

\begin{equation}
V_{Shenkin} = 2^S \times 6
\label{eq:shenkin}
\end{equation}
\myequations{Shenkin divergence score}

\begin{equation}
N_{Shenkin} = \frac{V_{Shenkin} - V_{Shenkin_{\text{min}}}}{V_{Shenkin_{\text{max}}} - V_{Shenkin_{\text{min}}}}
\label{eq:shenkin_norm}
\end{equation}
\myequations{Normalised Shenkin divergence score}

\subsection{Protein sequence}

XXX.

\subsubsection{Conservation and function}

XXX.

\subsection{Protein structure}

XXX.

\subsubsection{Primary structure}

XXX.

\subsubsection{Secondary structure}

XXX.

\subsubsection{Tertiary structure}

XXX.

\subsubsection{Quaternary structure}

XXX.

\subsection{Protein structure determination}

XXX.

\subsubsection{Nuclear magnetic resonance}

XXX.

\subsubsection{Cryo-electron microscopy}

XXX.

\subsubsection{X-ray crystallography}

XXX.

\subsection{Protein structure prediction}

XXX.

\subsubsection{Homology-based}

XXX.

\subsubsection{De novo}

XXX.

\subsection{Protein structure characterisation}

XXX.

\subsubsection{Secondary structure}

XXX.

\subsubsection{Solvent accessibility}

XXX.

\subsubsection{Hydrophobicity}

XXX.

\subsubsection{Charge}

XXX.

\subsubsection{Binding pockets}

XXX.

\section{The conservation plane}

XXX.

\section{Drug discovery}

XXX.

\subsection{Drug discovery pipeline}

XXX.

\subsubsection{Target identification}

XXX.

\subsubsection{Target validation}

XXX.

\subsubsection{Lead identification}

XXX.

\subsubsection{Lead optimisation}

XXX.

\subsubsection{Pre-clinical development}

XXX.

\subsubsection{Clinical studies}

XXX.

\subsubsection{Drug approval}

XXX.

\subsection{Fragment screening}

XXX.

\section{Machine learning}

XXX.

\section{Databases}

XXX.

\subsection{UniProt}

XXX.

\subsection{PDB}

XXX.

\subsection{gnomAD}

XXX.

\section{Thesis scope}

XXX.


% WHAT IS LEFT?